#!/bin/bash

DIR_IN=/expanse/projects/sebat1/miladm/UCSD/LONG_READ_COHORT/TRs_LongTR/FAM
VCF_IN=$DIR_IN/longtr_merge.sorted.vep.addInfo.vcf.gz
VEP_OUT=vep.tsv

date

echo "Making VEP table..."

echo -n -e "CHROM\tPOS\tEND\tID\tGENCODE\tPERIOD\tZS_SAMPLES\tZS_SAMPLES_NUM\tZS_SAMPLES_PB\tZS_SAMPLES_ONT\tSN_GB\tDENOVO\tDENOVO_GB\tDENOVO_NUM\tDENOVO_MAT\tDENOVO_MAT_GB\tDENOVO_MAT_NUM\tDENOVO_PAT\tDENOVO_PAT_GB\tDENOVO_PAT_NUM\tLARGE25_DEV_GB\tLARGE50_DEV_GB\tLARGE100_DEV_GB\tLARGE500_DEV_GB\tMEAN_GB\tSTD_GB\tQ90_SAMPLES_GB\tHP1_SAMPLES_PDP\tHP2_SAMPLES_PDP\tHEM_SAMPLES_DP\tNGT_SAMPLES\tGNOCCHI\tFB_PR_ENH_M\tFB_PR_ENH_F\tFANTOM_ENH\tEV_CONS_EL\tConsequence\tIMPACT\tSYMBOL\tGENE\tBIOTYPE\tDISTANCE\tTSSDistance\tEXON\tINTRON\n" > $VEP_OUT

bcftools +split-vep -X -c Gene,BIOTYPE,SYMBOL,IMPACT,EXON,INTRON,Consequence,DISTANCE,STRAND,OverlapBP,OverlapPC,TSSDistance,GENCODE,GNOCCHI,FB_PR_ENH_M,FB_PR_ENH_F,FANTOM_ENH,EV_CONS_EL -f '%CHROM\t%POS\t%END\t%ID\t%GENCODE\t%PERIOD\t%ZS_SAMPLES\t%ZS_SAMPLES_NUM\t%ZS_SAMPLES_PB\t%ZS_SAMPLES_ONT\t%SN_GB\t%DENOVO\t%DENOVO_GB\t%DENOVO_NUM\t%DENOVO_MAT\t%DENOVO_MAT_GB\t%DENOVO_MAT_NUM\t%DENOVO_PAT\t%DENOVO_PAT_GB\t%DENOVO_PAT_NUM\t%LARGE25_DEV_GB\t%LARGE50_DEV_GB\t%LARGE100_DEV_GB\t%LARGE500_DEV_GB\t%MEAN_GB\t%STD_GB\t%Q90_SAMPLES_GB\t%HP1_SAMPLES_PDP\t%HP2_SAMPLES_PDP\t%HEM_SAMPLES_DP\t%NGT_SAMPLES\t%GNOCCHI\t%FB_PR_ENH_M\t%FB_PR_ENH_F\t%FANTOM_ENH\t%EV_CONS_EL\t%Consequence\t%IMPACT\t%SYMBOL\t%Gene\t%BIOTYPE\t%DISTANCE\t%TSSDistance\t%EXON\t%INTRON' $VCF_IN >> $VEP_OUT

date

