{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "20179b83-b2a7-4e8e-bc1d-039539c1a4bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "#import matplotlib.pyplot as plt\n",
    "#import seaborn as sb\n",
    "import numpy as np\n",
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)\n",
    "from scipy.stats import binomtest\n",
    "#from scipy.stats import chisquare, binomtest\n",
    "#from sklearn.linear_model import LinearRegression\n",
    "#import statsmodels.api as sm\n",
    "#import warnings\n",
    "#warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)\n",
    "#from sklearn.exceptions import ConvergenceWarning\n",
    "#warnings.simplefilter(\"ignore\", category=ConvergenceWarning)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e2138513-690b-4b2c-8f4b-e6c8ba0fe1d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir_data = '../data'\n",
    "dir_out = 'outputs'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "90ac9ee1-58db-4340-8bfb-5d6ca81d0973",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>bp_dev_h1</th>\n",
       "      <th>bp_dev_h2</th>\n",
       "      <th>bp_dev_h0</th>\n",
       "      <th>rn_h1</th>\n",
       "      <th>rn_h2</th>\n",
       "      <th>rn_h0</th>\n",
       "      <th>ps_h1</th>\n",
       "      <th>ps_h2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147911920</td>\n",
       "      <td>147912179</td>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>.;.;.;91|95|100|99|96|100|99|98|98|96|96|97|94...</td>\n",
       "      <td>.;.;.;64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|...</td>\n",
       "      <td>.;0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3...</td>\n",
       "      <td>.;.;.;32;.;.;30;.;31,30;34;27,27,27,29;.;.;.;3...</td>\n",
       "      <td>.;.;.;31,43,30;.;.;30,28,28,31;.;30,30,32;34,2...</td>\n",
       "      <td>.;51,51,51,57,51,51,52;30;.;.;30;.;34,31;.;.;....</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/27724288/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/75497573/ccs,m5434...</td>\n",
       "      <td>.;m54340U_210523_044800/52758275/ccs,m54340U_2...</td>\n",
       "      <td>.;.;.;10185;.;.;10215;.;10215,10215;10185;1018...</td>\n",
       "      <td>.;.;.;10185,10185,10185;.;.;10215,10215,10215,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149505139</td>\n",
       "      <td>149505306</td>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>.;.;.;8|2|2|1|0|0|0|0|0|1|0|7|1|0|3|2|1|4|0|0|...</td>\n",
       "      <td>.;.;.;62|51|80|91|75|95|89|98|99|98|97|99|99|9...</td>\n",
       "      <td>.;0|2|0|1|0|1|6|2|1|8|9|13|6|3|4|8|5|0|1|0|0|0...</td>\n",
       "      <td>.;.;.;1;.;.;.;.;.;2;.;.;.;.;0,0,2;.;.;-1,1,0;....</td>\n",
       "      <td>.;.;.;0;.;.;.;.;.;.;.;.;.;.;.;.;.;1,0,1;.;.;.;...</td>\n",
       "      <td>.;0,2;3;.;0,0;0;-1,1,0,1;0,-2,0;0,0,-1,0;0,0,1...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/128059025/ccs;.;.;...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/133105829/ccs;.;.;...</td>\n",
       "      <td>.;m54340U_200806_070905/88277013/ccs,m54340U_2...</td>\n",
       "      <td>.;.;.;10185;.;.;.;.;.;10185;.;.;.;.;10185,1018...</td>\n",
       "      <td>.;.;.;10185;.;.;.;.;.;.;.;.;.;.;.;.;.;10185,10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149631582</td>\n",
       "      <td>149631792</td>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;95|91|35|70|18|12|12|62|41|61|26|6...</td>\n",
       "      <td>.;.;.;.;73|60|29|4|2|7|3|1|1|1|1|0|0|0|8|17|11...</td>\n",
       "      <td>5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...</td>\n",
       "      <td>.;.;.;.;.;.;-21,-21,-21;.;-21,-15;.;15,9;.;0;....</td>\n",
       "      <td>.;.;.;.;-21;.;.,9,0;.;-21,-21;.;-21;.;-11;.;.;...</td>\n",
       "      <td>-21,-19;-21,-21,-21,-21,-18,-21,-21,-25;.;-19;...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/110166254/cc...</td>\n",
       "      <td>.;.;.;.;m54340U_200716_153006/94110941/ccs;.;m...</td>\n",
       "      <td>m54340U_200716_001732/174262380/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215,10215;.;10215,10215;.;...</td>\n",
       "      <td>.;.;.;.;10185;.;10215,10215,10215;.;10215,1021...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150363318</td>\n",
       "      <td>150363530</td>\n",
       "      <td>MAMLD1_5UTR</td>\n",
       "      <td>.;.;.;3|2|5|5|4|1|3|3|1|0|0|0|0|0|0|0|0|0|0|4|...</td>\n",
       "      <td>.;.;.;.;.;.;71|65|46|44|7|4|0|0|0|0|0|2|64|66|...</td>\n",
       "      <td>0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...</td>\n",
       "      <td>.;.;.;0;.;.;-1;.;0,0;32,33,35,29,29,34,34;1,-1...</td>\n",
       "      <td>.;.;.;.;.;.;1,3;.;.;1,2,2;0,1,3;.;1,1;.;1,0;.;...</td>\n",
       "      <td>-1;0,0,0;.;.;0,3;4,1;1,0,1,1;0,1,0,0;.;.;.;37,...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/71501671/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_200806_223416/98175724/ccs...</td>\n",
       "      <td>m54340U_200716_001732/24969595/ccs;m54340U_210...</td>\n",
       "      <td>.;.;.;10185;.;.;10215;.;10215,10215;10185,1018...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215;.;.;10185,10185,10185;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150568622</td>\n",
       "      <td>150568662</td>\n",
       "      <td>MTM1_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;3|13|4|2|0|0,98|99|98|95|97|...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;100|100|100|100|100|100,45|5...</td>\n",
       "      <td>1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;0,0;0,0;.;0,0,0;.;.;.;.;0,0;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;0,2;0;.;0,-1;.;.;.;.;1;.;.;....</td>\n",
       "      <td>0,0,0,0,0;0,0,0;0;.;0,0;.;2,0,0,0;.;.,0,0,1,0,...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/158271...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/107350...</td>\n",
       "      <td>m54340U_200716_001732/162859070/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185,10185;10185,10185;.;10...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185,10185;10185;.;10185,10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150693397</td>\n",
       "      <td>150693530</td>\n",
       "      <td>MTMR1_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;2|1|2|10|11|5|3|2|3|1|1|5|18...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;2|0|0|5|2|5|12|23|34|4...</td>\n",
       "      <td>6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;1,0,1,0,2,0;1,2,2,1,1;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;-1,1,-1;.;0,2,3;.;.;0,...</td>\n",
       "      <td>1,0,0;0,0,0,0,2,0,1;.;0;0,0;0;0,0,0,1,1;0,-1,0...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/609751...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;m54340U_200611_224317/...</td>\n",
       "      <td>m54340U_200716_001732/53740679/ccs,m54340U_200...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185,10185,10185,10185,1018...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;.;.;10185,10185,10185;.;10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150898590</td>\n",
       "      <td>150898668</td>\n",
       "      <td>CD99L2_5UTR</td>\n",
       "      <td>.;.;.;8|7|9|52|43|81|48|25;.;.;88|47|83|82|98|...</td>\n",
       "      <td>.;.;.;2|0|0|1|0|0|0|0;.;.;0|0|0|0|0|0|0|13,62|...</td>\n",
       "      <td>0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69;0|0|0|0|1|1...</td>\n",
       "      <td>.;.;.;1;.;.;0,1;.;0;0,0,-1;0,0;.;0,0,1;.;.;.;....</td>\n",
       "      <td>.;.;.;1;.;.;0,0,0;.;.;0,0,1;2,1;.;1,0;.;.;.;.;...</td>\n",
       "      <td>0,0;0,0;.;1,3;2,0,1;.;.;-2,2,0;.;.;.;1,0;.;0;0...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/24904108/ccs;.;.;m...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/134351559/ccs;.;.;...</td>\n",
       "      <td>m54340U_200716_001732/156107929/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;10185;.;.;10215,10215;.;10215;10185,1018...</td>\n",
       "      <td>.;.;.;10185;.;.;10215,10215,10215;.;.;10185,10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150983336</td>\n",
       "      <td>150983376</td>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;100|100|100|100|100,11|61|12|40|64...</td>\n",
       "      <td>.;.;.;.;.;.;29|50|50|41;.;93|98|98|85|95;70|67...</td>\n",
       "      <td>.;2|11|5|2|6,2|1|15|3|11,5|9|6|2|0;.;.;80|76|5...</td>\n",
       "      <td>.;.;.;.;.;.;0,1,0;.;.;0,0,0;0,0,0;.;0,0;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;0;0;0,0,0;.;2,0,0,0;.;0;0;.;-1...</td>\n",
       "      <td>.;0,0,0;.;.;0;0,0,0;.;0,0,0;0;0,0;.;0,0,0;.;0,...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/127796270/cc...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_200806_223416/168034751/cc...</td>\n",
       "      <td>.;m54340U_210523_044800/100468860/ccs,m54340U_...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215,10215;.;.;10185,10185,...</td>\n",
       "      <td>.;.;.;.;.;.;10215;.;10215;10185;10185,10185,10...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom        pos        end      region_id  \\\n",
       "1   chrX  147911920  147912179      FMR1_5UTR   \n",
       "7   chrX  149505139  149505306       IDS_5UTR   \n",
       "8   chrX  149631582  149631792  TMEM185A_5UTR   \n",
       "9   chrX  150363318  150363530    MAMLD1_5UTR   \n",
       "10  chrX  150568622  150568662      MTM1_5UTR   \n",
       "11  chrX  150693397  150693530     MTMR1_5UTR   \n",
       "12  chrX  150898590  150898668    CD99L2_5UTR   \n",
       "13  chrX  150983336  150983376     HMGB3_5UTR   \n",
       "\n",
       "                                            methyl_h1  \\\n",
       "1   .;.;.;91|95|100|99|96|100|99|98|98|96|96|97|94...   \n",
       "7   .;.;.;8|2|2|1|0|0|0|0|0|1|0|7|1|0|3|2|1|4|0|0|...   \n",
       "8   .;.;.;.;.;.;95|91|35|70|18|12|12|62|41|61|26|6...   \n",
       "9   .;.;.;3|2|5|5|4|1|3|3|1|0|0|0|0|0|0|0|0|0|0|4|...   \n",
       "10  .;.;.;.;.;.;.;.;.;3|13|4|2|0|0,98|99|98|95|97|...   \n",
       "11  .;.;.;.;.;.;.;.;.;2|1|2|10|11|5|3|2|3|1|1|5|18...   \n",
       "12  .;.;.;8|7|9|52|43|81|48|25;.;.;88|47|83|82|98|...   \n",
       "13  .;.;.;.;.;.;100|100|100|100|100,11|61|12|40|64...   \n",
       "\n",
       "                                            methyl_h2  \\\n",
       "1   .;.;.;64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|...   \n",
       "7   .;.;.;62|51|80|91|75|95|89|98|99|98|97|99|99|9...   \n",
       "8   .;.;.;.;73|60|29|4|2|7|3|1|1|1|1|0|0|0|8|17|11...   \n",
       "9   .;.;.;.;.;.;71|65|46|44|7|4|0|0|0|0|0|2|64|66|...   \n",
       "10  .;.;.;.;.;.;.;.;.;100|100|100|100|100|100,45|5...   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;2|0|0|5|2|5|12|23|34|4...   \n",
       "12  .;.;.;2|0|0|1|0|0|0|0;.;.;0|0|0|0|0|0|0|13,62|...   \n",
       "13  .;.;.;.;.;.;29|50|50|41;.;93|98|98|85|95;70|67...   \n",
       "\n",
       "                                            methyl_h0  \\\n",
       "1   .;0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3...   \n",
       "7   .;0|2|0|1|0|1|6|2|1|8|9|13|6|3|4|8|5|0|1|0|0|0...   \n",
       "8   5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...   \n",
       "9   0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...   \n",
       "10  1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...   \n",
       "11  6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...   \n",
       "12  0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69;0|0|0|0|1|1...   \n",
       "13  .;2|11|5|2|6,2|1|15|3|11,5|9|6|2|0;.;.;80|76|5...   \n",
       "\n",
       "                                            bp_dev_h1  \\\n",
       "1   .;.;.;32;.;.;30;.;31,30;34;27,27,27,29;.;.;.;3...   \n",
       "7   .;.;.;1;.;.;.;.;.;2;.;.;.;.;0,0,2;.;.;-1,1,0;....   \n",
       "8   .;.;.;.;.;.;-21,-21,-21;.;-21,-15;.;15,9;.;0;....   \n",
       "9   .;.;.;0;.;.;-1;.;0,0;32,33,35,29,29,34,34;1,-1...   \n",
       "10  .;.;.;.;.;.;.;.;.;0,0;0,0;.;0,0,0;.;.;.;.;0,0;...   \n",
       "11  .;.;.;.;.;.;.;.;.;1,0,1,0,2,0;1,2,2,1,1;.;.;.;...   \n",
       "12  .;.;.;1;.;.;0,1;.;0;0,0,-1;0,0;.;0,0,1;.;.;.;....   \n",
       "13  .;.;.;.;.;.;0,1,0;.;.;0,0,0;0,0,0;.;0,0;.;.;.;...   \n",
       "\n",
       "                                            bp_dev_h2  \\\n",
       "1   .;.;.;31,43,30;.;.;30,28,28,31;.;30,30,32;34,2...   \n",
       "7   .;.;.;0;.;.;.;.;.;.;.;.;.;.;.;.;.;1,0,1;.;.;.;...   \n",
       "8   .;.;.;.;-21;.;.,9,0;.;-21,-21;.;-21;.;-11;.;.;...   \n",
       "9   .;.;.;.;.;.;1,3;.;.;1,2,2;0,1,3;.;1,1;.;1,0;.;...   \n",
       "10  .;.;.;.;.;.;.;.;.;0,2;0;.;0,-1;.;.;.;.;1;.;.;....   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;-1,1,-1;.;0,2,3;.;.;0,...   \n",
       "12  .;.;.;1;.;.;0,0,0;.;.;0,0,1;2,1;.;1,0;.;.;.;.;...   \n",
       "13  .;.;.;.;.;.;.;.;0;0;0,0,0;.;2,0,0,0;.;0;0;.;-1...   \n",
       "\n",
       "                                            bp_dev_h0  \\\n",
       "1   .;51,51,51,57,51,51,52;30;.;.;30;.;34,31;.;.;....   \n",
       "7   .;0,2;3;.;0,0;0;-1,1,0,1;0,-2,0;0,0,-1,0;0,0,1...   \n",
       "8   -21,-19;-21,-21,-21,-21,-18,-21,-21,-25;.;-19;...   \n",
       "9   -1;0,0,0;.;.;0,3;4,1;1,0,1,1;0,1,0,0;.;.;.;37,...   \n",
       "10  0,0,0,0,0;0,0,0;0;.;0,0;.;2,0,0,0;.;.,0,0,1,0,...   \n",
       "11  1,0,0;0,0,0,0,2,0,1;.;0;0,0;0;0,0,0,1,1;0,-1,0...   \n",
       "12  0,0;0,0;.;1,3;2,0,1;.;.;-2,2,0;.;.;.;1,0;.;0;0...   \n",
       "13  .;0,0,0;.;.;0;0,0,0;.;0,0,0;0;0,0;.;0,0,0;.;0,...   \n",
       "\n",
       "                                                rn_h1  \\\n",
       "1   .;.;.;m54340U_200716_153006/27724288/ccs;.;.;m...   \n",
       "7   .;.;.;m54340U_200716_153006/128059025/ccs;.;.;...   \n",
       "8   .;.;.;.;.;.;m54340U_210703_193517/110166254/cc...   \n",
       "9   .;.;.;m54340U_200716_153006/71501671/ccs;.;.;m...   \n",
       "10  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/158271...   \n",
       "11  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/609751...   \n",
       "12  .;.;.;m54340U_200716_153006/24904108/ccs;.;.;m...   \n",
       "13  .;.;.;.;.;.;m54340U_210703_193517/127796270/cc...   \n",
       "\n",
       "                                                rn_h2  \\\n",
       "1   .;.;.;m54340U_200716_153006/75497573/ccs,m5434...   \n",
       "7   .;.;.;m54340U_200716_153006/133105829/ccs;.;.;...   \n",
       "8   .;.;.;.;m54340U_200716_153006/94110941/ccs;.;m...   \n",
       "9   .;.;.;.;.;.;m54340U_200806_223416/98175724/ccs...   \n",
       "10  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/107350...   \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;m54340U_200611_224317/...   \n",
       "12  .;.;.;m54340U_200716_153006/134351559/ccs;.;.;...   \n",
       "13  .;.;.;.;.;.;m54340U_200806_223416/168034751/cc...   \n",
       "\n",
       "                                                rn_h0  \\\n",
       "1   .;m54340U_210523_044800/52758275/ccs,m54340U_2...   \n",
       "7   .;m54340U_200806_070905/88277013/ccs,m54340U_2...   \n",
       "8   m54340U_200716_001732/174262380/ccs,m54340U_20...   \n",
       "9   m54340U_200716_001732/24969595/ccs;m54340U_210...   \n",
       "10  m54340U_200716_001732/162859070/ccs,m54340U_20...   \n",
       "11  m54340U_200716_001732/53740679/ccs,m54340U_200...   \n",
       "12  m54340U_200716_001732/156107929/ccs,m54340U_20...   \n",
       "13  .;m54340U_210523_044800/100468860/ccs,m54340U_...   \n",
       "\n",
       "                                                ps_h1  \\\n",
       "1   .;.;.;10185;.;.;10215;.;10215,10215;10185;1018...   \n",
       "7   .;.;.;10185;.;.;.;.;.;10185;.;.;.;.;10185,1018...   \n",
       "8   .;.;.;.;.;.;10215,10215,10215;.;10215,10215;.;...   \n",
       "9   .;.;.;10185;.;.;10215;.;10215,10215;10185,1018...   \n",
       "10  .;.;.;.;.;.;.;.;.;10185,10185;10185,10185;.;10...   \n",
       "11  .;.;.;.;.;.;.;.;.;10185,10185,10185,10185,1018...   \n",
       "12  .;.;.;10185;.;.;10215,10215;.;10215;10185,1018...   \n",
       "13  .;.;.;.;.;.;10215,10215,10215;.;.;10185,10185,...   \n",
       "\n",
       "                                                ps_h2  \n",
       "1   .;.;.;10185,10185,10185;.;.;10215,10215,10215,...  \n",
       "7   .;.;.;10185;.;.;.;.;.;.;.;.;.;.;.;.;.;10185,10...  \n",
       "8   .;.;.;.;10185;.;10215,10215,10215;.;10215,1021...  \n",
       "9   .;.;.;.;.;.;10215,10215;.;.;10185,10185,10185;...  \n",
       "10  .;.;.;.;.;.;.;.;.;10185,10185;10185;.;10185,10...  \n",
       "11  .;.;.;.;.;.;.;.;.;.;.;.;10185,10185,10185;.;10...  \n",
       "12  .;.;.;10185;.;.;10215,10215,10215;.;.;10185,10...  \n",
       "13  .;.;.;.;.;.;10215;.;10215;10185;10185,10185,10...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of samples: 270\n"
     ]
    }
   ],
   "source": [
    "### read methylation data and sample order files\n",
    "file_in = f'{dir_data}/methyl_data.tsv'\n",
    "df = pd.read_table(file_in, sep='\\t', header=0, keep_default_na=False)\n",
    "df = df.loc[df.region_id.str.contains('_5UTR')].copy()\n",
    "print('df:')\n",
    "display(df)\n",
    "\n",
    "file_sam = f'{dir_data}/methyl_data_sample_order.txt'\n",
    "with open(file_sam, 'r') as fh:\n",
    "    sample_order = fh.readline().split(',')[:-1] # last one is an empty string\n",
    "print(f'number of samples: {len(sample_order)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "12c461a2-c218-4473-8bf6-8067bff7d51f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>REACH000026_methyl_h1</th>\n",
       "      <th>REACH000026_methyl_h2</th>\n",
       "      <th>REACH000026_methyl_h0</th>\n",
       "      <th>REACH000026_bp_dev_h1</th>\n",
       "      <th>REACH000026_bp_dev_h2</th>\n",
       "      <th>REACH000026_bp_dev_h0</th>\n",
       "      <th>...</th>\n",
       "      <th>REACH000685_methyl_h2</th>\n",
       "      <th>REACH000685_methyl_h0</th>\n",
       "      <th>REACH000685_bp_dev_h1</th>\n",
       "      <th>REACH000685_bp_dev_h2</th>\n",
       "      <th>REACH000685_bp_dev_h0</th>\n",
       "      <th>REACH000685_rn_h1</th>\n",
       "      <th>REACH000685_rn_h2</th>\n",
       "      <th>REACH000685_rn_h0</th>\n",
       "      <th>REACH000685_ps_h1</th>\n",
       "      <th>REACH000685_ps_h2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147911920</td>\n",
       "      <td>147912179</td>\n",
       "      <td>FMR1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>13,17,33,29,16</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>658567b8-0bb0-48a8-a85d-619f26f01f60,ec6386b7-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149505139</td>\n",
       "      <td>149505306</td>\n",
       "      <td>IDS_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>11|0|0|0|0|15|0|0|0|4|89|91|2|0|0|0|1|9|93|89|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6,-14,-3,-4,-29,-24,-17,-8,-69</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>dbdc51c7-868d-4c1e-aef8-54b4be7616b1,ee73bd80-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chrX</td>\n",
       "      <td>149631582</td>\n",
       "      <td>149631792</td>\n",
       "      <td>TMEM185A_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-21,-19</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>100|4|30|7|40|97|0,0|0|0|0|2|0|0|1|0|0|0|0|2|2...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-23,-13,-2,-8,-21,.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0bdcf4ce-425a-4468-9796-785e0a2817f8,8d6dac54-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150363318</td>\n",
       "      <td>150363530</td>\n",
       "      <td>MAMLD1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>1|0|1|0|0|0|4|0|3|61|9|12|46|79|84|75|0|1|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-8,-3,-11,-2,-17,8,-5,-2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>b0ddb187-c71b-4f31-8e70-da57f39b8cff,9b1c13e4-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150568622</td>\n",
       "      <td>150568662</td>\n",
       "      <td>MTM1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,0,0,0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>40|36|9|41|79|30|22|2|45,4|2|0|0|0|0|5,0|4|0|1...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9,-1,-2,2,6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7178be63-b143-43d2-b646-7f250eee984a,3152af87-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150693397</td>\n",
       "      <td>150693530</td>\n",
       "      <td>MTMR1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1,0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>0|9|0|1|0|0|0|1|13|0|16|0|0|2|0|5,55|70|9|2|1|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-10,-2,5,0,-3,3,5,-5,-16,-4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>33acaac3-8252-404b-b9e8-e6c5fde090e9,21350c97-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150898590</td>\n",
       "      <td>150898668</td>\n",
       "      <td>CD99L2_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,0</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0|0|7|7|0|0|0|2,45|0|21|0|0|2|0|2</td>\n",
       "      <td>46|85,0|49|2|0|0|4|0|0,0|80|0|2|0|3|0|36,10|0</td>\n",
       "      <td>-3</td>\n",
       "      <td>4,0</td>\n",
       "      <td>-6,-6,-5,.</td>\n",
       "      <td>5af74e3f-fde8-469d-9c44-8ca2cce771d2</td>\n",
       "      <td>34a1e889-e5f6-4400-8f70-0d32c5afd4d6,7d1b38e0-...</td>\n",
       "      <td>1ae72adc-1d37-4141-bfa6-590e0ec211d1,46f0b406-...</td>\n",
       "      <td>10185</td>\n",
       "      <td>10185,10185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrX</td>\n",
       "      <td>150983336</td>\n",
       "      <td>150983376</td>\n",
       "      <td>HMGB3_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|12|0,0|0|0|0|2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1,-6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>519f9439-750b-421f-9828-4705fb07f845,94933348-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows × 2974 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom        pos        end      region_id REACH000026_methyl_h1  \\\n",
       "1   chrX  147911920  147912179      FMR1_5UTR                     .   \n",
       "7   chrX  149505139  149505306       IDS_5UTR                     .   \n",
       "8   chrX  149631582  149631792  TMEM185A_5UTR                     .   \n",
       "9   chrX  150363318  150363530    MAMLD1_5UTR                     .   \n",
       "10  chrX  150568622  150568662      MTM1_5UTR                     .   \n",
       "11  chrX  150693397  150693530     MTMR1_5UTR                     .   \n",
       "12  chrX  150898590  150898668    CD99L2_5UTR                     .   \n",
       "13  chrX  150983336  150983376     HMGB3_5UTR                     .   \n",
       "\n",
       "   REACH000026_methyl_h2                              REACH000026_methyl_h0  \\\n",
       "1                      .                                                  .   \n",
       "7                      .                                                  .   \n",
       "8                      .  5|0|0|0|0|13|5|7|20|56|43|61|72|96|91|64|21|11...   \n",
       "9                      .  0|2|2|2|0|0|1|6|1|3|2|49|71|55|13|6|11|26|7|5|...   \n",
       "10                     .  1|0|0|0|0|0,2|4|2|0|0|6,7|14|13|9|1|0,2|13|6|0...   \n",
       "11                     .  6|4|6|5|10|63|82|90|76|34|11|6|1|2|2|1|1|3|2|4...   \n",
       "12                     .                 0|0|1|0|1|1|6|4,1|1|1|4|3|16|46|69   \n",
       "13                     .                                                  .   \n",
       "\n",
       "   REACH000026_bp_dev_h1 REACH000026_bp_dev_h2 REACH000026_bp_dev_h0  ...  \\\n",
       "1                      .                     .                     .  ...   \n",
       "7                      .                     .                     .  ...   \n",
       "8                      .                     .               -21,-19  ...   \n",
       "9                      .                     .                    -1  ...   \n",
       "10                     .                     .             0,0,0,0,0  ...   \n",
       "11                     .                     .                 1,0,0  ...   \n",
       "12                     .                     .                   0,0  ...   \n",
       "13                     .                     .                     .  ...   \n",
       "\n",
       "                  REACH000685_methyl_h2  \\\n",
       "1                                     .   \n",
       "7                                     .   \n",
       "8                                     .   \n",
       "9                                     .   \n",
       "10                                    .   \n",
       "11                                    .   \n",
       "12  1|0|0|7|7|0|0|0|2,45|0|21|0|0|2|0|2   \n",
       "13                                    .   \n",
       "\n",
       "                                REACH000685_methyl_h0 REACH000685_bp_dev_h1  \\\n",
       "1   2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...                     .   \n",
       "7   11|0|0|0|0|15|0|0|0|4|89|91|2|0|0|0|1|9|93|89|...                     .   \n",
       "8   100|4|30|7|40|97|0,0|0|0|0|2|0|0|1|0|0|0|0|2|2...                     .   \n",
       "9   1|0|1|0|0|0|4|0|3|61|9|12|46|79|84|75|0|1|0|0|...                     .   \n",
       "10  40|36|9|41|79|30|22|2|45,4|2|0|0|0|0|5,0|4|0|1...                     .   \n",
       "11  0|9|0|1|0|0|0|1|13|0|16|0|0|2|0|5,55|70|9|2|1|...                     .   \n",
       "12      46|85,0|49|2|0|0|4|0|0,0|80|0|2|0|3|0|36,10|0                    -3   \n",
       "13                                 0|0|12|0,0|0|0|0|2                     .   \n",
       "\n",
       "   REACH000685_bp_dev_h2           REACH000685_bp_dev_h0  \\\n",
       "1                      .                  13,17,33,29,16   \n",
       "7                      .  6,-14,-3,-4,-29,-24,-17,-8,-69   \n",
       "8                      .             -23,-13,-2,-8,-21,.   \n",
       "9                      .        -8,-3,-11,-2,-17,8,-5,-2   \n",
       "10                     .                     9,-1,-2,2,6   \n",
       "11                     .     -10,-2,5,0,-3,3,5,-5,-16,-4   \n",
       "12                   4,0                      -6,-6,-5,.   \n",
       "13                     .                           -1,-6   \n",
       "\n",
       "                       REACH000685_rn_h1  \\\n",
       "1                                      .   \n",
       "7                                      .   \n",
       "8                                      .   \n",
       "9                                      .   \n",
       "10                                     .   \n",
       "11                                     .   \n",
       "12  5af74e3f-fde8-469d-9c44-8ca2cce771d2   \n",
       "13                                     .   \n",
       "\n",
       "                                    REACH000685_rn_h2  \\\n",
       "1                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12  34a1e889-e5f6-4400-8f70-0d32c5afd4d6,7d1b38e0-...   \n",
       "13                                                  .   \n",
       "\n",
       "                                    REACH000685_rn_h0 REACH000685_ps_h1  \\\n",
       "1   658567b8-0bb0-48a8-a85d-619f26f01f60,ec6386b7-...                 .   \n",
       "7   dbdc51c7-868d-4c1e-aef8-54b4be7616b1,ee73bd80-...                 .   \n",
       "8   0bdcf4ce-425a-4468-9796-785e0a2817f8,8d6dac54-...                 .   \n",
       "9   b0ddb187-c71b-4f31-8e70-da57f39b8cff,9b1c13e4-...                 .   \n",
       "10  7178be63-b143-43d2-b646-7f250eee984a,3152af87-...                 .   \n",
       "11  33acaac3-8252-404b-b9e8-e6c5fde090e9,21350c97-...                 .   \n",
       "12  1ae72adc-1d37-4141-bfa6-590e0ec211d1,46f0b406-...             10185   \n",
       "13  519f9439-750b-421f-9828-4705fb07f845,94933348-...                 .   \n",
       "\n",
       "   REACH000685_ps_h2  \n",
       "1                  .  \n",
       "7                  .  \n",
       "8                  .  \n",
       "9                  .  \n",
       "10                 .  \n",
       "11                 .  \n",
       "12       10185,10185  \n",
       "13                 .  \n",
       "\n",
       "[8 rows x 2974 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### expand columns for each sample, because each column consists of semicolon-separated sample values\n",
    "df_exnd = df[['chrom' ,'pos', 'end', 'region_id']].copy()\n",
    "cols = df.columns.tolist()[4:]\n",
    "#print(cols)\n",
    "\n",
    "for idx, sample in enumerate(sample_order):\n",
    "    for col in cols:\n",
    "        df_exnd[f'{sample}_{col}'] = df[col].str.split(';').apply(lambda row: row[idx])\n",
    "display(df_exnd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bd951fa3-bf03-4f1e-87b7-9a3467d43f4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>MEAN_COVERAGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG004</td>\n",
       "      <td>ONT</td>\n",
       "      <td>81.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000626</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACH000546</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACH000683</td>\n",
       "      <td>ONT</td>\n",
       "      <td>9.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACH000563</td>\n",
       "      <td>ONT</td>\n",
       "      <td>8.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>REACH000284</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>REACH000672</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>REACH000292</td>\n",
       "      <td>PB</td>\n",
       "      <td>9.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>REACH000066</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>2602-2-1</td>\n",
       "      <td>PB</td>\n",
       "      <td>0.89</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>288 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SAMPLE COHORT  MEAN_COVERAGE\n",
       "0          HG004    ONT          81.44\n",
       "1    REACH000626    ONT          10.51\n",
       "2    REACH000546    ONT          10.87\n",
       "3    REACH000683    ONT           9.97\n",
       "4    REACH000563    ONT           8.68\n",
       "..           ...    ...            ...\n",
       "283  REACH000284     PB           3.43\n",
       "284  REACH000672     PB           5.02\n",
       "285  REACH000292     PB           9.00\n",
       "286  REACH000066     PB           1.61\n",
       "287     2602-2-1     PB           0.89\n",
       "\n",
       "[288 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Get coverage information for each sample\n",
    "file_cov = f'{dir_data}/coverages_mod.tsv'\n",
    "df_cov = pd.read_table(file_cov, header=0, sep='\\t')\n",
    "multiplat_samples = ['REACH000236', 'REACH000530', 'REACH000531', 'REACH000532', 'REACH000436']\n",
    "df_cov['SAMPLE'] = df_cov.apply(lambda row: row['SAMPLE']+'_'+row['COHORT'] if row['SAMPLE'] in multiplat_samples else row['SAMPLE'], axis=1)\n",
    "display(df_cov)\n",
    "\n",
    "def get_plat(sample):\n",
    "    try:\n",
    "        ret = df_cov[df_cov.SAMPLE == sample]['COHORT'].values[0]\n",
    "    except:\n",
    "        print(f'problem with samples: {sample}')\n",
    "        ret = ''\n",
    "    return ret\n",
    "\n",
    "def get_cov(sample):\n",
    "    try:\n",
    "        ret = df_cov[df_cov.SAMPLE == sample]['MEAN_COVERAGE'].values[0]\n",
    "    except:\n",
    "        print(f'problem with samples: {sample} in get_cov')\n",
    "        ret = ''\n",
    "    return ret\n",
    "\n",
    "plat_dict = {sample: get_plat(sample) for sample in df_cov.SAMPLE.tolist()}\n",
    "plat_dict['REACH000226_blood'] = plat_dict['REACH000226']\n",
    "plat_dict['REACH000226_sperm'] = plat_dict['REACH000226']\n",
    "#print('plat_dict:')\n",
    "#print(plat_dict)\n",
    "\n",
    "cov_dict = {sample: get_cov(sample) for sample in df_cov.SAMPLE.tolist()}\n",
    "#print('cov_dict:')\n",
    "#print(cov_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6cafdb05-6258-4789-9e32-0a9677accf24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_trios:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(117, 6)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_samples:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>plat</th>\n",
       "      <th>coverage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>REACH000026</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>REACH000058</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>6.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>REACH000065</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>REACH000086</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>2.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>REACH000089</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>REACH000672</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>REACH000195</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>2.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>REACH000651</td>\n",
       "      <td>M</td>\n",
       "      <td>2</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACH000226_blood</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>ONT</td>\n",
       "      <td>1.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000226_sperm</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>ONT</td>\n",
       "      <td>27.87</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>245 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample_id rel  sex plat  coverage\n",
       "10        REACH000026   C    1   PB      3.22\n",
       "11        REACH000058   C    1   PB      6.19\n",
       "12        REACH000065   C    1   PB      1.22\n",
       "15        REACH000086   C    1   PB      2.70\n",
       "18        REACH000089   C    2   PB      3.27\n",
       "..                ...  ..  ...  ...       ...\n",
       "60        REACH000672   M    2   PB      5.02\n",
       "61        REACH000195   M    2   PB      2.67\n",
       "62        REACH000651   M    2   PB      5.75\n",
       "0   REACH000226_blood   D    1  ONT      1.92\n",
       "1   REACH000226_sperm   D    1  ONT     27.87\n",
       "\n",
       "[245 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Get sample relation information from a psam file\n",
    "file_psam = f'{dir_data}/REACH_LR_platform.psam'\n",
    "df_p = pd.read_table(file_psam, sep='\\t', header=None, names = ['famid', 'sample_id', 'dad', 'mom', 'sex', 'phen'])\n",
    "#print('df_p:')\n",
    "#display(df_p)\n",
    "ds_sex = df_p.set_index('sample_id').sex\n",
    "#display(ds_sex)\n",
    "\n",
    "# make mom and dad dictionary\n",
    "sample_dad_dict = {}\n",
    "sample_mom_dict = {}\n",
    "for sample, dad, mom in zip(df_p.sample_id, df_p.dad, df_p.mom):\n",
    "    #print(sample, dad, mom)\n",
    "    sample_dad_dict[sample] = dad\n",
    "    sample_mom_dict[sample] = mom\n",
    "\n",
    "#print('complete trios:')\n",
    "df_trios = df_p.loc[df_p.dad.isin(df_p.sample_id) & df_p.mom.isin(df_p.sample_id)]\n",
    "print('df_trios:')\n",
    "display(df_trios.shape)\n",
    "\n",
    "### make a samples dataframe containing all samples in complete trios\n",
    "df_samples = pd.DataFrame()\n",
    "df_samples['sample_id'] = df_trios['sample_id']\n",
    "df_samples['rel'] = 'C'\n",
    "\n",
    "df_dads = pd.DataFrame()\n",
    "df_dads['sample_id'] =  list(set(df_trios['dad'].tolist()))\n",
    "df_dads['rel'] = 'D'\n",
    "df_samples = pd.concat([df_samples, df_dads])\n",
    "\n",
    "df_moms = pd.DataFrame()\n",
    "df_moms['sample_id'] =  list(set(df_trios['mom'].tolist()))\n",
    "df_moms['rel'] = 'M'\n",
    "df_samples = pd.concat([df_samples, df_moms])\n",
    "\n",
    "df_samples['sex'] = df_samples.sample_id.map(ds_sex)\n",
    "df_samples['plat'] = df_samples.sample_id.map(plat_dict)\n",
    "df_samples['coverage'] = df_samples.sample_id.map(cov_dict)\n",
    "### add blood/sperm sections for REACH000226\n",
    "row_add = pd.DataFrame({'sample_id':['REACH000226_blood', 'REACH000226_sperm'] ,'rel':['D', 'D'] ,'sex':[1, 1] ,'plat':['ONT', 'ONT'] ,'coverage':[1.92, 27.87]})\n",
    "df_samples = pd.concat([df_samples, row_add])\n",
    "pd.set_option('display.max_rows', 10)\n",
    "print('df_samples:')\n",
    "display(df_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7f69ccf6-93dd-47d5-93ab-e30524d7a4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read meta data of subjects\n",
    "meta_file = f'{dir_data}/REACH_sample_info.tsv'\n",
    "df_meta = pd.read_table(meta_file, sep='\\t', header=0)\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_meta['Sample_ID_orig'] = df_meta['Sample_ID']\n",
    "df_meta['Sample_ID'] = df_meta.apply(lambda row: rename_samples_dict[row['Sample_ID']] \n",
    "                                     if row['Sample_ID'] in rename_samples_dict else row['Sample_ID'], axis=1)\n",
    "\n",
    "aff_dict = {}\n",
    "for sample, aff in zip(df_meta['Sample_ID'].tolist(), df_meta['Affected'].tolist()):\n",
    "    aff_dict[sample] = aff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "825a3593-80c2-42be-ad81-5cc92b903c4a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_pca_flt:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phen</th>\n",
       "      <th>sex</th>\n",
       "      <th>cohort</th>\n",
       "      <th>duo</th>\n",
       "      <th>family</th>\n",
       "      <th>is_eur</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>...</th>\n",
       "      <th>PC10_MAT</th>\n",
       "      <th>rel</th>\n",
       "      <th>Sample_ID</th>\n",
       "      <th>Affected</th>\n",
       "      <th>Affected_ASD</th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>MEAN_COVERAGE</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Platform</th>\n",
       "      <th>iid_orig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.008627</td>\n",
       "      <td>-0.005607</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.014983</td>\n",
       "      <td>C</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>10.15</td>\n",
       "      <td>1</td>\n",
       "      <td>ONT</td>\n",
       "      <td>REACH000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.095364</td>\n",
       "      <td>-0.045230</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011925</td>\n",
       "      <td>C</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>3.22</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>...</td>\n",
       "      <td>0.016383</td>\n",
       "      <td>C</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>6.19</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>F0065</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.252774</td>\n",
       "      <td>0.107933</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>C</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.044785</td>\n",
       "      <td>0.038065</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.018303</td>\n",
       "      <td>C</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>2.70</td>\n",
       "      <td>1</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.204041</td>\n",
       "      <td>0.127559</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.030630</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>3.36</td>\n",
       "      <td>0</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.027327</td>\n",
       "      <td>-0.004878</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>1.20</td>\n",
       "      <td>0</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.091442</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.086006</td>\n",
       "      <td>-0.003008</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>1.11</td>\n",
       "      <td>0</td>\n",
       "      <td>PB</td>\n",
       "      <td>REACH000769</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       fid          iid phen     sex  cohort   duo  family  is_eur       PC1  \\\n",
       "0    F0001  REACH000001  ASD    Male       1  trio     1.0       1  0.008627   \n",
       "1    F0026  REACH000026  ASD    Male       1  trio     1.0       0  0.095364   \n",
       "2    F0058  REACH000058  ASD    Male       1  trio     1.0       1  0.002422   \n",
       "3    F0065  REACH000065  ASD    Male       1  trio     1.0       0  0.252774   \n",
       "4    F0078  REACH000086  ASD    Male       1  trio     2.0       0  0.044785   \n",
       "..     ...          ...  ...     ...     ...   ...     ...     ...       ...   \n",
       "238  F0288  REACH000756    .    Male       1     .     NaN       0  0.204041   \n",
       "239  F0290  REACH000763    .  Female       1     .     NaN       1 -0.030630   \n",
       "240  F0290  REACH000764    .    Male       1     .     NaN       1 -0.027327   \n",
       "241  F0291  REACH000768    .  Female       1     .     NaN       0  0.091442   \n",
       "242  F0291  REACH000769    .    Male       1     .     NaN       0  0.086006   \n",
       "\n",
       "          PC2  ...  PC10_MAT  rel    Sample_ID  Affected  Affected_ASD  \\\n",
       "0   -0.005607  ... -0.014983    C  REACH000001       Yes           Yes   \n",
       "1   -0.045230  ... -0.011925    C  REACH000026       Yes           Yes   \n",
       "2   -0.009216  ...  0.016383    C  REACH000058       Yes           Yes   \n",
       "3    0.107933  ... -0.009906    C  REACH000065       Yes           Yes   \n",
       "4    0.038065  ... -0.018303    C  REACH000086       Yes           Yes   \n",
       "..        ...  ...       ...  ...          ...       ...           ...   \n",
       "238  0.127559  ...       NaN    P  REACH000756        No            No   \n",
       "239  0.003052  ...       NaN    P  REACH000763        No            No   \n",
       "240 -0.004878  ...       NaN    P  REACH000764        No            No   \n",
       "241 -0.016917  ...       NaN    P  REACH000768        No            No   \n",
       "242 -0.003008  ...       NaN    P  REACH000769        No            No   \n",
       "\n",
       "          SAMPLE  MEAN_COVERAGE  Phenotype  Platform     iid_orig  \n",
       "0    REACH000001          10.15          1       ONT  REACH000001  \n",
       "1    REACH000026           3.22          1        PB  REACH000026  \n",
       "2    REACH000058           6.19          1        PB  REACH000058  \n",
       "3    REACH000065           1.22          1        PB  REACH000065  \n",
       "4    REACH000086           2.70          1        PB  REACH000086  \n",
       "..           ...            ...        ...       ...          ...  \n",
       "238  REACH000756           1.55          0        PB  REACH000756  \n",
       "239  REACH000763           3.36          0        PB  REACH000763  \n",
       "240  REACH000764           1.20          0        PB  REACH000764  \n",
       "241  REACH000768           0.10          0        PB  REACH000768  \n",
       "242  REACH000769           1.11          0        PB  REACH000769  \n",
       "\n",
       "[243 rows x 47 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Read PCA data of subjects\n",
    "file_pca = f'{dir_data}/master_phen_4.SBayesR.20210915.csv'\n",
    "\n",
    "df_pca = pd.read_table(file_pca, sep=',', header=0)\n",
    "cols = ['fid', 'iid', 'phen','sex','cohort','duo','family','is_eur','PC1','PC2','PC3','PC4','PC5','PC6','PC7','PC8','PC9','PC10', \n",
    "        'PC1_PAT','PC2_PAT','PC3_PAT','PC4_PAT','PC5_PAT','PC6_PAT','PC7_PAT','PC8_PAT','PC9_PAT','PC10_PAT', \n",
    "        'PC1_MAT','PC2_MAT','PC3_MAT','PC4_MAT','PC5_MAT','PC6_MAT','PC7_MAT','PC8_MAT','PC9_MAT','PC10_MAT']\n",
    "df_pca = df_pca[cols]\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_pca['iid'] = df_pca.apply(lambda row: rename_samples_dict[row['iid']] if row['iid'] in rename_samples_dict else row['iid'], axis=1)\n",
    "#display(df_pca)\n",
    "\n",
    "# first filter for the kids, then add parents to them\n",
    "df_pca_flt = df_pca.loc[df_pca.iid.isin(df_trios.sample_id)].copy()\n",
    "df_pca_flt['rel'] = 'C'\n",
    "#print('df_pca_flt:')\n",
    "#display(df_pca_flt)\n",
    "\n",
    "df_parents = pd.DataFrame()\n",
    "for index, row in df_pca_flt.iterrows():\n",
    "    sample = row['iid']\n",
    "    fid = row['fid']\n",
    "    phen = row['phen']\n",
    "    is_eur = row['is_eur']\n",
    "    pc_mat = [row[f'PC{x}_MAT'] for x in range(1,11)]\n",
    "    pc_pat = [row[f'PC{x}_PAT'] for x in range(1,11)]\n",
    "    dad = sample_dad_dict[sample]\n",
    "    mom = sample_mom_dict[sample]\n",
    "    \n",
    "    dict_mat = {'fid': [fid], 'iid': [mom], 'phen': ['.'], 'sex': ['Female'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_mat.update({f'PC{x}': [pc_mat[x-1]] for x in range(1,11)})\n",
    "    df_mat = pd.DataFrame(dict_mat)\n",
    "\n",
    "    dict_pat = {'fid': [fid], 'iid': [dad], 'phen': ['.'], 'sex': ['Male'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_pat.update({f'PC{x}': [pc_pat[x-1]] for x in range(1,11)})\n",
    "    df_pat = pd.DataFrame(dict_pat)\n",
    "    \n",
    "    if df_parents.shape[0] == 0 or mom not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_mat])\n",
    "    if df_parents.shape[0] == 0 or dad not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_pat])\n",
    "\n",
    "df_parents['rel'] = 'P'\n",
    "#print('df_parents:')\n",
    "#display(df_parents)\n",
    "\n",
    "df_pca_flt = pd.concat([df_pca_flt, df_parents])\n",
    "\n",
    "df_pca_flt = pd.merge(df_pca_flt, df_meta[['Sample_ID', 'Affected', 'Affected_ASD']], how='inner', left_on='iid', right_on='Sample_ID')\n",
    "df_pca_flt = pd.merge(df_pca_flt, df_cov[['SAMPLE', 'MEAN_COVERAGE']], how='inner', left_on='iid', right_on='SAMPLE')\n",
    "\n",
    "df_pca_flt['Phenotype'] = df_pca_flt.apply(lambda row: 1 if row.Affected == 'Yes' else 0, axis=1)\n",
    "#df_pca_flt['Phenotype_ASD'] = df_pca_flt.apply(lambda row: 1 if row.Affected_ASD == 'Yes' else 0, axis=1)\n",
    "df_pca_flt['Platform'] = df_pca_flt.apply(lambda row: get_plat(row.iid), axis=1)\n",
    "\n",
    "### make a original iid w/o platform for the handfull of samples\n",
    "df_pca_flt['iid_orig'] = df_pca_flt.iid.str.split('_').str[0]\n",
    "\n",
    "print('df_pca_flt:')\n",
    "display(df_pca_flt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "24930164-1f38-4c77-ad5a-8949350426f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bp_dev_h1</th>\n",
       "      <th>xci_read_mean_methyl_lik_h1</th>\n",
       "      <th>bp_dev_h2</th>\n",
       "      <th>xci_read_mean_methyl_lik_h2</th>\n",
       "      <th>bp_dev_h0</th>\n",
       "      <th>xci_read_mean_methyl_lik_h0</th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>repeat_length_h1</th>\n",
       "      <th>repeat_length_h2</th>\n",
       "      <th>repeat_length_h0</th>\n",
       "      <th>mean_repeat_length_h1</th>\n",
       "      <th>mean_repeat_length_h2</th>\n",
       "      <th>mean_repeat_length_h0</th>\n",
       "      <th>read_mean_methyl_lik_h1</th>\n",
       "      <th>read_mean_methyl_lik_h2</th>\n",
       "      <th>read_mean_methyl_lik_h0</th>\n",
       "      <th>num_methyl_reads_h1</th>\n",
       "      <th>num_unmethyl_reads_h1</th>\n",
       "      <th>frac_hap_methyl_h1</th>\n",
       "      <th>num_methyl_reads_h2</th>\n",
       "      <th>num_unmethyl_reads_h2</th>\n",
       "      <th>frac_hap_methyl_h2</th>\n",
       "      <th>num_methyl_reads_h0</th>\n",
       "      <th>num_unmethyl_reads_h0</th>\n",
       "      <th>frac_hap_methyl_h0</th>\n",
       "      <th>xci_num_methyl_reads_h1</th>\n",
       "      <th>xci_num_unmethyl_reads_h1</th>\n",
       "      <th>xci_frac_hap_methyl_h1</th>\n",
       "      <th>xci_num_methyl_reads_h2</th>\n",
       "      <th>xci_num_unmethyl_reads_h2</th>\n",
       "      <th>xci_frac_hap_methyl_h2</th>\n",
       "      <th>xci_num_methyl_reads_h0</th>\n",
       "      <th>xci_num_unmethyl_reads_h0</th>\n",
       "      <th>xci_frac_hap_methyl_h0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000026</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>22,10,10,0,2,7,4,2,20,19,20,2,18</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>51,51,51,57,51,51,52</td>\n",
       "      <td>3,4,9,10,2,2,7,2,39,8,12,1,17,0,0,1,1,14,29,10...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>37,37,37,39,37,37,37</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>37.285714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,9,11,17,23,5,17</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000065</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>30</td>\n",
       "      <td>9,10</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>62|37|36|51|8|2|0|0|0|0|0|7|46|88|91|87|84|66|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>22</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000066</th>\n",
       "      <td>32</td>\n",
       "      <td>1,12,34</td>\n",
       "      <td>31,43,30</td>\n",
       "      <td>87,0</td>\n",
       "      <td>.</td>\n",
       "      <td>71,77,1,88</td>\n",
       "      <td>91|95|100|99|96|100|99|98|98|96|96|97|94|99|98...</td>\n",
       "      <td>64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|2|6|2|...</td>\n",
       "      <td>.</td>\n",
       "      <td>31</td>\n",
       "      <td>30,34,30</td>\n",
       "      <td>.</td>\n",
       "      <td>31.0</td>\n",
       "      <td>31.333333</td>\n",
       "      <td>.</td>\n",
       "      <td>79</td>\n",
       "      <td>8,42,3</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000067</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>15</td>\n",
       "      <td>.</td>\n",
       "      <td>7,0,11,23,55,0,2,3,4,0,0,8,46</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9|14|13|23|14|22|13|13|29|12|20|2|7|5|9|18|6|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>0.076923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000681</th>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>16</td>\n",
       "      <td>-6,-8,-6,25</td>\n",
       "      <td>3,47,10,23,3,1,9,9,2,17,1,3,0,6,2,16,4,6,14,11...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|9|59|2|56|0|0|0|0|0|0|0|0|0|6|0|0|11|0|1|9|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>18,17,18,28</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>20.25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4,25,5,11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>0.03125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000682</th>\n",
       "      <td>31,28,.,24</td>\n",
       "      <td>2,63,56,6,31,89,81,3,69,52,66,69,2,17,60,2,78,...</td>\n",
       "      <td>12,14</td>\n",
       "      <td>71,20,2,54,41,43,8,3,7,5,8,58,54,94,57,48,4,84...</td>\n",
       "      <td>30,26,-12</td>\n",
       "      <td>80,73,0,0</td>\n",
       "      <td>0|0|0|9|8|4|3|3|0|0|0|0|21|0|0|0|2|8|0|0|3|5|1...</td>\n",
       "      <td>45|97|100|100|100|100|100|22|89|17|92|100|95|1...</td>\n",
       "      <td>100|100|100|81|100|95|100|100|100|100|100|35|1...</td>\n",
       "      <td>30,29,.,28</td>\n",
       "      <td>24,25</td>\n",
       "      <td>30,29,16</td>\n",
       "      <td>29.0</td>\n",
       "      <td>24.5</td>\n",
       "      <td>25.0</td>\n",
       "      <td>8,2,5,3</td>\n",
       "      <td>74,86</td>\n",
       "      <td>70,8,12</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>13</td>\n",
       "      <td>18</td>\n",
       "      <td>0.419355</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000683</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>31,26,23,31</td>\n",
       "      <td>2,6,4,1,1,20,7,4,33,9,19,4,1,1,0,0,6,6,3,2,1,2...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|18|0|0|0|0|0|0|0|0|0|0|1|33|0|0|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30,29,28,30</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>29.25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2,4,29,6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000684</th>\n",
       "      <td>30,24,24</td>\n",
       "      <td></td>\n",
       "      <td>23,15,24</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>2,4,5,7,3,0,27,1,10,4,2,6,6,0,28,1,6,12,1,87,5...</td>\n",
       "      <td>3|58|0|2|5|0|0|0|0|0|0|0|0|0|0|0|0|0|0|14|0|0|...</td>\n",
       "      <td>0|0|0|8|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|2|4|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>30,28,28</td>\n",
       "      <td>28,25,28</td>\n",
       "      <td>.</td>\n",
       "      <td>28.666667</td>\n",
       "      <td>27.0</td>\n",
       "      <td>.</td>\n",
       "      <td>2,7,3</td>\n",
       "      <td>1,13,3</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>0.0625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000685</th>\n",
       "      <td>.</td>\n",
       "      <td>23</td>\n",
       "      <td>.</td>\n",
       "      <td>2,9</td>\n",
       "      <td>13,17,33,29,16</td>\n",
       "      <td>18,31,2,13,35,6,2,8,21,40,4,3,24,17,2,17,16,6,...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>24,26,31,30,25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>27.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,1,5,6,3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>43</td>\n",
       "      <td>0.022727</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>270 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              bp_dev_h1                        xci_read_mean_methyl_lik_h1  \\\n",
       "REACH000026           .                                                      \n",
       "REACH000058           .                                                      \n",
       "REACH000065           .                                                      \n",
       "REACH000066          32                                            1,12,34   \n",
       "REACH000067           .                                                      \n",
       "...                 ...                                                ...   \n",
       "REACH000681           .                                                  5   \n",
       "REACH000682  31,28,.,24  2,63,56,6,31,89,81,3,69,52,66,69,2,17,60,2,78,...   \n",
       "REACH000683           .                                                      \n",
       "REACH000684    30,24,24                                                      \n",
       "REACH000685           .                                                 23   \n",
       "\n",
       "            bp_dev_h2                        xci_read_mean_methyl_lik_h2  \\\n",
       "REACH000026         .                                                      \n",
       "REACH000058         .                                                      \n",
       "REACH000065         .                                                      \n",
       "REACH000066  31,43,30                                               87,0   \n",
       "REACH000067         .                                                 15   \n",
       "...               ...                                                ...   \n",
       "REACH000681         .                                                 16   \n",
       "REACH000682     12,14  71,20,2,54,41,43,8,3,7,5,8,58,54,94,57,48,4,84...   \n",
       "REACH000683         .                                                      \n",
       "REACH000684  23,15,24                                                      \n",
       "REACH000685         .                                                2,9   \n",
       "\n",
       "                        bp_dev_h0  \\\n",
       "REACH000026                     .   \n",
       "REACH000058  51,51,51,57,51,51,52   \n",
       "REACH000065                    30   \n",
       "REACH000066                     .   \n",
       "REACH000067                     .   \n",
       "...                           ...   \n",
       "REACH000681           -6,-8,-6,25   \n",
       "REACH000682             30,26,-12   \n",
       "REACH000683           31,26,23,31   \n",
       "REACH000684                     .   \n",
       "REACH000685        13,17,33,29,16   \n",
       "\n",
       "                                   xci_read_mean_methyl_lik_h0  \\\n",
       "REACH000026                   22,10,10,0,2,7,4,2,20,19,20,2,18   \n",
       "REACH000058  3,4,9,10,2,2,7,2,39,8,12,1,17,0,0,1,1,14,29,10...   \n",
       "REACH000065                                               9,10   \n",
       "REACH000066                                         71,77,1,88   \n",
       "REACH000067                      7,0,11,23,55,0,2,3,4,0,0,8,46   \n",
       "...                                                        ...   \n",
       "REACH000681  3,47,10,23,3,1,9,9,2,17,1,3,0,6,2,16,4,6,14,11...   \n",
       "REACH000682                                          80,73,0,0   \n",
       "REACH000683  2,6,4,1,1,20,7,4,33,9,19,4,1,1,0,0,6,6,3,2,1,2...   \n",
       "REACH000684  2,4,5,7,3,0,27,1,10,4,2,6,6,0,28,1,6,12,1,87,5...   \n",
       "REACH000685  18,31,2,13,35,6,2,8,21,40,4,3,24,17,2,17,16,6,...   \n",
       "\n",
       "                                                     methyl_h1  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058                                                  .   \n",
       "REACH000065                                                  .   \n",
       "REACH000066  91|95|100|99|96|100|99|98|98|96|96|97|94|99|98...   \n",
       "REACH000067                                                  .   \n",
       "...                                                        ...   \n",
       "REACH000681                                                  .   \n",
       "REACH000682  0|0|0|9|8|4|3|3|0|0|0|0|21|0|0|0|2|8|0|0|3|5|1...   \n",
       "REACH000683                                                  .   \n",
       "REACH000684  3|58|0|2|5|0|0|0|0|0|0|0|0|0|0|0|0|0|0|14|0|0|...   \n",
       "REACH000685                                                  .   \n",
       "\n",
       "                                                     methyl_h2  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058                                                  .   \n",
       "REACH000065                                                  .   \n",
       "REACH000066  64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|2|6|2|...   \n",
       "REACH000067                                                  .   \n",
       "...                                                        ...   \n",
       "REACH000681                                                  .   \n",
       "REACH000682  45|97|100|100|100|100|100|22|89|17|92|100|95|1...   \n",
       "REACH000683                                                  .   \n",
       "REACH000684  0|0|0|8|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|2|4|0|...   \n",
       "REACH000685                                                  .   \n",
       "\n",
       "                                                     methyl_h0  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058  0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3|4...   \n",
       "REACH000065  62|37|36|51|8|2|0|0|0|0|0|7|46|88|91|87|84|66|...   \n",
       "REACH000066                                                  .   \n",
       "REACH000067  9|14|13|23|14|22|13|13|29|12|20|2|7|5|9|18|6|4...   \n",
       "...                                                        ...   \n",
       "REACH000681  0|9|59|2|56|0|0|0|0|0|0|0|0|0|6|0|0|11|0|1|9|0...   \n",
       "REACH000682  100|100|100|81|100|95|100|100|100|100|100|35|1...   \n",
       "REACH000683  0|0|0|18|0|0|0|0|0|0|0|0|0|0|1|33|0|0|0|0|0|0|...   \n",
       "REACH000684                                                  .   \n",
       "REACH000685  2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...   \n",
       "\n",
       "            repeat_length_h1 repeat_length_h2      repeat_length_h0  \\\n",
       "REACH000026                .                .                     .   \n",
       "REACH000058                .                .  37,37,37,39,37,37,37   \n",
       "REACH000065                .                .                    30   \n",
       "REACH000066               31         30,34,30                     .   \n",
       "REACH000067                .                .                     .   \n",
       "...                      ...              ...                   ...   \n",
       "REACH000681                .                .           18,17,18,28   \n",
       "REACH000682       30,29,.,28            24,25              30,29,16   \n",
       "REACH000683                .                .           30,29,28,30   \n",
       "REACH000684         30,28,28         28,25,28                     .   \n",
       "REACH000685                .                .        24,26,31,30,25   \n",
       "\n",
       "            mean_repeat_length_h1 mean_repeat_length_h2 mean_repeat_length_h0  \\\n",
       "REACH000026                     .                     .                     .   \n",
       "REACH000058                     .                     .             37.285714   \n",
       "REACH000065                     .                     .                  30.0   \n",
       "REACH000066                  31.0             31.333333                     .   \n",
       "REACH000067                     .                     .                     .   \n",
       "...                           ...                   ...                   ...   \n",
       "REACH000681                     .                     .                 20.25   \n",
       "REACH000682                  29.0                  24.5                  25.0   \n",
       "REACH000683                     .                     .                 29.25   \n",
       "REACH000684             28.666667                  27.0                     .   \n",
       "REACH000685                     .                     .                  27.2   \n",
       "\n",
       "            read_mean_methyl_lik_h1 read_mean_methyl_lik_h2  \\\n",
       "REACH000026                       .                       .   \n",
       "REACH000058                       .                       .   \n",
       "REACH000065                       .                       .   \n",
       "REACH000066                      79                  8,42,3   \n",
       "REACH000067                       .                       .   \n",
       "...                             ...                     ...   \n",
       "REACH000681                       .                       .   \n",
       "REACH000682                 8,2,5,3                   74,86   \n",
       "REACH000683                       .                       .   \n",
       "REACH000684                   2,7,3                  1,13,3   \n",
       "REACH000685                       .                       .   \n",
       "\n",
       "            read_mean_methyl_lik_h0 num_methyl_reads_h1 num_unmethyl_reads_h1  \\\n",
       "REACH000026                       .                   .                     .   \n",
       "REACH000058       8,9,11,17,23,5,17                   .                     .   \n",
       "REACH000065                      22                   .                     .   \n",
       "REACH000066                       .                   1                     0   \n",
       "REACH000067                      11                   .                     .   \n",
       "...                             ...                 ...                   ...   \n",
       "REACH000681               4,25,5,11                   .                     .   \n",
       "REACH000682                 70,8,12                   0                     4   \n",
       "REACH000683                2,4,29,6                   .                     .   \n",
       "REACH000684                       .                   0                     3   \n",
       "REACH000685               8,1,5,6,3                   .                     .   \n",
       "\n",
       "            frac_hap_methyl_h1 num_methyl_reads_h2 num_unmethyl_reads_h2  \\\n",
       "REACH000026                  .                   .                     .   \n",
       "REACH000058                  .                   .                     .   \n",
       "REACH000065                  .                   .                     .   \n",
       "REACH000066                1.0                   0                     3   \n",
       "REACH000067                  .                   .                     .   \n",
       "...                        ...                 ...                   ...   \n",
       "REACH000681                  .                   .                     .   \n",
       "REACH000682                0.0                   2                     0   \n",
       "REACH000683                  .                   .                     .   \n",
       "REACH000684                0.0                   0                     3   \n",
       "REACH000685                  .                   .                     .   \n",
       "\n",
       "            frac_hap_methyl_h2 num_methyl_reads_h0 num_unmethyl_reads_h0  \\\n",
       "REACH000026                  .                   .                     .   \n",
       "REACH000058                  .                   0                     7   \n",
       "REACH000065                  .                   0                     1   \n",
       "REACH000066                0.0                   .                     .   \n",
       "REACH000067                  .                   0                     1   \n",
       "...                        ...                 ...                   ...   \n",
       "REACH000681                  .                   0                     4   \n",
       "REACH000682                1.0                   1                     2   \n",
       "REACH000683                  .                   0                     4   \n",
       "REACH000684                0.0                   .                     .   \n",
       "REACH000685                  .                   0                     5   \n",
       "\n",
       "            frac_hap_methyl_h0 xci_num_methyl_reads_h1  \\\n",
       "REACH000026                  .                       .   \n",
       "REACH000058                0.0                       .   \n",
       "REACH000065                0.0                       .   \n",
       "REACH000066                  .                       0   \n",
       "REACH000067                0.0                       .   \n",
       "...                        ...                     ...   \n",
       "REACH000681                0.0                       0   \n",
       "REACH000682           0.333333                      13   \n",
       "REACH000683                0.0                       .   \n",
       "REACH000684                  .                       .   \n",
       "REACH000685                0.0                       0   \n",
       "\n",
       "            xci_num_unmethyl_reads_h1 xci_frac_hap_methyl_h1  \\\n",
       "REACH000026                         .                      .   \n",
       "REACH000058                         .                      .   \n",
       "REACH000065                         .                      .   \n",
       "REACH000066                         3                    0.0   \n",
       "REACH000067                         .                      .   \n",
       "...                               ...                    ...   \n",
       "REACH000681                         1                    0.0   \n",
       "REACH000682                        18               0.419355   \n",
       "REACH000683                         .                      .   \n",
       "REACH000684                         .                      .   \n",
       "REACH000685                         1                    0.0   \n",
       "\n",
       "            xci_num_methyl_reads_h2 xci_num_unmethyl_reads_h2  \\\n",
       "REACH000026                       .                         .   \n",
       "REACH000058                       .                         .   \n",
       "REACH000065                       .                         .   \n",
       "REACH000066                       1                         1   \n",
       "REACH000067                       0                         1   \n",
       "...                             ...                       ...   \n",
       "REACH000681                       0                         1   \n",
       "REACH000682                       8                        20   \n",
       "REACH000683                       .                         .   \n",
       "REACH000684                       .                         .   \n",
       "REACH000685                       0                         2   \n",
       "\n",
       "            xci_frac_hap_methyl_h2 xci_num_methyl_reads_h0  \\\n",
       "REACH000026                      .                       0   \n",
       "REACH000058                      .                       0   \n",
       "REACH000065                      .                       0   \n",
       "REACH000066                    0.5                       3   \n",
       "REACH000067                    0.0                       1   \n",
       "...                            ...                     ...   \n",
       "REACH000681                    0.0                       1   \n",
       "REACH000682               0.285714                       2   \n",
       "REACH000683                      .                       0   \n",
       "REACH000684                      .                       2   \n",
       "REACH000685                    0.0                       1   \n",
       "\n",
       "            xci_num_unmethyl_reads_h0 xci_frac_hap_methyl_h0  \n",
       "REACH000026                        13                    0.0  \n",
       "REACH000058                        28                    0.0  \n",
       "REACH000065                         2                    0.0  \n",
       "REACH000066                         1                   0.75  \n",
       "REACH000067                        12               0.076923  \n",
       "...                               ...                    ...  \n",
       "REACH000681                        31                0.03125  \n",
       "REACH000682                         2                    0.5  \n",
       "REACH000683                        29                    0.0  \n",
       "REACH000684                        30                 0.0625  \n",
       "REACH000685                        43               0.022727  \n",
       "\n",
       "[270 rows x 36 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### make a table of raw data.\n",
    "pd.set_option('display.max_columns', None)\n",
    "region_id = 'FMR1_5UTR'\n",
    "xci_region_ids = ['IDS_5UTR', 'TMEM185A_5UTR', 'MAMLD1_5UTR', 'MTM1_5UTR', 'MTMR1_5UTR', 'CD99L2_5UTR', 'HMGB3_5UTR']\n",
    "df_methyl = pd.DataFrame()\n",
    "\n",
    "this_idx = df_exnd.loc[df_exnd.region_id==region_id].index.tolist()[0]\n",
    "cols_bp = [f'bp_dev_h1', f'bp_dev_h2', f'bp_dev_h0']\n",
    "cols_met = [f'methyl_h1', f'methyl_h2', f'methyl_h0']\n",
    "for col in cols_bp+cols_met:\n",
    "    col_out = f'xci_read_mean_methyl_lik_{col.split(\"_\")[-1]}'\n",
    "    df_methyl[col] = pd.Series(dtype='str')\n",
    "    df_methyl[col_out] = pd.Series(dtype='str')\n",
    "\n",
    "for sample in sample_order:\n",
    "    \n",
    "    ### set bp deviation columns\n",
    "    for col in cols_bp:\n",
    "        df_methyl.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "    \n",
    "    ### set FMR1 methylation columns\n",
    "    for col in cols_met:\n",
    "        df_methyl.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "\n",
    "    ### set 7 XCI gene methylation columns\n",
    "    for col in cols_met:\n",
    "        col_out = f'xci_read_mean_methyl_lik_{col.split(\"_\")[-1]}'\n",
    "        read_met_list = [y for x in df_exnd.loc[df_exnd.region_id.isin(xci_region_ids)][f'{sample}_{col}'].tolist() for y in x.split(',') if y!='.']\n",
    "        df_methyl.loc[sample, col_out] = ','.join([str(round(np.mean([float(xx) for xx in x.split('|')]))) for x in read_met_list])\n",
    "\n",
    "### correct phasing of REACH000563 and REACH000561 manually due to poor phasing\n",
    "bp_dev_thr = 60\n",
    "if ('REACH000563' in df_methyl.index):\n",
    "    this_sample = 'REACH000563'\n",
    "    # h1 and h2 are empty and we want to make them from h0 based on bp deviation\n",
    "    df_methyl.loc[this_sample, 'bp_dev_h2'] = ','.join([x for x in df_methyl.loc[this_sample, 'bp_dev_h0'].split(',') if x != '.' and float(x)>bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'bp_dev_h1'] = ','.join([x for x in df_methyl.loc[this_sample, 'bp_dev_h0'].split(',') if x != '.' and float(x)<bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'methyl_h2'] = ','.join([m for b, m in zip(df_methyl.loc[this_sample, 'bp_dev_h0'].split(','),  \n",
    "                                                                          df_methyl.loc[this_sample, 'methyl_h0'].split(',')) if b != '.' and float(b)>bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'methyl_h1'] = ','.join([m for b, m in zip(df_methyl.loc[this_sample, 'bp_dev_h0'].split(','),  \n",
    "                                                                          df_methyl.loc[this_sample, 'methyl_h0'].split(',')) if b != '.' and float(b)<bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'methyl_h0'] = '.'\n",
    "    df_methyl.loc[this_sample, 'bp_dev_h0'] = '.'\n",
    "\n",
    "if ('REACH000561' in df_methyl.index):\n",
    "    this_sample = 'REACH000561'\n",
    "    ### first fix h2: bring one read from h1\n",
    "    # one read in h1 has bp=83 and belongs to h2. we want to bring it in h2\n",
    "    df_methyl.loc[this_sample, 'bp_dev_h2'] = ','.join(df_methyl.loc[this_sample, 'bp_dev_h2'].split(',') + \n",
    "                                               [x for x in df_methyl.loc[this_sample, 'bp_dev_h1'].split(',') if x != '.' and float(x)>bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'methyl_h2'] = ','.join(df_methyl.loc[this_sample, 'methyl_h2'].split(',') + \n",
    "                                                       [m for b, m in zip(df_methyl.loc[this_sample, 'bp_dev_h1'].split(','), \n",
    "                                                                          df_methyl.loc[this_sample, 'methyl_h1'].split(',')) if b != '.' and float(b)>bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'bp_dev_h1'] = ','.join([x for x in df_methyl.loc[this_sample, 'bp_dev_h1'].split(',') if x != '.' and float(x)<bp_dev_thr])\n",
    "    df_methyl.loc[this_sample, 'methyl_h1'] = ','.join([m for b, m in zip(df_methyl.loc[this_sample, 'bp_dev_h1'].split(','),  \n",
    "                                                                          df_methyl.loc[this_sample, 'methyl_h1'].split(',')) if b != '.' and float(b)<bp_dev_thr])\n",
    "\n",
    "### compute repeat length based on bp deviation\n",
    "for col in cols_bp:\n",
    "    col_out = f'repeat_length_{col.split(\"_\")[-1]}'\n",
    "    df_methyl[col_out] = df_methyl[col].apply(lambda x: ','.join([str(round(float(xx) / 3) + 20) if xx!='.' else '.' for xx in x.split(',')]))\n",
    "    \n",
    "### take average of repeat length\n",
    "cols_rep_len = [f'repeat_length_h1', f'repeat_length_h2', f'repeat_length_h0']\n",
    "for col in cols_rep_len:\n",
    "    df_methyl[f'mean_{col}'] = df_methyl.apply(lambda row: np.mean([float(x) for x in row[col].split(',') if x != '.']) \n",
    "                                                 if len([x for x in row[col].split(',') if x != '.'])>0 else '.', axis=1)\n",
    "\n",
    "### compute mean methylation likelihood for each read\n",
    "for col in cols_met:\n",
    "    df_methyl[f'read_mean_methyl_lik_{col.split(\"_\")[-1]}'] = df_methyl.apply(lambda row: ','.join([str(round(np.mean([float(xx) for xx in x.split('|')]))) for x in row[col].split(',') if x != '.'])\n",
    "                                                                             if row[col] != '.' else '.', axis=1)\n",
    "### set number of methylation/unmethylation reads\n",
    "for col in cols_met:\n",
    "    hp = col.split(\"_\")[-1]\n",
    "    df_methyl[f'num_methyl_reads_{hp}'] = df_methyl[f'read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)>=50]) if x != '.' else '.')\n",
    "    df_methyl[f'num_unmethyl_reads_{hp}'] = df_methyl[f'read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)<50]) if x != '.' else '.')\n",
    "    df_methyl[f'frac_hap_methyl_{hp}'] = df_methyl.apply(lambda row: int(row[f'num_methyl_reads_{hp}'])/(int(row[f'num_methyl_reads_{hp}'])+int(row[f'num_unmethyl_reads_{hp}'])) \n",
    "                                                           if (row[f'num_methyl_reads_{hp}']!='.' and row[f'num_unmethyl_reads_{hp}']!='.') else '.', axis=1)\n",
    "\n",
    "### set 7 XCI gene number of methylation/unmethylation reads\n",
    "for col in cols_met:\n",
    "    hp = col.split(\"_\")[-1]\n",
    "    df_methyl[f'xci_num_methyl_reads_{hp}'] = df_methyl[f'xci_read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)>=50]) if (x != '.' and x != '') else '.')\n",
    "    df_methyl[f'xci_num_unmethyl_reads_{hp}'] = df_methyl[f'xci_read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)<50]) if (x != '.' and x != '') else '.')\n",
    "    df_methyl[f'xci_frac_hap_methyl_{hp}'] = df_methyl.apply(lambda row: int(row[f'xci_num_methyl_reads_{hp}'])/(int(row[f'xci_num_methyl_reads_{hp}'])+int(row[f'xci_num_unmethyl_reads_{hp}'])) \n",
    "                                                           if (row[f'xci_num_methyl_reads_{hp}']!='.' and row[f'xci_num_unmethyl_reads_{hp}']!='.') else '.', axis=1)\n",
    "\n",
    "\n",
    "display(df_methyl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "6e78f735-66d7-42f0-8d5b-75e683d8930e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hp_c_h1: 35.0\n",
      "hp_c1_h2: 20.833333333333332\n",
      "hp_c2_h2: 29.444444444444443\n"
     ]
    }
   ],
   "source": [
    "### Repeat length correction for some large deviation families \n",
    "### after averaging over corresponding haplotypes in children and parents for mendelian consistence\n",
    "### subjects corrected: \n",
    "#REACH000519, C, F\n",
    "#REACH000518, C, F\n",
    "#REACH000523, M, F\n",
    "#REACH000524, D, M\n",
    "\n",
    "hp_c_h1 = (df_methyl.loc['REACH000519', 'repeat_length_h1'] + ',' + \n",
    "           df_methyl.loc['REACH000518', 'repeat_length_h1'] + ',' + \n",
    "           df_methyl.loc['REACH000524', 'repeat_length_h0']).split(',')\n",
    "hp_c_h1 = np.mean([float(x) for x in hp_c_h1])\n",
    "print(f'hp_c_h1: {hp_c_h1}')\n",
    "df_methyl.loc['REACH000519', 'mean_repeat_length_h1'] = hp_c_h1\n",
    "df_methyl.loc['REACH000518', 'mean_repeat_length_h1'] = hp_c_h1\n",
    "df_methyl.loc['REACH000524', 'mean_repeat_length_h0'] = hp_c_h1\n",
    "\n",
    "hp_c1_h2 = (df_methyl.loc['REACH000519', 'repeat_length_h2'] + ',' + \n",
    "            df_methyl.loc['REACH000523', 'repeat_length_h1']).split(',')\n",
    "hp_c1_h2 = np.mean([float(x) for x in hp_c1_h2])\n",
    "print(f'hp_c1_h2: {hp_c1_h2}')\n",
    "df_methyl.loc['REACH000519', 'mean_repeat_length_h2'] = hp_c1_h2\n",
    "df_methyl.loc['REACH000523', 'mean_repeat_length_h1'] = hp_c1_h2\n",
    "\n",
    "hp_c2_h2 = (df_methyl.loc['REACH000518', 'repeat_length_h2'] + ',' + \n",
    "            df_methyl.loc['REACH000523', 'repeat_length_h2']).split(',')\n",
    "hp_c2_h2 = np.mean([float(x) for x in hp_c2_h2])\n",
    "print(f'hp_c2_h2: {hp_c2_h2}')\n",
    "df_methyl.loc['REACH000518', 'mean_repeat_length_h2'] = hp_c2_h2\n",
    "df_methyl.loc['REACH000523', 'mean_repeat_length_h2'] = hp_c2_h2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e8fd4747-b50b-4594-8b52-14ef4fb14742",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hp_c_h1: 26.375\n",
      "hp_c1_h2: 49.111111111111114\n",
      "hp_m_h1: 28.25\n"
     ]
    }
   ],
   "source": [
    "### subjects corrected: \n",
    "#REACH000563, C, F\n",
    "#REACH000560, C, M\n",
    "#REACH000561, M, F\n",
    "#REACH000562, D, M\n",
    "\n",
    "hp_c_h1 = (df_methyl.loc['REACH000563', 'repeat_length_h1'] + ',' + \n",
    "           df_methyl.loc['REACH000562', 'repeat_length_h0']).split(',')\n",
    "hp_c_h1 = np.mean([float(x) for x in hp_c_h1])\n",
    "print(f'hp_c_h1: {hp_c_h1}')\n",
    "df_methyl.loc['REACH000563', 'mean_repeat_length_h1'] = hp_c_h1\n",
    "df_methyl.loc['REACH000562', 'mean_repeat_length_h0'] = hp_c_h1\n",
    "\n",
    "hp_c1_h2 = (df_methyl.loc['REACH000563', 'repeat_length_h2'] + ',' + \n",
    "            df_methyl.loc['REACH000561', 'repeat_length_h2']).split(',')\n",
    "hp_c1_h2 = np.mean([float(x) for x in hp_c1_h2])\n",
    "print(f'hp_c1_h2: {hp_c1_h2}')\n",
    "df_methyl.loc['REACH000563', 'mean_repeat_length_h2'] = hp_c1_h2\n",
    "df_methyl.loc['REACH000561', 'mean_repeat_length_h2'] = hp_c1_h2\n",
    "\n",
    "hp_m_h1 = (df_methyl.loc['REACH000561', 'repeat_length_h1']).split(',')\n",
    "hp_m_h1 = np.mean([float(x) for x in hp_m_h1])\n",
    "print(f'hp_m_h1: {hp_m_h1}')\n",
    "df_methyl.loc['REACH000561', 'mean_repeat_length_h1'] = hp_m_h1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9be7dc38-6a31-491d-a2be-7f227c3fce3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hp_c1_h2: 49.4\n",
      "hp_c_h1: 31.2\n",
      "hp_m_h1: 28.8\n"
     ]
    }
   ],
   "source": [
    "### subjects corrected: \n",
    "#REACH000453, C, F\n",
    "#REACH000451, M, F\n",
    "\n",
    "hp_c1_h2 = (df_methyl.loc['REACH000453', 'repeat_length_h2'] + ',' + \n",
    "            df_methyl.loc['REACH000451', 'repeat_length_h2']).split(',')\n",
    "hp_c1_h2 = np.mean([float(x) for x in hp_c1_h2])\n",
    "print(f'hp_c1_h2: {hp_c1_h2}')\n",
    "df_methyl.loc['REACH000453', 'mean_repeat_length_h2'] = hp_c1_h2\n",
    "df_methyl.loc['REACH000451', 'mean_repeat_length_h2'] = hp_c1_h2\n",
    "\n",
    "hp_c_h1 = (df_methyl.loc['REACH000453', 'repeat_length_h1']).split(',')\n",
    "hp_c_h1 = np.mean([float(x) for x in hp_c_h1])\n",
    "print(f'hp_c_h1: {hp_c_h1}')\n",
    "df_methyl.loc['REACH000453', 'mean_repeat_length_h1'] = hp_c_h1\n",
    "\n",
    "hp_m_h1 = (df_methyl.loc['REACH000451', 'repeat_length_h1']).split(',')\n",
    "hp_m_h1 = np.mean([float(x) for x in hp_m_h1])\n",
    "print(f'hp_m_h1: {hp_m_h1}')\n",
    "df_methyl.loc['REACH000451', 'mean_repeat_length_h1'] = hp_m_h1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "55a216b9-07ac-4818-8271-fb8e0e033b0e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in_trios\n",
      "TRUE     242\n",
      "FALSE     28\n",
      "Name: count, dtype: int64\n",
      "Affected\n",
      "No     181\n",
      "Yes     89\n",
      "Name: count, dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bp_dev_h1</th>\n",
       "      <th>xci_read_mean_methyl_lik_h1</th>\n",
       "      <th>bp_dev_h2</th>\n",
       "      <th>xci_read_mean_methyl_lik_h2</th>\n",
       "      <th>bp_dev_h0</th>\n",
       "      <th>xci_read_mean_methyl_lik_h0</th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>repeat_length_h1</th>\n",
       "      <th>repeat_length_h2</th>\n",
       "      <th>repeat_length_h0</th>\n",
       "      <th>mean_repeat_length_h1</th>\n",
       "      <th>mean_repeat_length_h2</th>\n",
       "      <th>mean_repeat_length_h0</th>\n",
       "      <th>read_mean_methyl_lik_h1</th>\n",
       "      <th>read_mean_methyl_lik_h2</th>\n",
       "      <th>read_mean_methyl_lik_h0</th>\n",
       "      <th>num_methyl_reads_h1</th>\n",
       "      <th>num_unmethyl_reads_h1</th>\n",
       "      <th>frac_hap_methyl_h1</th>\n",
       "      <th>num_methyl_reads_h2</th>\n",
       "      <th>num_unmethyl_reads_h2</th>\n",
       "      <th>frac_hap_methyl_h2</th>\n",
       "      <th>num_methyl_reads_h0</th>\n",
       "      <th>num_unmethyl_reads_h0</th>\n",
       "      <th>frac_hap_methyl_h0</th>\n",
       "      <th>xci_num_methyl_reads_h1</th>\n",
       "      <th>xci_num_unmethyl_reads_h1</th>\n",
       "      <th>xci_frac_hap_methyl_h1</th>\n",
       "      <th>xci_num_methyl_reads_h2</th>\n",
       "      <th>xci_num_unmethyl_reads_h2</th>\n",
       "      <th>xci_frac_hap_methyl_h2</th>\n",
       "      <th>xci_num_methyl_reads_h0</th>\n",
       "      <th>xci_num_unmethyl_reads_h0</th>\n",
       "      <th>xci_frac_hap_methyl_h0</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>PC9</th>\n",
       "      <th>PC10</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>fid</th>\n",
       "      <th>rel</th>\n",
       "      <th>sex</th>\n",
       "      <th>Platform</th>\n",
       "      <th>coverage</th>\n",
       "      <th>Affected</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>in_trios</th>\n",
       "      <th>frac_hap_methyl_large_rep</th>\n",
       "      <th>frac_hap_methyl_small_rep</th>\n",
       "      <th>mean_large_rep</th>\n",
       "      <th>mean_small_rep</th>\n",
       "      <th>has_both_hps</th>\n",
       "      <th>binomial_pval</th>\n",
       "      <th>binomial_phred</th>\n",
       "      <th>repeat_group</th>\n",
       "      <th>both_hps_2_reads</th>\n",
       "      <th>both_hps_3_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000026</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>22,10,10,0,2,7,4,2,20,19,20,2,18</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.095364</td>\n",
       "      <td>-0.045230</td>\n",
       "      <td>-0.019472</td>\n",
       "      <td>0.155381</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>-0.004595</td>\n",
       "      <td>-0.023594</td>\n",
       "      <td>0.038160</td>\n",
       "      <td>0.011138</td>\n",
       "      <td>-0.011087</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>F0026</td>\n",
       "      <td>Proband</td>\n",
       "      <td>Male</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.22</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000058</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>51,51,51,57,51,51,52</td>\n",
       "      <td>3,4,9,10,2,2,7,2,39,8,12,1,17,0,0,1,1,14,29,10...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>37,37,37,39,37,37,37</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>37.285714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,9,11,17,23,5,17</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>-0.007762</td>\n",
       "      <td>0.073817</td>\n",
       "      <td>0.015720</td>\n",
       "      <td>0.006303</td>\n",
       "      <td>0.055403</td>\n",
       "      <td>-0.038327</td>\n",
       "      <td>-0.011381</td>\n",
       "      <td>0.032498</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>F0058</td>\n",
       "      <td>Proband</td>\n",
       "      <td>Male</td>\n",
       "      <td>PB</td>\n",
       "      <td>6.19</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000065</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>30</td>\n",
       "      <td>9,10</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>62|37|36|51|8|2|0|0|0|0|0|7|46|88|91|87|84|66|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>22</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.252774</td>\n",
       "      <td>0.107933</td>\n",
       "      <td>0.199378</td>\n",
       "      <td>0.002387</td>\n",
       "      <td>0.001311</td>\n",
       "      <td>0.008149</td>\n",
       "      <td>0.012198</td>\n",
       "      <td>-0.000825</td>\n",
       "      <td>-0.002550</td>\n",
       "      <td>-0.030490</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>F0065</td>\n",
       "      <td>Proband</td>\n",
       "      <td>Male</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.22</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000066</th>\n",
       "      <td>32</td>\n",
       "      <td>1,12,34</td>\n",
       "      <td>31,43,30</td>\n",
       "      <td>87,0</td>\n",
       "      <td>.</td>\n",
       "      <td>71,77,1,88</td>\n",
       "      <td>91|95|100|99|96|100|99|98|98|96|96|97|94|99|98...</td>\n",
       "      <td>64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|2|6|2|...</td>\n",
       "      <td>.</td>\n",
       "      <td>31</td>\n",
       "      <td>30,34,30</td>\n",
       "      <td>.</td>\n",
       "      <td>31.0</td>\n",
       "      <td>31.333333</td>\n",
       "      <td>.</td>\n",
       "      <td>79</td>\n",
       "      <td>8,42,3</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.75</td>\n",
       "      <td>0.243117</td>\n",
       "      <td>0.049799</td>\n",
       "      <td>0.158347</td>\n",
       "      <td>0.011411</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.007690</td>\n",
       "      <td>-0.006605</td>\n",
       "      <td>0.008261</td>\n",
       "      <td>-0.012000</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>REACH000066</td>\n",
       "      <td>F0065</td>\n",
       "      <td>Mom</td>\n",
       "      <td>Female</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.61</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31.333333</td>\n",
       "      <td>31.0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>0.125</td>\n",
       "      <td>9.0309</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000067</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>15</td>\n",
       "      <td>.</td>\n",
       "      <td>7,0,11,23,55,0,2,3,4,0,0,8,46</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>9|14|13|23|14|22|13|13|29|12|20|2|7|5|9|18|6|4...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.245851</td>\n",
       "      <td>0.155945</td>\n",
       "      <td>0.240343</td>\n",
       "      <td>0.002973</td>\n",
       "      <td>0.002622</td>\n",
       "      <td>-0.000228</td>\n",
       "      <td>0.004678</td>\n",
       "      <td>-0.002355</td>\n",
       "      <td>0.011371</td>\n",
       "      <td>-0.027117</td>\n",
       "      <td>REACH000067</td>\n",
       "      <td>F0065</td>\n",
       "      <td>Dad</td>\n",
       "      <td>Male</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.42</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000681</th>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>16</td>\n",
       "      <td>-6,-8,-6,25</td>\n",
       "      <td>3,47,10,23,3,1,9,9,2,17,1,3,0,6,2,16,4,6,14,11...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|9|59|2|56|0|0|0|0|0|0|0|0|0|6|0|0|11|0|1|9|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>18,17,18,28</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>20.25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4,25,5,11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>0.03125</td>\n",
       "      <td>-0.030336</td>\n",
       "      <td>-0.005133</td>\n",
       "      <td>-0.003167</td>\n",
       "      <td>-0.015673</td>\n",
       "      <td>-0.001873</td>\n",
       "      <td>0.015788</td>\n",
       "      <td>0.017784</td>\n",
       "      <td>-0.002266</td>\n",
       "      <td>-0.019622</td>\n",
       "      <td>-0.022165</td>\n",
       "      <td>REACH000681</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Proband</td>\n",
       "      <td>Male</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.04</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000682</th>\n",
       "      <td>31,28,.,24</td>\n",
       "      <td>2,63,56,6,31,89,81,3,69,52,66,69,2,17,60,2,78,...</td>\n",
       "      <td>12,14</td>\n",
       "      <td>71,20,2,54,41,43,8,3,7,5,8,58,54,94,57,48,4,84...</td>\n",
       "      <td>30,26,-12</td>\n",
       "      <td>80,73,0,0</td>\n",
       "      <td>0|0|0|9|8|4|3|3|0|0|0|0|21|0|0|0|2|8|0|0|3|5|1...</td>\n",
       "      <td>45|97|100|100|100|100|100|22|89|17|92|100|95|1...</td>\n",
       "      <td>100|100|100|81|100|95|100|100|100|100|100|35|1...</td>\n",
       "      <td>30,29,.,28</td>\n",
       "      <td>24,25</td>\n",
       "      <td>30,29,16</td>\n",
       "      <td>29.0</td>\n",
       "      <td>24.5</td>\n",
       "      <td>25.0</td>\n",
       "      <td>8,2,5,3</td>\n",
       "      <td>74,86</td>\n",
       "      <td>70,8,12</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>13</td>\n",
       "      <td>18</td>\n",
       "      <td>0.419355</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>-0.022231</td>\n",
       "      <td>-0.005122</td>\n",
       "      <td>-0.006389</td>\n",
       "      <td>-0.005036</td>\n",
       "      <td>0.004410</td>\n",
       "      <td>0.021141</td>\n",
       "      <td>0.003401</td>\n",
       "      <td>0.009329</td>\n",
       "      <td>-0.021366</td>\n",
       "      <td>0.002156</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Mom</td>\n",
       "      <td>Female</td>\n",
       "      <td>ONT</td>\n",
       "      <td>8.63</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>24.5</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>0.03125</td>\n",
       "      <td>15.0515</td>\n",
       "      <td>≤25 CGG</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000683</th>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td></td>\n",
       "      <td>31,26,23,31</td>\n",
       "      <td>2,6,4,1,1,20,7,4,33,9,19,4,1,1,0,0,6,6,3,2,1,2...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|18|0|0|0|0|0|0|0|0|0|0|1|33|0|0|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30,29,28,30</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>29.25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2,4,29,6</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.029616</td>\n",
       "      <td>-0.004138</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.011085</td>\n",
       "      <td>0.006907</td>\n",
       "      <td>0.004983</td>\n",
       "      <td>0.000198</td>\n",
       "      <td>-0.006297</td>\n",
       "      <td>0.002731</td>\n",
       "      <td>-0.031329</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Dad</td>\n",
       "      <td>Male</td>\n",
       "      <td>ONT</td>\n",
       "      <td>9.97</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000684</th>\n",
       "      <td>30,24,24</td>\n",
       "      <td></td>\n",
       "      <td>23,15,24</td>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>2,4,5,7,3,0,27,1,10,4,2,6,6,0,28,1,6,12,1,87,5...</td>\n",
       "      <td>3|58|0|2|5|0|0|0|0|0|0|0|0|0|0|0|0|0|0|14|0|0|...</td>\n",
       "      <td>0|0|0|8|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|2|4|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>30,28,28</td>\n",
       "      <td>28,25,28</td>\n",
       "      <td>.</td>\n",
       "      <td>28.666667</td>\n",
       "      <td>27.0</td>\n",
       "      <td>.</td>\n",
       "      <td>2,7,3</td>\n",
       "      <td>1,13,3</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>0.0625</td>\n",
       "      <td>-0.021597</td>\n",
       "      <td>-0.007737</td>\n",
       "      <td>-0.005243</td>\n",
       "      <td>-0.009369</td>\n",
       "      <td>0.003756</td>\n",
       "      <td>0.019086</td>\n",
       "      <td>0.016933</td>\n",
       "      <td>-0.002215</td>\n",
       "      <td>-0.021031</td>\n",
       "      <td>-0.017064</td>\n",
       "      <td>REACH000684</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>Male</td>\n",
       "      <td>ONT</td>\n",
       "      <td>7.84</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000685</th>\n",
       "      <td>.</td>\n",
       "      <td>23</td>\n",
       "      <td>.</td>\n",
       "      <td>2,9</td>\n",
       "      <td>13,17,33,29,16</td>\n",
       "      <td>18,31,2,13,35,6,2,8,21,40,4,3,24,17,2,17,16,6,...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>24,26,31,30,25</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>27.2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,1,5,6,3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>43</td>\n",
       "      <td>0.022727</td>\n",
       "      <td>-0.029528</td>\n",
       "      <td>-0.007553</td>\n",
       "      <td>-0.003812</td>\n",
       "      <td>-0.009731</td>\n",
       "      <td>0.007237</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>0.003038</td>\n",
       "      <td>0.001101</td>\n",
       "      <td>-0.023349</td>\n",
       "      <td>-0.004458</td>\n",
       "      <td>REACH000685</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>Male</td>\n",
       "      <td>ONT</td>\n",
       "      <td>11.88</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>FALSE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>270 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              bp_dev_h1                        xci_read_mean_methyl_lik_h1  \\\n",
       "REACH000026           .                                                      \n",
       "REACH000058           .                                                      \n",
       "REACH000065           .                                                      \n",
       "REACH000066          32                                            1,12,34   \n",
       "REACH000067           .                                                      \n",
       "...                 ...                                                ...   \n",
       "REACH000681           .                                                  5   \n",
       "REACH000682  31,28,.,24  2,63,56,6,31,89,81,3,69,52,66,69,2,17,60,2,78,...   \n",
       "REACH000683           .                                                      \n",
       "REACH000684    30,24,24                                                      \n",
       "REACH000685           .                                                 23   \n",
       "\n",
       "            bp_dev_h2                        xci_read_mean_methyl_lik_h2  \\\n",
       "REACH000026         .                                                      \n",
       "REACH000058         .                                                      \n",
       "REACH000065         .                                                      \n",
       "REACH000066  31,43,30                                               87,0   \n",
       "REACH000067         .                                                 15   \n",
       "...               ...                                                ...   \n",
       "REACH000681         .                                                 16   \n",
       "REACH000682     12,14  71,20,2,54,41,43,8,3,7,5,8,58,54,94,57,48,4,84...   \n",
       "REACH000683         .                                                      \n",
       "REACH000684  23,15,24                                                      \n",
       "REACH000685         .                                                2,9   \n",
       "\n",
       "                        bp_dev_h0  \\\n",
       "REACH000026                     .   \n",
       "REACH000058  51,51,51,57,51,51,52   \n",
       "REACH000065                    30   \n",
       "REACH000066                     .   \n",
       "REACH000067                     .   \n",
       "...                           ...   \n",
       "REACH000681           -6,-8,-6,25   \n",
       "REACH000682             30,26,-12   \n",
       "REACH000683           31,26,23,31   \n",
       "REACH000684                     .   \n",
       "REACH000685        13,17,33,29,16   \n",
       "\n",
       "                                   xci_read_mean_methyl_lik_h0  \\\n",
       "REACH000026                   22,10,10,0,2,7,4,2,20,19,20,2,18   \n",
       "REACH000058  3,4,9,10,2,2,7,2,39,8,12,1,17,0,0,1,1,14,29,10...   \n",
       "REACH000065                                               9,10   \n",
       "REACH000066                                         71,77,1,88   \n",
       "REACH000067                      7,0,11,23,55,0,2,3,4,0,0,8,46   \n",
       "...                                                        ...   \n",
       "REACH000681  3,47,10,23,3,1,9,9,2,17,1,3,0,6,2,16,4,6,14,11...   \n",
       "REACH000682                                          80,73,0,0   \n",
       "REACH000683  2,6,4,1,1,20,7,4,33,9,19,4,1,1,0,0,6,6,3,2,1,2...   \n",
       "REACH000684  2,4,5,7,3,0,27,1,10,4,2,6,6,0,28,1,6,12,1,87,5...   \n",
       "REACH000685  18,31,2,13,35,6,2,8,21,40,4,3,24,17,2,17,16,6,...   \n",
       "\n",
       "                                                     methyl_h1  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058                                                  .   \n",
       "REACH000065                                                  .   \n",
       "REACH000066  91|95|100|99|96|100|99|98|98|96|96|97|94|99|98...   \n",
       "REACH000067                                                  .   \n",
       "...                                                        ...   \n",
       "REACH000681                                                  .   \n",
       "REACH000682  0|0|0|9|8|4|3|3|0|0|0|0|21|0|0|0|2|8|0|0|3|5|1...   \n",
       "REACH000683                                                  .   \n",
       "REACH000684  3|58|0|2|5|0|0|0|0|0|0|0|0|0|0|0|0|0|0|14|0|0|...   \n",
       "REACH000685                                                  .   \n",
       "\n",
       "                                                     methyl_h2  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058                                                  .   \n",
       "REACH000065                                                  .   \n",
       "REACH000066  64|20|7|2|0|0|0|0|0|0|2|0|1|0|0|0|0|0|3|2|6|2|...   \n",
       "REACH000067                                                  .   \n",
       "...                                                        ...   \n",
       "REACH000681                                                  .   \n",
       "REACH000682  45|97|100|100|100|100|100|22|89|17|92|100|95|1...   \n",
       "REACH000683                                                  .   \n",
       "REACH000684  0|0|0|8|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|2|4|0|...   \n",
       "REACH000685                                                  .   \n",
       "\n",
       "                                                     methyl_h0  \\\n",
       "REACH000026                                                  .   \n",
       "REACH000058  0|0|0|0|0|0|0|1|2|32|27|13|7|5|46|27|29|15|3|4...   \n",
       "REACH000065  62|37|36|51|8|2|0|0|0|0|0|7|46|88|91|87|84|66|...   \n",
       "REACH000066                                                  .   \n",
       "REACH000067  9|14|13|23|14|22|13|13|29|12|20|2|7|5|9|18|6|4...   \n",
       "...                                                        ...   \n",
       "REACH000681  0|9|59|2|56|0|0|0|0|0|0|0|0|0|6|0|0|11|0|1|9|0...   \n",
       "REACH000682  100|100|100|81|100|95|100|100|100|100|100|35|1...   \n",
       "REACH000683  0|0|0|18|0|0|0|0|0|0|0|0|0|0|1|33|0|0|0|0|0|0|...   \n",
       "REACH000684                                                  .   \n",
       "REACH000685  2|0|15|0|2|4|35|71|4|0|1|7|0|10|0|2|0|0|0|0|58...   \n",
       "\n",
       "            repeat_length_h1 repeat_length_h2      repeat_length_h0  \\\n",
       "REACH000026                .                .                     .   \n",
       "REACH000058                .                .  37,37,37,39,37,37,37   \n",
       "REACH000065                .                .                    30   \n",
       "REACH000066               31         30,34,30                     .   \n",
       "REACH000067                .                .                     .   \n",
       "...                      ...              ...                   ...   \n",
       "REACH000681                .                .           18,17,18,28   \n",
       "REACH000682       30,29,.,28            24,25              30,29,16   \n",
       "REACH000683                .                .           30,29,28,30   \n",
       "REACH000684         30,28,28         28,25,28                     .   \n",
       "REACH000685                .                .        24,26,31,30,25   \n",
       "\n",
       "            mean_repeat_length_h1 mean_repeat_length_h2 mean_repeat_length_h0  \\\n",
       "REACH000026                     .                     .                     .   \n",
       "REACH000058                     .                     .             37.285714   \n",
       "REACH000065                     .                     .                  30.0   \n",
       "REACH000066                  31.0             31.333333                     .   \n",
       "REACH000067                     .                     .                     .   \n",
       "...                           ...                   ...                   ...   \n",
       "REACH000681                     .                     .                 20.25   \n",
       "REACH000682                  29.0                  24.5                  25.0   \n",
       "REACH000683                     .                     .                 29.25   \n",
       "REACH000684             28.666667                  27.0                     .   \n",
       "REACH000685                     .                     .                  27.2   \n",
       "\n",
       "            read_mean_methyl_lik_h1 read_mean_methyl_lik_h2  \\\n",
       "REACH000026                       .                       .   \n",
       "REACH000058                       .                       .   \n",
       "REACH000065                       .                       .   \n",
       "REACH000066                      79                  8,42,3   \n",
       "REACH000067                       .                       .   \n",
       "...                             ...                     ...   \n",
       "REACH000681                       .                       .   \n",
       "REACH000682                 8,2,5,3                   74,86   \n",
       "REACH000683                       .                       .   \n",
       "REACH000684                   2,7,3                  1,13,3   \n",
       "REACH000685                       .                       .   \n",
       "\n",
       "            read_mean_methyl_lik_h0 num_methyl_reads_h1 num_unmethyl_reads_h1  \\\n",
       "REACH000026                       .                   .                     .   \n",
       "REACH000058       8,9,11,17,23,5,17                   .                     .   \n",
       "REACH000065                      22                   .                     .   \n",
       "REACH000066                       .                   1                     0   \n",
       "REACH000067                      11                   .                     .   \n",
       "...                             ...                 ...                   ...   \n",
       "REACH000681               4,25,5,11                   .                     .   \n",
       "REACH000682                 70,8,12                   0                     4   \n",
       "REACH000683                2,4,29,6                   .                     .   \n",
       "REACH000684                       .                   0                     3   \n",
       "REACH000685               8,1,5,6,3                   .                     .   \n",
       "\n",
       "            frac_hap_methyl_h1 num_methyl_reads_h2 num_unmethyl_reads_h2  \\\n",
       "REACH000026                  .                   .                     .   \n",
       "REACH000058                  .                   .                     .   \n",
       "REACH000065                  .                   .                     .   \n",
       "REACH000066                1.0                   0                     3   \n",
       "REACH000067                  .                   .                     .   \n",
       "...                        ...                 ...                   ...   \n",
       "REACH000681                  .                   .                     .   \n",
       "REACH000682                0.0                   2                     0   \n",
       "REACH000683                  .                   .                     .   \n",
       "REACH000684                0.0                   0                     3   \n",
       "REACH000685                  .                   .                     .   \n",
       "\n",
       "            frac_hap_methyl_h2 num_methyl_reads_h0 num_unmethyl_reads_h0  \\\n",
       "REACH000026                  .                   .                     .   \n",
       "REACH000058                  .                   0                     7   \n",
       "REACH000065                  .                   0                     1   \n",
       "REACH000066                0.0                   .                     .   \n",
       "REACH000067                  .                   0                     1   \n",
       "...                        ...                 ...                   ...   \n",
       "REACH000681                  .                   0                     4   \n",
       "REACH000682                1.0                   1                     2   \n",
       "REACH000683                  .                   0                     4   \n",
       "REACH000684                0.0                   .                     .   \n",
       "REACH000685                  .                   0                     5   \n",
       "\n",
       "            frac_hap_methyl_h0 xci_num_methyl_reads_h1  \\\n",
       "REACH000026                  .                       .   \n",
       "REACH000058                0.0                       .   \n",
       "REACH000065                0.0                       .   \n",
       "REACH000066                  .                       0   \n",
       "REACH000067                0.0                       .   \n",
       "...                        ...                     ...   \n",
       "REACH000681                0.0                       0   \n",
       "REACH000682           0.333333                      13   \n",
       "REACH000683                0.0                       .   \n",
       "REACH000684                  .                       .   \n",
       "REACH000685                0.0                       0   \n",
       "\n",
       "            xci_num_unmethyl_reads_h1 xci_frac_hap_methyl_h1  \\\n",
       "REACH000026                         .                      .   \n",
       "REACH000058                         .                      .   \n",
       "REACH000065                         .                      .   \n",
       "REACH000066                         3                    0.0   \n",
       "REACH000067                         .                      .   \n",
       "...                               ...                    ...   \n",
       "REACH000681                         1                    0.0   \n",
       "REACH000682                        18               0.419355   \n",
       "REACH000683                         .                      .   \n",
       "REACH000684                         .                      .   \n",
       "REACH000685                         1                    0.0   \n",
       "\n",
       "            xci_num_methyl_reads_h2 xci_num_unmethyl_reads_h2  \\\n",
       "REACH000026                       .                         .   \n",
       "REACH000058                       .                         .   \n",
       "REACH000065                       .                         .   \n",
       "REACH000066                       1                         1   \n",
       "REACH000067                       0                         1   \n",
       "...                             ...                       ...   \n",
       "REACH000681                       0                         1   \n",
       "REACH000682                       8                        20   \n",
       "REACH000683                       .                         .   \n",
       "REACH000684                       .                         .   \n",
       "REACH000685                       0                         2   \n",
       "\n",
       "            xci_frac_hap_methyl_h2 xci_num_methyl_reads_h0  \\\n",
       "REACH000026                      .                       0   \n",
       "REACH000058                      .                       0   \n",
       "REACH000065                      .                       0   \n",
       "REACH000066                    0.5                       3   \n",
       "REACH000067                    0.0                       1   \n",
       "...                            ...                     ...   \n",
       "REACH000681                    0.0                       1   \n",
       "REACH000682               0.285714                       2   \n",
       "REACH000683                      .                       0   \n",
       "REACH000684                      .                       2   \n",
       "REACH000685                    0.0                       1   \n",
       "\n",
       "            xci_num_unmethyl_reads_h0 xci_frac_hap_methyl_h0       PC1  \\\n",
       "REACH000026                        13                    0.0  0.095364   \n",
       "REACH000058                        28                    0.0  0.002422   \n",
       "REACH000065                         2                    0.0  0.252774   \n",
       "REACH000066                         1                   0.75  0.243117   \n",
       "REACH000067                        12               0.076923  0.245851   \n",
       "...                               ...                    ...       ...   \n",
       "REACH000681                        31                0.03125 -0.030336   \n",
       "REACH000682                         2                    0.5 -0.022231   \n",
       "REACH000683                        29                    0.0 -0.029616   \n",
       "REACH000684                        30                 0.0625 -0.021597   \n",
       "REACH000685                        43               0.022727 -0.029528   \n",
       "\n",
       "                  PC2       PC3       PC4       PC5       PC6       PC7  \\\n",
       "REACH000026 -0.045230 -0.019472  0.155381  0.000061 -0.004595 -0.023594   \n",
       "REACH000058 -0.009216 -0.007762  0.073817  0.015720  0.006303  0.055403   \n",
       "REACH000065  0.107933  0.199378  0.002387  0.001311  0.008149  0.012198   \n",
       "REACH000066  0.049799  0.158347  0.011411  0.004728  0.007690 -0.006605   \n",
       "REACH000067  0.155945  0.240343  0.002973  0.002622 -0.000228  0.004678   \n",
       "...               ...       ...       ...       ...       ...       ...   \n",
       "REACH000681 -0.005133 -0.003167 -0.015673 -0.001873  0.015788  0.017784   \n",
       "REACH000682 -0.005122 -0.006389 -0.005036  0.004410  0.021141  0.003401   \n",
       "REACH000683 -0.004138  0.000015 -0.011085  0.006907  0.004983  0.000198   \n",
       "REACH000684 -0.007737 -0.005243 -0.009369  0.003756  0.019086  0.016933   \n",
       "REACH000685 -0.007553 -0.003812 -0.009731  0.007237  0.016614  0.003038   \n",
       "\n",
       "                  PC8       PC9      PC10    sample_id    fid      rel  \\\n",
       "REACH000026  0.038160  0.011138 -0.011087  REACH000026  F0026  Proband   \n",
       "REACH000058 -0.038327 -0.011381  0.032498  REACH000058  F0058  Proband   \n",
       "REACH000065 -0.000825 -0.002550 -0.030490  REACH000065  F0065  Proband   \n",
       "REACH000066  0.008261 -0.012000 -0.009906  REACH000066  F0065      Mom   \n",
       "REACH000067 -0.002355  0.011371 -0.027117  REACH000067  F0065      Dad   \n",
       "...               ...       ...       ...          ...    ...      ...   \n",
       "REACH000681 -0.002266 -0.019622 -0.022165  REACH000681  F0270  Proband   \n",
       "REACH000682  0.009329 -0.021366  0.002156  REACH000682  F0270      Mom   \n",
       "REACH000683 -0.006297  0.002731 -0.031329  REACH000683  F0270      Dad   \n",
       "REACH000684 -0.002215 -0.021031 -0.017064  REACH000684  F0270  Sibling   \n",
       "REACH000685  0.001101 -0.023349 -0.004458  REACH000685  F0270  Sibling   \n",
       "\n",
       "                sex Platform  coverage Affected  Phenotype in_trios  \\\n",
       "REACH000026    Male       PB      3.22      Yes          1     TRUE   \n",
       "REACH000058    Male       PB      6.19      Yes          1     TRUE   \n",
       "REACH000065    Male       PB      1.22      Yes          1     TRUE   \n",
       "REACH000066  Female       PB      1.61       No          0     TRUE   \n",
       "REACH000067    Male       PB      1.42       No          0     TRUE   \n",
       "...             ...      ...       ...      ...        ...      ...   \n",
       "REACH000681    Male      ONT     10.04      Yes          1     TRUE   \n",
       "REACH000682  Female      ONT      8.63       No          0     TRUE   \n",
       "REACH000683    Male      ONT      9.97       No          0     TRUE   \n",
       "REACH000684    Male      ONT      7.84       No          0     TRUE   \n",
       "REACH000685    Male      ONT     11.88       No          0     TRUE   \n",
       "\n",
       "            frac_hap_methyl_large_rep frac_hap_methyl_small_rep  \\\n",
       "REACH000026                         .                         .   \n",
       "REACH000058                         .                         .   \n",
       "REACH000065                         .                         .   \n",
       "REACH000066                       0.0                       1.0   \n",
       "REACH000067                         .                         .   \n",
       "...                               ...                       ...   \n",
       "REACH000681                         .                         .   \n",
       "REACH000682                       0.0                       1.0   \n",
       "REACH000683                         .                         .   \n",
       "REACH000684                         .                         .   \n",
       "REACH000685                         .                         .   \n",
       "\n",
       "            mean_large_rep mean_small_rep has_both_hps binomial_pval  \\\n",
       "REACH000026              .              .        FALSE             .   \n",
       "REACH000058              .              .        FALSE             .   \n",
       "REACH000065              .              .        FALSE             .   \n",
       "REACH000066      31.333333           31.0         TRUE         0.125   \n",
       "REACH000067              .              .        FALSE             .   \n",
       "...                    ...            ...          ...           ...   \n",
       "REACH000681              .              .        FALSE             .   \n",
       "REACH000682           29.0           24.5         TRUE       0.03125   \n",
       "REACH000683              .              .        FALSE             .   \n",
       "REACH000684              .              .        FALSE             .   \n",
       "REACH000685              .              .        FALSE             .   \n",
       "\n",
       "            binomial_phred repeat_group both_hps_2_reads both_hps_3_reads  \n",
       "REACH000026              .            .                .                .  \n",
       "REACH000058              .            .                .                .  \n",
       "REACH000065              .            .                .                .  \n",
       "REACH000066         9.0309    26-34 CGG            FALSE            FALSE  \n",
       "REACH000067              .            .                .                .  \n",
       "...                    ...          ...              ...              ...  \n",
       "REACH000681              .            .                .                .  \n",
       "REACH000682        15.0515      ≤25 CGG             TRUE            FALSE  \n",
       "REACH000683              .            .                .                .  \n",
       "REACH000684              .            .                .                .  \n",
       "REACH000685              .            .                .                .  \n",
       "\n",
       "[270 rows x 65 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputs/methylation_table_complete.tsv\n"
     ]
    }
   ],
   "source": [
    "### write the final table\n",
    "for i in range(1,11):\n",
    "    df_methyl[f'PC{i}'] = df_methyl.index.str.split('_').str[0].map(df_pca_flt.set_index('iid_orig')[f'PC{i}'])\n",
    "df_methyl['sample_id'] = df_methyl.index.map(lambda x: x if 'REACH000226' not in x else 'REACH000226')\n",
    "df_methyl['fid'] = df_methyl.sample_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Family_ID)\n",
    "df_methyl['rel'] = df_methyl.sample_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Rel_Proband)\n",
    "df_methyl['sex'] = df_methyl.sample_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Sex.apply(lambda x: 'Male' if x=='M' else 'Female'))\n",
    "df_methyl['Platform'] = df_methyl.sample_id.map(df_cov.set_index('SAMPLE').COHORT) # df_cov has the sample/platform naming inside\n",
    "df_methyl['coverage'] = df_methyl.sample_id.map(df_cov.set_index('SAMPLE').MEAN_COVERAGE) # df_cov has the sample/platform naming inside\n",
    "df_methyl['Affected'] = df_methyl.sample_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Affected)\n",
    "df_methyl['Phenotype'] = df_methyl.Affected.map(lambda x: 1 if x=='Yes' else 0)\n",
    "df_methyl['in_trios'] = df_methyl.index.map(lambda x: 'TRUE' if x in df_pca_flt.iid.tolist() or x=='REACH000226_sperm' else 'FALSE')\n",
    "\n",
    "### for those subjects with both haplotypes present (must be females), compute large and small haplotype stats (frac, mean rep, skewness)\n",
    "### haplotype of col_in_1 must correspond to col_out_1 and the same for _2\n",
    "### output is large: first, small: second\n",
    "def get_large_small_vals(row, col_in_1, col_in_2, col_out_1, col_out_2):\n",
    "    if row.sex != 'Female':\n",
    "        return '.', '.'\n",
    "    if row.num_methyl_reads_h1 == '.' or row.num_methyl_reads_h2 == '.':\n",
    "        return '.', '.'\n",
    "    if row[col_in_1] >= row[col_in_2]:\n",
    "        out = (row[col_out_1], row[col_out_2])\n",
    "    else:\n",
    "        out = (row[col_out_2], row[col_out_1])\n",
    "    return out\n",
    "\n",
    "\n",
    "### figure out which hp is the large and with is the small CGG repeat\n",
    "col_in_1 = 'mean_repeat_length_h1'\n",
    "col_in_2 = 'mean_repeat_length_h2'\n",
    "col_out_1 = 'frac_hap_methyl_h1'\n",
    "col_out_2 = 'frac_hap_methyl_h2'\n",
    "df_methyl[['frac_hap_methyl_large_rep', 'frac_hap_methyl_small_rep']] = df_methyl.apply(lambda row: get_large_small_vals(row, col_in_1, col_in_2, \n",
    "                                                                                                                         col_out_1, col_out_2), axis=1, result_type = 'expand')\n",
    "\n",
    "col_in_1 = 'mean_repeat_length_h1'\n",
    "col_in_2 = 'mean_repeat_length_h2'\n",
    "col_out_1 = 'mean_repeat_length_h1'\n",
    "col_out_2 = 'mean_repeat_length_h2'\n",
    "df_methyl[['mean_large_rep', 'mean_small_rep']] = df_methyl.apply(lambda row: get_large_small_vals(row, col_in_1, col_in_2, \n",
    "                                                                                                   col_out_1, col_out_2), axis=1, result_type = 'expand')\n",
    "\n",
    "df_methyl['has_both_hps'] = df_methyl.mean_large_rep.map(lambda x: 'TRUE' if x!='.' else 'FALSE')\n",
    "\n",
    "def get_binomtest_pval(row):\n",
    "    if row.has_both_hps == 'FALSE':\n",
    "        return '.'\n",
    "    k = row.num_methyl_reads_h1 + row.num_unmethyl_reads_h2\n",
    "    n = row.num_methyl_reads_h1 + row.num_unmethyl_reads_h1 + row.num_methyl_reads_h2 + row.num_unmethyl_reads_h2\n",
    "    return binomtest(k, n, p=0.5, alternative='two-sided').pvalue\n",
    "\n",
    "df_methyl['binomial_pval'] = df_methyl.apply(lambda row: get_binomtest_pval(row), axis=1)\n",
    "df_methyl['binomial_phred'] = df_methyl['binomial_pval'].apply(lambda x: -10*np.log10(x) if x!='.' else '.')\n",
    "\n",
    "def get_rep_group(row):\n",
    "    if row.has_both_hps == 'FALSE':\n",
    "        return '.'\n",
    "    out = ('≥35 CGG' if ((row.mean_repeat_length_h1 >= 35) or \n",
    "                         (row.mean_repeat_length_h2 >= 35)) else \n",
    "           ('≤25 CGG' if ((row.mean_repeat_length_h1 <= 25) or \n",
    "                          (row.mean_repeat_length_h2 <= 25)) else '26-34 CGG'))\n",
    "    return out\n",
    "df_methyl['repeat_group'] = df_methyl.apply(lambda row: get_rep_group(row), axis=1)\n",
    "\n",
    "### last column: both haplotype has certain number of reads\n",
    "def get_min_reads(row, num_reads_thr=2):\n",
    "    if row.sex != 'Female':\n",
    "        return '.'\n",
    "    if row.num_methyl_reads_h1 == '.' or row.num_methyl_reads_h2 == '.':\n",
    "        return '.'\n",
    "    if (row.num_methyl_reads_h1 + row.num_unmethyl_reads_h1 >= num_reads_thr) and (row.num_methyl_reads_h2 + row.num_unmethyl_reads_h2 >= num_reads_thr):\n",
    "        return 'TRUE'\n",
    "    else:\n",
    "        return 'FALSE'\n",
    "\n",
    "num_reads_thr = 2\n",
    "df_methyl['both_hps_2_reads'] = df_methyl.apply(lambda row: get_min_reads(row, num_reads_thr), axis=1)\n",
    "num_reads_thr = 3\n",
    "df_methyl['both_hps_3_reads'] = df_methyl.apply(lambda row: get_min_reads(row, num_reads_thr), axis=1)\n",
    "\n",
    "print(df_methyl['in_trios'].value_counts())\n",
    "print(df_methyl['Affected'].value_counts())\n",
    "\n",
    "display(df_methyl)\n",
    "### write the table\n",
    "file_name = f'{dir_out}/methylation_table_complete.tsv'\n",
    "print(file_name)\n",
    "df_methyl.to_csv(file_name, sep='\\t', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ac422ec5-ebfd-436b-a414-18d895b51401",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_out:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/smmortazavi/micromamba/envs/bioinf/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.\n",
      "  return _methods._mean(a, axis=axis, dtype=dtype,\n",
      "/home/smmortazavi/micromamba/envs/bioinf/lib/python3.10/site-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject_id</th>\n",
       "      <th>sex</th>\n",
       "      <th>haplotype</th>\n",
       "      <th>mean_repeat_length</th>\n",
       "      <th>repeat_group</th>\n",
       "      <th>frac_hap_methyl</th>\n",
       "      <th>xci_frac_hap_methyl</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>xci_num_reads</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>PC9</th>\n",
       "      <th>PC10</th>\n",
       "      <th>fid</th>\n",
       "      <th>rel</th>\n",
       "      <th>Platform</th>\n",
       "      <th>coverage</th>\n",
       "      <th>Affected</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>in_trios</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>REACH000066</td>\n",
       "      <td>Female</td>\n",
       "      <td>h2</td>\n",
       "      <td>31.333333</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.243117</td>\n",
       "      <td>0.049799</td>\n",
       "      <td>0.158347</td>\n",
       "      <td>0.011411</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.007690</td>\n",
       "      <td>-0.006605</td>\n",
       "      <td>0.008261</td>\n",
       "      <td>-0.012000</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>F0065</td>\n",
       "      <td>Mom</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.61</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000087</td>\n",
       "      <td>Female</td>\n",
       "      <td>h2</td>\n",
       "      <td>29.500000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>4.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.129968</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "      <td>F0078</td>\n",
       "      <td>Mom</td>\n",
       "      <td>PB</td>\n",
       "      <td>2.75</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACH000089</td>\n",
       "      <td>Female</td>\n",
       "      <td>h2</td>\n",
       "      <td>30.333333</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.049880</td>\n",
       "      <td>0.047204</td>\n",
       "      <td>0.069405</td>\n",
       "      <td>0.015002</td>\n",
       "      <td>0.011334</td>\n",
       "      <td>0.011519</td>\n",
       "      <td>-0.001072</td>\n",
       "      <td>0.014418</td>\n",
       "      <td>-0.017609</td>\n",
       "      <td>-0.005676</td>\n",
       "      <td>F0078</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.27</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACH000092</td>\n",
       "      <td>Female</td>\n",
       "      <td>h2</td>\n",
       "      <td>29.800000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.028511</td>\n",
       "      <td>-0.001002</td>\n",
       "      <td>-0.001921</td>\n",
       "      <td>0.004426</td>\n",
       "      <td>-0.001891</td>\n",
       "      <td>0.006074</td>\n",
       "      <td>0.027474</td>\n",
       "      <td>0.004611</td>\n",
       "      <td>0.015596</td>\n",
       "      <td>0.030866</td>\n",
       "      <td>F0081</td>\n",
       "      <td>Proband</td>\n",
       "      <td>PB</td>\n",
       "      <td>4.89</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACH000093</td>\n",
       "      <td>Female</td>\n",
       "      <td>h1</td>\n",
       "      <td>29.250000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.277778</td>\n",
       "      <td>4.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>-0.027082</td>\n",
       "      <td>0.003555</td>\n",
       "      <td>-0.003174</td>\n",
       "      <td>-0.000206</td>\n",
       "      <td>-0.004555</td>\n",
       "      <td>0.000972</td>\n",
       "      <td>-0.004064</td>\n",
       "      <td>-0.011024</td>\n",
       "      <td>0.005814</td>\n",
       "      <td>-0.009496</td>\n",
       "      <td>F0081</td>\n",
       "      <td>Mom</td>\n",
       "      <td>PB</td>\n",
       "      <td>4.63</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>REACH000663</td>\n",
       "      <td>Male</td>\n",
       "      <td>hm</td>\n",
       "      <td>33.166667</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025000</td>\n",
       "      <td>6.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.464341</td>\n",
       "      <td>0.385613</td>\n",
       "      <td>-0.153238</td>\n",
       "      <td>-0.033060</td>\n",
       "      <td>0.029758</td>\n",
       "      <td>0.012759</td>\n",
       "      <td>0.018026</td>\n",
       "      <td>0.006101</td>\n",
       "      <td>-0.000833</td>\n",
       "      <td>-0.016008</td>\n",
       "      <td>F0266</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.64</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>REACH000681</td>\n",
       "      <td>Male</td>\n",
       "      <td>hm</td>\n",
       "      <td>20.250000</td>\n",
       "      <td>≤25 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.029412</td>\n",
       "      <td>4.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-0.030336</td>\n",
       "      <td>-0.005133</td>\n",
       "      <td>-0.003167</td>\n",
       "      <td>-0.015673</td>\n",
       "      <td>-0.001873</td>\n",
       "      <td>0.015788</td>\n",
       "      <td>0.017784</td>\n",
       "      <td>-0.002266</td>\n",
       "      <td>-0.019622</td>\n",
       "      <td>-0.022165</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Proband</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.04</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>REACH000683</td>\n",
       "      <td>Male</td>\n",
       "      <td>hm</td>\n",
       "      <td>29.250000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>-0.029616</td>\n",
       "      <td>-0.004138</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>-0.011085</td>\n",
       "      <td>0.006907</td>\n",
       "      <td>0.004983</td>\n",
       "      <td>0.000198</td>\n",
       "      <td>-0.006297</td>\n",
       "      <td>0.002731</td>\n",
       "      <td>-0.031329</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Dad</td>\n",
       "      <td>ONT</td>\n",
       "      <td>9.97</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>REACH000684</td>\n",
       "      <td>Male</td>\n",
       "      <td>hm</td>\n",
       "      <td>27.833333</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062500</td>\n",
       "      <td>6.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>-0.021597</td>\n",
       "      <td>-0.007737</td>\n",
       "      <td>-0.005243</td>\n",
       "      <td>-0.009369</td>\n",
       "      <td>0.003756</td>\n",
       "      <td>0.019086</td>\n",
       "      <td>0.016933</td>\n",
       "      <td>-0.002215</td>\n",
       "      <td>-0.021031</td>\n",
       "      <td>-0.017064</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>ONT</td>\n",
       "      <td>7.84</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>REACH000685</td>\n",
       "      <td>Male</td>\n",
       "      <td>hm</td>\n",
       "      <td>27.200000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021277</td>\n",
       "      <td>5.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.029528</td>\n",
       "      <td>-0.007553</td>\n",
       "      <td>-0.003812</td>\n",
       "      <td>-0.009731</td>\n",
       "      <td>0.007237</td>\n",
       "      <td>0.016614</td>\n",
       "      <td>0.003038</td>\n",
       "      <td>0.001101</td>\n",
       "      <td>-0.023349</td>\n",
       "      <td>-0.004458</td>\n",
       "      <td>F0270</td>\n",
       "      <td>Sibling</td>\n",
       "      <td>ONT</td>\n",
       "      <td>11.88</td>\n",
       "      <td>No</td>\n",
       "      <td>0</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      subject_id     sex haplotype  mean_repeat_length repeat_group  \\\n",
       "0    REACH000066  Female        h2           31.333333    26-34 CGG   \n",
       "1    REACH000087  Female        h2           29.500000    26-34 CGG   \n",
       "2    REACH000089  Female        h2           30.333333    26-34 CGG   \n",
       "3    REACH000092  Female        h2           29.800000    26-34 CGG   \n",
       "4    REACH000093  Female        h1           29.250000    26-34 CGG   \n",
       "..           ...     ...       ...                 ...          ...   \n",
       "128  REACH000663    Male        hm           33.166667    26-34 CGG   \n",
       "129  REACH000681    Male        hm           20.250000      ≤25 CGG   \n",
       "130  REACH000683    Male        hm           29.250000    26-34 CGG   \n",
       "131  REACH000684    Male        hm           27.833333    26-34 CGG   \n",
       "132  REACH000685    Male        hm           27.200000    26-34 CGG   \n",
       "\n",
       "     frac_hap_methyl  xci_frac_hap_methyl  num_reads  xci_num_reads       PC1  \\\n",
       "0           0.000000             0.500000        3.0            2.0  0.243117   \n",
       "1           0.750000             0.666667        4.0            9.0  0.129968   \n",
       "2           0.666667             1.000000        3.0            3.0  0.049880   \n",
       "3           0.600000             0.777778        5.0            9.0 -0.028511   \n",
       "4           1.000000             0.277778        4.0           18.0 -0.027082   \n",
       "..               ...                  ...        ...            ...       ...   \n",
       "128         0.000000             0.025000        6.0           40.0  0.464341   \n",
       "129         0.000000             0.029412        4.0           34.0 -0.030336   \n",
       "130         0.000000             0.000000        4.0           29.0 -0.029616   \n",
       "131         0.000000             0.062500        6.0           32.0 -0.021597   \n",
       "132         0.000000             0.021277        5.0           47.0 -0.029528   \n",
       "\n",
       "          PC2       PC3       PC4       PC5       PC6       PC7       PC8  \\\n",
       "0    0.049799  0.158347  0.011411  0.004728  0.007690 -0.006605  0.008261   \n",
       "1    0.080387  0.132056  0.029583  0.020156  0.007971 -0.002383  0.014468   \n",
       "2    0.047204  0.069405  0.015002  0.011334  0.011519 -0.001072  0.014418   \n",
       "3   -0.001002 -0.001921  0.004426 -0.001891  0.006074  0.027474  0.004611   \n",
       "4    0.003555 -0.003174 -0.000206 -0.004555  0.000972 -0.004064 -0.011024   \n",
       "..        ...       ...       ...       ...       ...       ...       ...   \n",
       "128  0.385613 -0.153238 -0.033060  0.029758  0.012759  0.018026  0.006101   \n",
       "129 -0.005133 -0.003167 -0.015673 -0.001873  0.015788  0.017784 -0.002266   \n",
       "130 -0.004138  0.000015 -0.011085  0.006907  0.004983  0.000198 -0.006297   \n",
       "131 -0.007737 -0.005243 -0.009369  0.003756  0.019086  0.016933 -0.002215   \n",
       "132 -0.007553 -0.003812 -0.009731  0.007237  0.016614  0.003038  0.001101   \n",
       "\n",
       "          PC9      PC10    fid      rel Platform  coverage Affected  \\\n",
       "0   -0.012000 -0.009906  F0065      Mom       PB      1.61       No   \n",
       "1    0.006253 -0.018303  F0078      Mom       PB      2.75       No   \n",
       "2   -0.017609 -0.005676  F0078  Sibling       PB      3.27      Yes   \n",
       "3    0.015596  0.030866  F0081  Proband       PB      4.89      Yes   \n",
       "4    0.005814 -0.009496  F0081      Mom       PB      4.63       No   \n",
       "..        ...       ...    ...      ...      ...       ...      ...   \n",
       "128 -0.000833 -0.016008  F0266  Sibling      ONT     10.64       No   \n",
       "129 -0.019622 -0.022165  F0270  Proband      ONT     10.04      Yes   \n",
       "130  0.002731 -0.031329  F0270      Dad      ONT      9.97       No   \n",
       "131 -0.021031 -0.017064  F0270  Sibling      ONT      7.84       No   \n",
       "132 -0.023349 -0.004458  F0270  Sibling      ONT     11.88       No   \n",
       "\n",
       "     Phenotype in_trios  \n",
       "0            0     TRUE  \n",
       "1            0     TRUE  \n",
       "2            1     TRUE  \n",
       "3            1     TRUE  \n",
       "4            0     TRUE  \n",
       "..         ...      ...  \n",
       "128          0     TRUE  \n",
       "129          1     TRUE  \n",
       "130          0     TRUE  \n",
       "131          0     TRUE  \n",
       "132          0     TRUE  \n",
       "\n",
       "[133 rows x 26 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputs/methylation_table_complete_hps.tsv\n"
     ]
    }
   ],
   "source": [
    "### Make a table of haplotype information for clogit/logit models\n",
    "### set females:\n",
    "df_methyl_f = df_methyl.loc[(df_methyl.sex=='Female')].copy()\n",
    "df_methyl_f = df_methyl_f.loc[(df_methyl_f.num_methyl_reads_h1!='.') | (df_methyl_f.num_methyl_reads_h2!='.')]\n",
    "df_methyl_f['num_reads_h1'] = df_methyl_f.apply(lambda row: int(row.num_methyl_reads_h1) + int(row.num_unmethyl_reads_h1) if row.num_methyl_reads_h1!='.' else 0, axis=1)\n",
    "df_methyl_f['num_reads_h2'] = df_methyl_f.apply(lambda row: int(row.num_methyl_reads_h2) + int(row.num_unmethyl_reads_h2) if row.num_methyl_reads_h2!='.' else 0, axis=1)\n",
    "df_methyl_f['xci_num_reads_h1'] = df_methyl_f.apply(lambda row: int(row.xci_num_methyl_reads_h1) + int(row.xci_num_unmethyl_reads_h1) if row.xci_num_methyl_reads_h1!='.' else 0, axis=1)\n",
    "df_methyl_f['xci_num_reads_h2'] = df_methyl_f.apply(lambda row: int(row.xci_num_methyl_reads_h2) + int(row.xci_num_unmethyl_reads_h2) if row.xci_num_methyl_reads_h2!='.' else 0, axis=1)\n",
    "df_methyl_f = df_methyl_f.loc[(df_methyl_f.num_reads_h1>=3) | (df_methyl_f.num_reads_h2>=3)]\n",
    "\n",
    "df_out_f = pd.DataFrame()\n",
    "for idx, row in df_methyl_f.iterrows():\n",
    "    sample = idx\n",
    "    for hp in ['h1', 'h2']:\n",
    "        if row[f'num_reads_{hp}'] >= 3:\n",
    "            mean_rep = row[f'mean_repeat_length_{hp}']\n",
    "            frac_hap = row[f'frac_hap_methyl_{hp}']\n",
    "            xci_frac_hap = row[f'xci_frac_hap_methyl_{hp}']\n",
    "            num_reads = row[f'num_reads_{hp}']\n",
    "            xci_num_reads = row[f'xci_num_reads_{hp}']\n",
    "            rep_group = row['repeat_group']\n",
    "            df_tmp = pd.DataFrame({'subject_id': [idx], 'sex': ['Female'], 'haplotype': [hp], \n",
    "                                   f'mean_repeat_length': [mean_rep], f'repeat_group': [rep_group], \n",
    "                                   f'frac_hap_methyl': [frac_hap], f'xci_frac_hap_methyl': [xci_frac_hap], \n",
    "                                   f'num_reads': [num_reads], f'xci_num_reads': [xci_num_reads]})\n",
    "            df_out_f = pd.concat([df_out_f, df_tmp], ignore_index=True)\n",
    "\n",
    "### also get rid of the haplotypes where xci_frac_hap_methyl are missing.\n",
    "df_out_f = df_out_f.loc[df_out_f.xci_frac_hap_methyl!='.']\n",
    "\n",
    "tot_cols = ['subject_id', 'sex', 'haplotype', 'mean_repeat_length', 'repeat_group', \n",
    "            'frac_hap_methyl', 'xci_frac_hap_methyl', 'num_reads', 'xci_num_reads']\n",
    "\n",
    "# set male subjects\n",
    "df_methyl_m = df_methyl.loc[(df_methyl.sex=='Male')].copy()\n",
    "df_methyl_m = df_methyl_m.loc[(df_methyl_m.num_methyl_reads_h1!='.') | (df_methyl_m.num_methyl_reads_h2!='.') | (df_methyl_m.num_methyl_reads_h0!='.')]\n",
    "df_methyl_m['repeat_lengths'] = df_methyl_m.apply(lambda row: ','.join([x for x in [row.repeat_length_h1, row.repeat_length_h2, row.repeat_length_h0] if x!='.']), axis=1)\n",
    "df_methyl_m['mean_repeat_length'] = df_methyl_m.repeat_lengths.map(lambda x: np.mean([int(xx) for xx in x.split(',') if xx!='.' and xx!='']))\n",
    "df_methyl_m['num_methyl_reads'] = df_methyl_m.apply(lambda row: np.sum([int(x) for x in [row.num_methyl_reads_h1, row.num_methyl_reads_h2, row.num_methyl_reads_h0] if x!='.']), axis=1)\n",
    "df_methyl_m['num_unmethyl_reads'] = df_methyl_m.apply(lambda row: np.sum([int(x) for x in [row.num_unmethyl_reads_h1, row.num_unmethyl_reads_h2, row.num_unmethyl_reads_h0] if x!='.']), axis=1)\n",
    "df_methyl_m['num_reads'] = df_methyl_m['num_methyl_reads'] + df_methyl_m['num_unmethyl_reads']\n",
    "df_methyl_m['frac_hap_methyl'] = df_methyl_m['num_methyl_reads'] / (df_methyl_m['num_methyl_reads'] + df_methyl_m['num_unmethyl_reads'])\n",
    "df_methyl_m['xci_num_methyl_reads'] = df_methyl_m.apply(lambda row: np.sum([int(x) for x in [row.xci_num_methyl_reads_h1, row.xci_num_methyl_reads_h2, row.xci_num_methyl_reads_h0] if x!='.']), axis=1)\n",
    "df_methyl_m['xci_num_unmethyl_reads'] = df_methyl_m.apply(lambda row: np.sum([int(x) for x in [row.xci_num_unmethyl_reads_h1, row.xci_num_unmethyl_reads_h2, row.xci_num_unmethyl_reads_h0] if x!='.']), axis=1)\n",
    "df_methyl_m['xci_num_reads'] = df_methyl_m['xci_num_methyl_reads'] + df_methyl_m['xci_num_unmethyl_reads']\n",
    "df_methyl_m['xci_frac_hap_methyl'] = df_methyl_m['xci_num_methyl_reads'] / (df_methyl_m['xci_num_methyl_reads'] + df_methyl_m['xci_num_unmethyl_reads'])\n",
    "df_methyl_m['subject_id'] = df_methyl_m.index\n",
    "df_methyl_m['haplotype'] = 'hm'\n",
    "\n",
    "df_out_m = df_methyl_m.loc[df_methyl_m.num_reads>=4][tot_cols]\n",
    "\n",
    "### concatinate female and male\n",
    "df_out = pd.concat([df_out_f, df_out_m], ignore_index=True)\n",
    "df_out['subject_id'] = df_out.subject_id.apply(lambda x: 'REACH000226' if x == 'REACH000226_sperm' else x)\n",
    "for i in range(1,11):\n",
    "    df_out[f'PC{i}'] = df_out.subject_id.str.split('_').str[0].map(df_pca_flt.set_index('iid_orig')[f'PC{i}'])\n",
    "df_out['fid'] = df_out.subject_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Family_ID)\n",
    "df_out['rel'] = df_out.subject_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Rel_Proband)\n",
    "df_out['Platform'] = df_out.subject_id.map(df_cov.set_index('SAMPLE').COHORT) # df_cov has the sample/platform naming inside\n",
    "df_out['coverage'] = df_out.subject_id.map(df_cov.set_index('SAMPLE').MEAN_COVERAGE) # df_cov has the sample/platform naming inside\n",
    "df_out['Affected'] = df_out.subject_id.str.split('_').str[0].map(df_meta.set_index('Sample_ID_orig').Affected)\n",
    "df_out['Phenotype'] = df_out.Affected.map(lambda x: 1 if x=='Yes' else 0)\n",
    "df_out['in_trios'] = df_out.subject_id.map(lambda x: 'TRUE' if x in df_pca_flt.iid.tolist() or x=='REACH000226_sperm' else 'FALSE')\n",
    "\n",
    "cols = ['mean_repeat_length', 'frac_hap_methyl', 'xci_frac_hap_methyl', 'xci_num_reads', 'num_reads']\n",
    "for col in cols:\n",
    "    df_out[col] = df_out[col].astype(float)\n",
    "\n",
    "def get_rep_group2(row):\n",
    "    if row.repeat_group != '.':\n",
    "        out = row.repeat_group\n",
    "    else:\n",
    "        out = ('≥35 CGG' if (row.mean_repeat_length >= 35) else \n",
    "               ('≤25 CGG' if (row.mean_repeat_length <= 25) else '26-34 CGG'))\n",
    "    return out\n",
    "df_out['repeat_group'] = df_out.apply(lambda row: get_rep_group2(row), axis=1)\n",
    "\n",
    "print('df_out:')\n",
    "display(df_out)\n",
    "\n",
    "### write the table\n",
    "file_name = f'{dir_out}/methylation_table_complete_hps.tsv'\n",
    "print(file_name)\n",
    "df_out.to_csv(file_name, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2b1df19-53c4-479d-97e5-34543139f67a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### make a table of raw data for other select genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "113d319b-b73e-4935-9147-5a2fb369a987",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_og:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>bp_dev_h1</th>\n",
       "      <th>bp_dev_h2</th>\n",
       "      <th>bp_dev_h0</th>\n",
       "      <th>rn_h1</th>\n",
       "      <th>rn_h2</th>\n",
       "      <th>rn_h0</th>\n",
       "      <th>ps_h1</th>\n",
       "      <th>ps_h2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>10620123</td>\n",
       "      <td>10620805</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;2|2|34|21|9|38|52|66|56|18|1...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;22|55|23|34|52|71|37|63|36|3...</td>\n",
       "      <td>2|2|11|4|0|2|33|10|1|0|1|1|1|0|1|4|11|3|5|1|0|...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;0;.;.;.;.;.;.;.;-3,.;.;.;.;....</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;3;.;.;1;.;.;.;.;-6,1;.;.;.;....</td>\n",
       "      <td>0,.;-1,4;1;8,2,3,0;.;8;4,1,0,4;4,0;-4,0;15;-1,...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_210709_004131/129893...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_210709_004131/139856...</td>\n",
       "      <td>m54340U_200716_001732/101845162/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185;.;.;.;.;.;.;.;10185,10...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185;.;.;10185;.;.;.;.;1018...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>11663929</td>\n",
       "      <td>11665723</td>\n",
       "      <td>ARHGAP6_5UTR</td>\n",
       "      <td>.;.;.;12|33|72|92|98|99|98|99|96|93|95|99|99|9...</td>\n",
       "      <td>.;.;.;16|14|19|79|98|80|90|52|57|28|4|5|4|9|24...</td>\n",
       "      <td>99|98|96|94|88|27|58|48|25|20|5|5|2|1|0|0|1|3|...</td>\n",
       "      <td>.;.;.;3;.;.;.;.;0;0;.,0,0,24,0,.;.;.;.;-2;.;.;...</td>\n",
       "      <td>.;.;.;8;.;.;.;.;.;0,25;16,17,1,0,.;.;.;.;2,7,6...</td>\n",
       "      <td>20,15;2,5,7,.;.;.;19,-6;.;.,1,14,-2,1;.,0,2;-1...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/176228322/ccs;.;.;...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/53151007/ccs;.;.;....</td>\n",
       "      <td>m54340U_200716_001732/98502494/ccs,m54340U_200...</td>\n",
       "      <td>.;.;.;10185;.;.;.;.;10215;10185;10185,10185,10...</td>\n",
       "      <td>.;.;.;10185;.;.;.;.;.;10185,10185;10185,10185,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>12791015</td>\n",
       "      <td>12791585</td>\n",
       "      <td>PRPS2_5UTR</td>\n",
       "      <td>.;.;.;96|82|42|36|92|94|91|74|94|99|99|97|97|9...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;1|0|0|0|0|0|0|0|0|7|15|19|8|4|...</td>\n",
       "      <td>0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|8|5|7|2|...</td>\n",
       "      <td>.;.;.;1;.;.;-1,2;.;-1;0,0;-5;.;7,5;.;8,0;.;.;....</td>\n",
       "      <td>.;.;.;.;.;.;.;.;-3;0,-2,0;.;.;0,0,-3,6,2,-1;.;...</td>\n",
       "      <td>1,-2,1;0,0,-2,2,5;.;.;5;3,1,1;.;0,1,1,10,.;.;....</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/120785681/ccs;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;m54340U_210703_193517/36570698...</td>\n",
       "      <td>m54340U_200716_001732/45089815/ccs,m54340U_200...</td>\n",
       "      <td>.;.;.;10185;.;.;10215,10215;.;10215;10185,1018...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;10215;10185,10185,10185;.;.;10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>13044401</td>\n",
       "      <td>13045071</td>\n",
       "      <td>FAM9C_5UTR</td>\n",
       "      <td>.;.;.;46|53|92|91|65|93|98|99|100|100|100|100|...</td>\n",
       "      <td>.;.;.;74|88|95|96|96|97|94|98|98|93|95|95|83|9...</td>\n",
       "      <td>94|90|88|72|76|79|29|41|98|98|100|100|99|100|1...</td>\n",
       "      <td>.;.;.;5,5;.;.;1;.;.;-1,3,1;.;.;1;.;.;.;.;.;.;....</td>\n",
       "      <td>.;.;.;5,1;.;.;.;.;-2;2,-1,.;.;.;2,8;.;.;.;.;2,...</td>\n",
       "      <td>3,2,0,3;2,2,12,.;.;.;-3;0,2,2,1,0;0,-2,4,-1,0;...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/162006345/ccs,m543...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/13370548/ccs,m5434...</td>\n",
       "      <td>m54340U_200716_001732/18612485/ccs,m54340U_200...</td>\n",
       "      <td>.;.;.;10185,10185;.;.;10215;.;.;10185,10185,10...</td>\n",
       "      <td>.;.;.;10185,10185;.;.;.;.;10215;10185,10185,10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrX</td>\n",
       "      <td>13816674</td>\n",
       "      <td>13817842</td>\n",
       "      <td>GPM6B_5UTR</td>\n",
       "      <td>.;.;.;21|34|70|55|72|63|57|59|24|7|15|13|9|7|8...</td>\n",
       "      <td>.;.;.;26|35|9|0|0|0|0|2|2|1|2|12|20|16|2|0|1|4...</td>\n",
       "      <td>100|99|100|100|99|99|100|99|99|99|99|99|100|99...</td>\n",
       "      <td>.;.;.;-1,3,3,5,.;.;.;.;.;1;-3,7,.;-3,0,-2;.;.,...</td>\n",
       "      <td>.;.;.;9,-4;.;-2,1;2;4;6,-5;-5,.,.,0,1,12,-2,0;...</td>\n",
       "      <td>0,2;-7,2,4,-3,.;1,7;.;.;.;.;2,.,-3;.;.;.;-6,.,...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/29294712/ccs,m5434...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/110494252/ccs,m543...</td>\n",
       "      <td>m54340U_200716_001732/148963558/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;10185,10185,10185,10185,10185;.;.;.;.;10...</td>\n",
       "      <td>.;.;.;10185,10185;.;10215,10215;10215;10215;10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154457737</td>\n",
       "      <td>154458815</td>\n",
       "      <td>PLXNA3_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;11|31|14|14|55|30|36|4|3|2...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;4|0|0|0|0|0|0|0|0|0|5|20|20|...</td>\n",
       "      <td>0|0|0|0|18|6|1|0|2|5|1|0|0|0|0|0|0|0|0|0|0|0|0...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;6,10,0;.;-1;.;.;.;.;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;6,4,1;5,16;.;7,0;.;.;.;.;.;....</td>\n",
       "      <td>5;-1,1;0;16;.;-4;0,1;0,.;2,-3;-1,.,0;.;2;.;.,3...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;m54340U_210709_004131/5721...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;m54340U_200611_224317/482370...</td>\n",
       "      <td>m54340U_200716_001732/6359572/ccs;m54340U_2105...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;.;10185,10185,10185;.;10185;...</td>\n",
       "      <td>.;.;.;.;.;.;.;.;.;10185,10185,10185;10185,1018...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154515851</td>\n",
       "      <td>154516567</td>\n",
       "      <td>FAM3A_5UTR</td>\n",
       "      <td>.;.;0|0|0|1|1|8|9|14|3|1|2|1|0|0|0|2|2|2|5|7|1...</td>\n",
       "      <td>.;.;.;.;.;.;6|0|3|16|15|23|13|11|36|68|27|9|19...</td>\n",
       "      <td>0|2|4|6|11|30|7|8|11|5|1|0|0|0|0|0|0|1|5|1|0|0...</td>\n",
       "      <td>.;.;6;6;.;.;-1,2;.;.;.;.;.;.;.;5,-1;.;.;-1;.;....</td>\n",
       "      <td>.;.;.;.;.;.;2,1;.;0,-1;.;.;.;.;.;.;.;.;.;.;.;....</td>\n",
       "      <td>0,3;0;.;.;.;0,12;.;-1,.,0,-1;.;4,5,18;0,3,.;5;...</td>\n",
       "      <td>.;.;m54340U_200716_153006/163972789/ccs;m54340...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_200806_223416/126420567/cc...</td>\n",
       "      <td>m54340U_200716_001732/108201510/ccs,m54340U_20...</td>\n",
       "      <td>.;.;10185;10185;.;.;10215,10215;.;.;.;.;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215;.;10215,10215;.;.;.;.;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154750529</td>\n",
       "      <td>154751289</td>\n",
       "      <td>GAB3_5UTR</td>\n",
       "      <td>.;.;.;.;.;.;76|99|96|94|98|99|85|79|35|26|9|1|...</td>\n",
       "      <td>.;.;.;0|2|5|45|88|93|81|91|92|90|44|32|94|99|9...</td>\n",
       "      <td>0|0|0|2|8|21|5|2|1|1|0|0|0|0|0|0|0|0|0|0|0|0|0...</td>\n",
       "      <td>.;.;.;.;.;.;5,-1;.;5;1,2,6,3,13,.;2;.;8,1;.;2,...</td>\n",
       "      <td>.;.;.;10,9;.;.;.;.;0;.;.;4,-1,1,2;.;2,0;.;.;.;...</td>\n",
       "      <td>1,-2,1,.;1,0;.;.;.;-5,3,2;13,2;.;.;1;3;.;2;.;....</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/70124537/ccs...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/98632169/ccs,m5434...</td>\n",
       "      <td>m54340U_200716_001732/121375499/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215;.;10215;10185,10185,10...</td>\n",
       "      <td>.;.;.;10185,10185;.;.;.;.;10215;.;.;10185,1018...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154762543</td>\n",
       "      <td>154763011</td>\n",
       "      <td>DKC1_5UTR</td>\n",
       "      <td>.;.;.;100|100|100|100|100|96|73|93|82|86|90|96...</td>\n",
       "      <td>.;.;1|4|7|29|38|7|20|14|31|11|2|1|77|18|3|0|0|...</td>\n",
       "      <td>3|4|46|86|52|73|74|65|47|17|14|7|4|2|5|7|11|16...</td>\n",
       "      <td>.;.;.;3,0;.;.;-1,1;.;12,2;0,4,6,-1,1,3,0,0,0,-...</td>\n",
       "      <td>.;.;0;5,6;.;.;.;.;0,-1;.;.;4,-1;.;1;0,4;.;.;.;...</td>\n",
       "      <td>1,-1;3,0;0,3;2;.;-1,.,1;.;.;.;.;0;0;2,9,11,2,0...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/161745075/ccs,m543...</td>\n",
       "      <td>.;.;m54340U_200716_153006/21431663/ccs;m54340U...</td>\n",
       "      <td>m54340U_200716_001732/126421125/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;10185,10185;.;.;10215,10215;.;10215,1021...</td>\n",
       "      <td>.;.;10185;10185,10185;.;.;.;.;10215,10215;.;.;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155263841</td>\n",
       "      <td>155264649</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>.;.;.;32|53|27|30|49|77|22|4|0|1|0|0|0|0|0|1|3...</td>\n",
       "      <td>.;.;.;.;.;.;84|80|38|22|26|54|71|7|21|63|69|56...</td>\n",
       "      <td>5|2|0|0|0|0|0|0|0|0|0|0|0|0|0|1|1|23|21|25|30|...</td>\n",
       "      <td>.;.;.;5;.;.;0,0;.;.;.;.;.;.;.;.;.;.;.;.;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;0,0,2;.;1,3;.;.;.;.;.;.;.;.;.;.;.;...</td>\n",
       "      <td>2,2,3;5,0,1,2,3;10;.;.;-2,0;.;1,4;.;1,0,6,0,.;...</td>\n",
       "      <td>.;.;.;m54340U_200716_153006/155322677/ccs;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;m54340U_210703_193517/105710335/cc...</td>\n",
       "      <td>m54340U_200716_001732/136054324/ccs,m54340U_20...</td>\n",
       "      <td>.;.;.;10185;.;.;10215,10215;.;.;.;.;.;.;.;.;.;...</td>\n",
       "      <td>.;.;.;.;.;.;10215,10215,10215;.;10215,10215;.;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>163 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom        pos        end     region_id  \\\n",
       "0    chrX   10620123   10620805     MID1_5UTR   \n",
       "1    chrX   11663929   11665723  ARHGAP6_5UTR   \n",
       "2    chrX   12791015   12791585    PRPS2_5UTR   \n",
       "3    chrX   13044401   13045071    FAM9C_5UTR   \n",
       "4    chrX   13816674   13817842    GPM6B_5UTR   \n",
       "..    ...        ...        ...           ...   \n",
       "158  chrX  154457737  154458815   PLXNA3_5UTR   \n",
       "159  chrX  154515851  154516567    FAM3A_5UTR   \n",
       "160  chrX  154750529  154751289     GAB3_5UTR   \n",
       "161  chrX  154762543  154763011     DKC1_5UTR   \n",
       "162  chrX  155263841  155264649   RAB39B_5UTR   \n",
       "\n",
       "                                             methyl_h1  \\\n",
       "0    .;.;.;.;.;.;.;.;.;2|2|34|21|9|38|52|66|56|18|1...   \n",
       "1    .;.;.;12|33|72|92|98|99|98|99|96|93|95|99|99|9...   \n",
       "2    .;.;.;96|82|42|36|92|94|91|74|94|99|99|97|97|9...   \n",
       "3    .;.;.;46|53|92|91|65|93|98|99|100|100|100|100|...   \n",
       "4    .;.;.;21|34|70|55|72|63|57|59|24|7|15|13|9|7|8...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;.;11|31|14|14|55|30|36|4|3|2...   \n",
       "159  .;.;0|0|0|1|1|8|9|14|3|1|2|1|0|0|0|2|2|2|5|7|1...   \n",
       "160  .;.;.;.;.;.;76|99|96|94|98|99|85|79|35|26|9|1|...   \n",
       "161  .;.;.;100|100|100|100|100|96|73|93|82|86|90|96...   \n",
       "162  .;.;.;32|53|27|30|49|77|22|4|0|1|0|0|0|0|0|1|3...   \n",
       "\n",
       "                                             methyl_h2  \\\n",
       "0    .;.;.;.;.;.;.;.;.;22|55|23|34|52|71|37|63|36|3...   \n",
       "1    .;.;.;16|14|19|79|98|80|90|52|57|28|4|5|4|9|24...   \n",
       "2    .;.;.;.;.;.;.;.;1|0|0|0|0|0|0|0|0|7|15|19|8|4|...   \n",
       "3    .;.;.;74|88|95|96|96|97|94|98|98|93|95|95|83|9...   \n",
       "4    .;.;.;26|35|9|0|0|0|0|2|2|1|2|12|20|16|2|0|1|4...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;4|0|0|0|0|0|0|0|0|0|5|20|20|...   \n",
       "159  .;.;.;.;.;.;6|0|3|16|15|23|13|11|36|68|27|9|19...   \n",
       "160  .;.;.;0|2|5|45|88|93|81|91|92|90|44|32|94|99|9...   \n",
       "161  .;.;1|4|7|29|38|7|20|14|31|11|2|1|77|18|3|0|0|...   \n",
       "162  .;.;.;.;.;.;84|80|38|22|26|54|71|7|21|63|69|56...   \n",
       "\n",
       "                                             methyl_h0  \\\n",
       "0    2|2|11|4|0|2|33|10|1|0|1|1|1|0|1|4|11|3|5|1|0|...   \n",
       "1    99|98|96|94|88|27|58|48|25|20|5|5|2|1|0|0|1|3|...   \n",
       "2    0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2|8|5|7|2|...   \n",
       "3    94|90|88|72|76|79|29|41|98|98|100|100|99|100|1...   \n",
       "4    100|99|100|100|99|99|100|99|99|99|99|99|100|99...   \n",
       "..                                                 ...   \n",
       "158  0|0|0|0|18|6|1|0|2|5|1|0|0|0|0|0|0|0|0|0|0|0|0...   \n",
       "159  0|2|4|6|11|30|7|8|11|5|1|0|0|0|0|0|0|1|5|1|0|0...   \n",
       "160  0|0|0|2|8|21|5|2|1|1|0|0|0|0|0|0|0|0|0|0|0|0|0...   \n",
       "161  3|4|46|86|52|73|74|65|47|17|14|7|4|2|5|7|11|16...   \n",
       "162  5|2|0|0|0|0|0|0|0|0|0|0|0|0|0|1|1|23|21|25|30|...   \n",
       "\n",
       "                                             bp_dev_h1  \\\n",
       "0    .;.;.;.;.;.;.;.;.;0;.;.;.;.;.;.;.;-3,.;.;.;.;....   \n",
       "1    .;.;.;3;.;.;.;.;0;0;.,0,0,24,0,.;.;.;.;-2;.;.;...   \n",
       "2    .;.;.;1;.;.;-1,2;.;-1;0,0;-5;.;7,5;.;8,0;.;.;....   \n",
       "3    .;.;.;5,5;.;.;1;.;.;-1,3,1;.;.;1;.;.;.;.;.;.;....   \n",
       "4    .;.;.;-1,3,3,5,.;.;.;.;.;1;-3,7,.;-3,0,-2;.;.,...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;.;6,10,0;.;-1;.;.;.;.;.;.;.;...   \n",
       "159  .;.;6;6;.;.;-1,2;.;.;.;.;.;.;.;5,-1;.;.;-1;.;....   \n",
       "160  .;.;.;.;.;.;5,-1;.;5;1,2,6,3,13,.;2;.;8,1;.;2,...   \n",
       "161  .;.;.;3,0;.;.;-1,1;.;12,2;0,4,6,-1,1,3,0,0,0,-...   \n",
       "162  .;.;.;5;.;.;0,0;.;.;.;.;.;.;.;.;.;.;.;.;.;.;.;...   \n",
       "\n",
       "                                             bp_dev_h2  \\\n",
       "0    .;.;.;.;.;.;.;.;.;3;.;.;1;.;.;.;.;-6,1;.;.;.;....   \n",
       "1    .;.;.;8;.;.;.;.;.;0,25;16,17,1,0,.;.;.;.;2,7,6...   \n",
       "2    .;.;.;.;.;.;.;.;-3;0,-2,0;.;.;0,0,-3,6,2,-1;.;...   \n",
       "3    .;.;.;5,1;.;.;.;.;-2;2,-1,.;.;.;2,8;.;.;.;.;2,...   \n",
       "4    .;.;.;9,-4;.;-2,1;2;4;6,-5;-5,.,.,0,1,12,-2,0;...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;6,4,1;5,16;.;7,0;.;.;.;.;.;....   \n",
       "159  .;.;.;.;.;.;2,1;.;0,-1;.;.;.;.;.;.;.;.;.;.;.;....   \n",
       "160  .;.;.;10,9;.;.;.;.;0;.;.;4,-1,1,2;.;2,0;.;.;.;...   \n",
       "161  .;.;0;5,6;.;.;.;.;0,-1;.;.;4,-1;.;1;0,4;.;.;.;...   \n",
       "162  .;.;.;.;.;.;0,0,2;.;1,3;.;.;.;.;.;.;.;.;.;.;.;...   \n",
       "\n",
       "                                             bp_dev_h0  \\\n",
       "0    0,.;-1,4;1;8,2,3,0;.;8;4,1,0,4;4,0;-4,0;15;-1,...   \n",
       "1    20,15;2,5,7,.;.;.;19,-6;.;.,1,14,-2,1;.,0,2;-1...   \n",
       "2    1,-2,1;0,0,-2,2,5;.;.;5;3,1,1;.;0,1,1,10,.;.;....   \n",
       "3    3,2,0,3;2,2,12,.;.;.;-3;0,2,2,1,0;0,-2,4,-1,0;...   \n",
       "4    0,2;-7,2,4,-3,.;1,7;.;.;.;.;2,.,-3;.;.;.;-6,.,...   \n",
       "..                                                 ...   \n",
       "158  5;-1,1;0;16;.;-4;0,1;0,.;2,-3;-1,.,0;.;2;.;.,3...   \n",
       "159  0,3;0;.;.;.;0,12;.;-1,.,0,-1;.;4,5,18;0,3,.;5;...   \n",
       "160  1,-2,1,.;1,0;.;.;.;-5,3,2;13,2;.;.;1;3;.;2;.;....   \n",
       "161  1,-1;3,0;0,3;2;.;-1,.,1;.;.;.;.;0;0;2,9,11,2,0...   \n",
       "162  2,2,3;5,0,1,2,3;10;.;.;-2,0;.;1,4;.;1,0,6,0,.;...   \n",
       "\n",
       "                                                 rn_h1  \\\n",
       "0    .;.;.;.;.;.;.;.;.;m54340U_210709_004131/129893...   \n",
       "1    .;.;.;m54340U_200716_153006/176228322/ccs;.;.;...   \n",
       "2    .;.;.;m54340U_200716_153006/120785681/ccs;.;.;...   \n",
       "3    .;.;.;m54340U_200716_153006/162006345/ccs,m543...   \n",
       "4    .;.;.;m54340U_200716_153006/29294712/ccs,m5434...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;.;m54340U_210709_004131/5721...   \n",
       "159  .;.;m54340U_200716_153006/163972789/ccs;m54340...   \n",
       "160  .;.;.;.;.;.;m54340U_210703_193517/70124537/ccs...   \n",
       "161  .;.;.;m54340U_200716_153006/161745075/ccs,m543...   \n",
       "162  .;.;.;m54340U_200716_153006/155322677/ccs;.;.;...   \n",
       "\n",
       "                                                 rn_h2  \\\n",
       "0    .;.;.;.;.;.;.;.;.;m54340U_210709_004131/139856...   \n",
       "1    .;.;.;m54340U_200716_153006/53151007/ccs;.;.;....   \n",
       "2    .;.;.;.;.;.;.;.;m54340U_210703_193517/36570698...   \n",
       "3    .;.;.;m54340U_200716_153006/13370548/ccs,m5434...   \n",
       "4    .;.;.;m54340U_200716_153006/110494252/ccs,m543...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;m54340U_200611_224317/482370...   \n",
       "159  .;.;.;.;.;.;m54340U_200806_223416/126420567/cc...   \n",
       "160  .;.;.;m54340U_200716_153006/98632169/ccs,m5434...   \n",
       "161  .;.;m54340U_200716_153006/21431663/ccs;m54340U...   \n",
       "162  .;.;.;.;.;.;m54340U_210703_193517/105710335/cc...   \n",
       "\n",
       "                                                 rn_h0  \\\n",
       "0    m54340U_200716_001732/101845162/ccs,m54340U_20...   \n",
       "1    m54340U_200716_001732/98502494/ccs,m54340U_200...   \n",
       "2    m54340U_200716_001732/45089815/ccs,m54340U_200...   \n",
       "3    m54340U_200716_001732/18612485/ccs,m54340U_200...   \n",
       "4    m54340U_200716_001732/148963558/ccs,m54340U_20...   \n",
       "..                                                 ...   \n",
       "158  m54340U_200716_001732/6359572/ccs;m54340U_2105...   \n",
       "159  m54340U_200716_001732/108201510/ccs,m54340U_20...   \n",
       "160  m54340U_200716_001732/121375499/ccs,m54340U_20...   \n",
       "161  m54340U_200716_001732/126421125/ccs,m54340U_20...   \n",
       "162  m54340U_200716_001732/136054324/ccs,m54340U_20...   \n",
       "\n",
       "                                                 ps_h1  \\\n",
       "0    .;.;.;.;.;.;.;.;.;10185;.;.;.;.;.;.;.;10185,10...   \n",
       "1    .;.;.;10185;.;.;.;.;10215;10185;10185,10185,10...   \n",
       "2    .;.;.;10185;.;.;10215,10215;.;10215;10185,1018...   \n",
       "3    .;.;.;10185,10185;.;.;10215;.;.;10185,10185,10...   \n",
       "4    .;.;.;10185,10185,10185,10185,10185;.;.;.;.;10...   \n",
       "..                                                 ...   \n",
       "158  .;.;.;.;.;.;.;.;.;.;10185,10185,10185;.;10185;...   \n",
       "159  .;.;10185;10185;.;.;10215,10215;.;.;.;.;.;.;.;...   \n",
       "160  .;.;.;.;.;.;10215,10215;.;10215;10185,10185,10...   \n",
       "161  .;.;.;10185,10185;.;.;10215,10215;.;10215,1021...   \n",
       "162  .;.;.;10185;.;.;10215,10215;.;.;.;.;.;.;.;.;.;...   \n",
       "\n",
       "                                                 ps_h2  \n",
       "0    .;.;.;.;.;.;.;.;.;10185;.;.;10185;.;.;.;.;1018...  \n",
       "1    .;.;.;10185;.;.;.;.;.;10185,10185;10185,10185,...  \n",
       "2    .;.;.;.;.;.;.;.;10215;10185,10185,10185;.;.;10...  \n",
       "3    .;.;.;10185,10185;.;.;.;.;10215;10185,10185,10...  \n",
       "4    .;.;.;10185,10185;.;10215,10215;10215;10215;10...  \n",
       "..                                                 ...  \n",
       "158  .;.;.;.;.;.;.;.;.;10185,10185,10185;10185,1018...  \n",
       "159  .;.;.;.;.;.;10215,10215;.;10215,10215;.;.;.;.;...  \n",
       "160  .;.;.;10185,10185;.;.;.;.;10215;.;.;10185,1018...  \n",
       "161  .;.;10185;10185,10185;.;.;.;.;10215,10215;.;.;...  \n",
       "162  .;.;.;.;.;.;10215,10215,10215;.;10215,10215;.;...  \n",
       "\n",
       "[163 rows x 15 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of samples: 270\n",
      "df_exnd_og:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>region_id</th>\n",
       "      <th>REACH000026_methyl_h1</th>\n",
       "      <th>...</th>\n",
       "      <th>REACH000685_rn_h1</th>\n",
       "      <th>REACH000685_rn_h2</th>\n",
       "      <th>REACH000685_rn_h0</th>\n",
       "      <th>REACH000685_ps_h1</th>\n",
       "      <th>REACH000685_ps_h2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrX</td>\n",
       "      <td>10620123</td>\n",
       "      <td>10620805</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>d8c3a034-713f-45e7-afc3-5bd725037e60,4ed0dc63-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrX</td>\n",
       "      <td>11663929</td>\n",
       "      <td>11665723</td>\n",
       "      <td>ARHGAP6_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>48fde9cd-8d16-4466-968f-bc2702649e69,72cf700b-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrX</td>\n",
       "      <td>12791015</td>\n",
       "      <td>12791585</td>\n",
       "      <td>PRPS2_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7ac125ba-e89e-4916-b232-f2de4a0176b2,74c8650b-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>13044401</td>\n",
       "      <td>13045071</td>\n",
       "      <td>FAM9C_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7b324b3e-8553-4a6b-a86f-218833ae9489,d050936f-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrX</td>\n",
       "      <td>13816674</td>\n",
       "      <td>13817842</td>\n",
       "      <td>GPM6B_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>e8d3ece7-d441-4bee-99fa-322791acda4d,441af2cd-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154457737</td>\n",
       "      <td>154458815</td>\n",
       "      <td>PLXNA3_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>206d001e-a23b-4bc8-b79f-7dfbe1ead309,cb5fa621-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154515851</td>\n",
       "      <td>154516567</td>\n",
       "      <td>FAM3A_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>efb7f9d6-7497-4cf0-ac98-7c3012616b7a,b03822db-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154750529</td>\n",
       "      <td>154751289</td>\n",
       "      <td>GAB3_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>129c6e04-18e0-4f22-af8e-c5ce9d3cfd01,37594a25-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154762543</td>\n",
       "      <td>154763011</td>\n",
       "      <td>DKC1_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>eb986a78-64c7-4ab1-b42e-566131904deb,44593820-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155263841</td>\n",
       "      <td>155264649</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>dd2af919-096b-4269-becc-8389e3051162,df83db44-...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>163 rows × 2974 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom        pos        end     region_id REACH000026_methyl_h1  ...  \\\n",
       "0    chrX   10620123   10620805     MID1_5UTR                     .  ...   \n",
       "1    chrX   11663929   11665723  ARHGAP6_5UTR                     .  ...   \n",
       "2    chrX   12791015   12791585    PRPS2_5UTR                     .  ...   \n",
       "3    chrX   13044401   13045071    FAM9C_5UTR                     .  ...   \n",
       "4    chrX   13816674   13817842    GPM6B_5UTR                     .  ...   \n",
       "..    ...        ...        ...           ...                   ...  ...   \n",
       "158  chrX  154457737  154458815   PLXNA3_5UTR                     .  ...   \n",
       "159  chrX  154515851  154516567    FAM3A_5UTR                     .  ...   \n",
       "160  chrX  154750529  154751289     GAB3_5UTR                     .  ...   \n",
       "161  chrX  154762543  154763011     DKC1_5UTR                     .  ...   \n",
       "162  chrX  155263841  155264649   RAB39B_5UTR                     .  ...   \n",
       "\n",
       "    REACH000685_rn_h1 REACH000685_rn_h2  \\\n",
       "0                   .                 .   \n",
       "1                   .                 .   \n",
       "2                   .                 .   \n",
       "3                   .                 .   \n",
       "4                   .                 .   \n",
       "..                ...               ...   \n",
       "158                 .                 .   \n",
       "159                 .                 .   \n",
       "160                 .                 .   \n",
       "161                 .                 .   \n",
       "162                 .                 .   \n",
       "\n",
       "                                     REACH000685_rn_h0 REACH000685_ps_h1  \\\n",
       "0    d8c3a034-713f-45e7-afc3-5bd725037e60,4ed0dc63-...                 .   \n",
       "1    48fde9cd-8d16-4466-968f-bc2702649e69,72cf700b-...                 .   \n",
       "2    7ac125ba-e89e-4916-b232-f2de4a0176b2,74c8650b-...                 .   \n",
       "3    7b324b3e-8553-4a6b-a86f-218833ae9489,d050936f-...                 .   \n",
       "4    e8d3ece7-d441-4bee-99fa-322791acda4d,441af2cd-...                 .   \n",
       "..                                                 ...               ...   \n",
       "158  206d001e-a23b-4bc8-b79f-7dfbe1ead309,cb5fa621-...                 .   \n",
       "159  efb7f9d6-7497-4cf0-ac98-7c3012616b7a,b03822db-...                 .   \n",
       "160  129c6e04-18e0-4f22-af8e-c5ce9d3cfd01,37594a25-...                 .   \n",
       "161  eb986a78-64c7-4ab1-b42e-566131904deb,44593820-...                 .   \n",
       "162  dd2af919-096b-4269-becc-8389e3051162,df83db44-...                 .   \n",
       "\n",
       "    REACH000685_ps_h2  \n",
       "0                   .  \n",
       "1                   .  \n",
       "2                   .  \n",
       "3                   .  \n",
       "4                   .  \n",
       "..                ...  \n",
       "158                 .  \n",
       "159                 .  \n",
       "160                 .  \n",
       "161                 .  \n",
       "162                 .  \n",
       "\n",
       "[163 rows x 2974 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### read select genes methylation for subjects in the skew plot\n",
    "file_in_og = f'{dir_data}/methyl_data_select_genes.tsv'\n",
    "df_og = pd.read_table(file_in_og, sep='\\t', header=0, keep_default_na=False)\n",
    "\n",
    "pd.set_option('display.max_columns', None)\n",
    "print('df_og:')\n",
    "display(df_og)\n",
    "pd.set_option('display.max_columns', 10)\n",
    "\n",
    "file_sam = f'{dir_data}/methyl_data_sample_order_select_genes.txt'\n",
    "with open(file_sam, 'r') as fh:\n",
    "    sample_order_og = fh.readline().split(',')[:-1] # last one is an empty string\n",
    "print(f'number of samples: {len(sample_order_og)}')\n",
    "#print(f'sample order: {sample_order_og}')\n",
    "\n",
    "df_exnd_og = df_og[['chrom' ,'pos', 'end', 'region_id']].copy()\n",
    "cols = df_og.columns.tolist()[4:]\n",
    "#print(cols)\n",
    "\n",
    "for idx, sample in enumerate(sample_order_og):\n",
    "    for col in cols:\n",
    "        df_exnd_og[f'{sample}_{col}'] = df_og[col].str.split(';').apply(lambda row: row[idx])\n",
    "print('df_exnd_og:')\n",
    "display(df_exnd_og)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8e14f45c-e5ce-4765-a6f6-f5920299a775",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_methyl_data(region_id, df_exnd, sample_order, samples_only=None):\n",
    "    df_out = pd.DataFrame()\n",
    "    \n",
    "    this_idx = df_exnd.loc[df_exnd.region_id==region_id].index.tolist()[0]\n",
    "    cols_met = [f'methyl_h1', f'methyl_h2', f'methyl_h0']\n",
    "    cols_ps = [f'ps_h1', f'ps_h2']\n",
    "    for col in cols_met+cols_ps:\n",
    "        df_out[col] = pd.Series(dtype='str')\n",
    "\n",
    "    if samples_only == None:\n",
    "        for sample in sample_order:        \n",
    "            ### set methylation columns\n",
    "            for col in cols_met:\n",
    "                df_out.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "            ### set phase set columns\n",
    "            for col in cols_ps:\n",
    "                df_out.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "    else:\n",
    "        for sample in sample_order:\n",
    "            if sample in samples_only:\n",
    "                ### set methylation columns\n",
    "                for col in cols_met:\n",
    "                    df_out.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "                ### set phase set columns\n",
    "                for col in cols_ps:\n",
    "                    df_out.loc[sample, col] = df_exnd.loc[df_exnd.region_id==region_id].loc[this_idx, f'{sample}_{col}']\n",
    "    \n",
    "    ### compute mean methylation likelihood for each read\n",
    "    for col in cols_met:\n",
    "        df_out[f'read_mean_methyl_lik_{col.split(\"_\")[-1]}'] = df_out.apply(lambda row: ','.join([str(round(np.mean([float(xx) for xx in x.split('|')]))) for x in row[col].split(',') if x != '.'])\n",
    "                                                                                 if row[col] != '.' else '.', axis=1)\n",
    "    for col in cols_met:\n",
    "        hp = col.split(\"_\")[-1]\n",
    "        df_out[f'num_methyl_reads_{hp}'] = df_out[f'read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)>=50]) if x != '.' else '.')\n",
    "        df_out[f'num_unmethyl_reads_{hp}'] = df_out[f'read_mean_methyl_lik_{hp}'].apply(lambda x: len([xx for xx in x.split(',') if float(xx)<50]) if x != '.' else '.')\n",
    "        df_out[f'frac_hap_methyl_{hp}'] = df_out.apply(lambda row: int(row[f'num_methyl_reads_{hp}'])/(int(row[f'num_methyl_reads_{hp}'])+int(row[f'num_unmethyl_reads_{hp}'])) \n",
    "                                                               if (row[f'num_methyl_reads_{hp}']!='.' and row[f'num_unmethyl_reads_{hp}']!='.') else '.', axis=1)\n",
    "    \n",
    "    return df_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "5a3e12e8-419a-4058-99f1-4fde21f8b1bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 ARHGAP6_5UTR\n",
      "11 PRDX4_5UTR\n",
      "21 TAB3_5UTR\n",
      "31 CASK_5UTR\n",
      "41 RBM3_5UTR\n",
      "51 PHF8_5UTR\n",
      "61 EDA_5UTR\n",
      "71 SLC16A2_5UTR\n",
      "81 HNRNPH2_5UTR\n",
      "91 BEX2_5UTR\n",
      "101 PSMD10_5UTR\n",
      "111 NKAP_5UTR\n",
      "121 UTP14A_5UTR\n",
      "131 RTL8B_5UTR\n",
      "141 FMR1_5UTR\n",
      "151 ZNF185_5UTR\n",
      "161 DKC1_5UTR\n",
      "df_methyl_sample:\n",
      "(3749, 21)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>ps_h1</th>\n",
       "      <th>ps_h2</th>\n",
       "      <th>read_mean_methyl_lik_h1</th>\n",
       "      <th>read_mean_methyl_lik_h2</th>\n",
       "      <th>read_mean_methyl_lik_h0</th>\n",
       "      <th>num_methyl_reads_h1</th>\n",
       "      <th>num_unmethyl_reads_h1</th>\n",
       "      <th>frac_hap_methyl_h1</th>\n",
       "      <th>num_methyl_reads_h2</th>\n",
       "      <th>num_unmethyl_reads_h2</th>\n",
       "      <th>frac_hap_methyl_h2</th>\n",
       "      <th>num_methyl_reads_h0</th>\n",
       "      <th>num_unmethyl_reads_h0</th>\n",
       "      <th>frac_hap_methyl_h0</th>\n",
       "      <th>region_id</th>\n",
       "      <th>num_reads_h1</th>\n",
       "      <th>num_reads_h2</th>\n",
       "      <th>Read count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000220</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>10|12|16|13|4|5|2|4|1|0|0|0|0|0|0|0|1|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2,44,2,22,51,1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000234</th>\n",
       "      <td>77|81|91|72|65|59|19|2|0|1|2|0|2|21|28|53|51|7...</td>\n",
       "      <td>0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...</td>\n",
       "      <td>78|99|99|100|99|97|98|95|26|5|1|10|12|5|29|26|...</td>\n",
       "      <td>10185,10185</td>\n",
       "      <td>10185,10185,10185</td>\n",
       "      <td>32,2</td>\n",
       "      <td>0,51,9</td>\n",
       "      <td>52,7,7,26,32,73</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000365</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,72,0,73,1,0,2,11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000436_PB</th>\n",
       "      <td>44|93|96|98|97|100|99|98|73|9|2|3|0|11|66|68|7...</td>\n",
       "      <td>98|99|98|93|96|84|38|4|0|0|0|1|4|9|14|81|95|85...</td>\n",
       "      <td>.</td>\n",
       "      <td>10532334,10532334,10532334,10532334,10532334</td>\n",
       "      <td>10532334,10532334,10532334</td>\n",
       "      <td>39,9,55,2,67</td>\n",
       "      <td>50,0,48</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000451</th>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|9|2|0|0|0|0|0|0|0|0|0|1|1|1|0|0|0|0|...</td>\n",
       "      <td>82|95|98|97|82|82|62|72|33|14|32|56|29|5|3|4|5...</td>\n",
       "      <td>.</td>\n",
       "      <td>10185</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>53,2,43,0,1,53,79,0,89</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000518</th>\n",
       "      <td>0|0|1|0|0|0|13|0|7|11|0|0|0|6|0|0|0|0|0|0|0|4|...</td>\n",
       "      <td>1|10|4|75|1|0|1|0|0|0|1|1|60|0|0|8|2|0|0|0|0|1...</td>\n",
       "      <td>0|100|14|98|98|98|0|100|96|0|100|100|0|5|0|1|1...</td>\n",
       "      <td>10185,10185,10185,10185</td>\n",
       "      <td>10185,10185</td>\n",
       "      <td>3,4,61,13</td>\n",
       "      <td>8,12</td>\n",
       "      <td>32,5,10,7,70,7,17,2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.125</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000519</th>\n",
       "      <td>7|0|3|0|0|0|6|41|0|0|0|3|0|0|0|0|1|0|0|0|0|2|0...</td>\n",
       "      <td>1|68|41|2|1|0|79|82|0|64|4|2|23|0|0|5|0|3|2|0|...</td>\n",
       "      <td>100|53|0|0|0|0|3|14|0|0|2|8|0|40|0|2|1|0|29|0|...</td>\n",
       "      <td>10185</td>\n",
       "      <td>10185</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>8,17,53,45,52,11,2,59,16,11,29,55</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000547</th>\n",
       "      <td>33|0|5|23|59|96|3|98|0|2|30|14|18|1|16|38|11|5...</td>\n",
       "      <td>100|55|92|100|99|45|86|99|99|91|99|100|100|100...</td>\n",
       "      <td>0|0|1|29|8|0|0</td>\n",
       "      <td>12790,12790,12790,12790,12790,12790,12790</td>\n",
       "      <td>12790,12790,12790</td>\n",
       "      <td>16,36,64,47,2,9,75</td>\n",
       "      <td>73,2,22</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000561</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7|65|0|0|0|0|7|0|0|0|0|0|0|0|9|0|0|0|0|1|2|57|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,44,60,80,76,81,58,59,52,47,12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000581</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5|27|100|100|98|100|100|100|84|0|0|0|99|69|0|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>77,0,8,75,68,7,15,6,89,51,6,48,21</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>0.384615</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3749 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        methyl_h1  \\\n",
       "REACH000220                                                     .   \n",
       "REACH000234     77|81|91|72|65|59|19|2|0|1|2|0|2|21|28|53|51|7...   \n",
       "REACH000365                                                     .   \n",
       "REACH000436_PB  44|93|96|98|97|100|99|98|73|9|2|3|0|11|66|68|7...   \n",
       "REACH000451                                                     .   \n",
       "...                                                           ...   \n",
       "REACH000518     0|0|1|0|0|0|13|0|7|11|0|0|0|6|0|0|0|0|0|0|0|4|...   \n",
       "REACH000519     7|0|3|0|0|0|6|41|0|0|0|3|0|0|0|0|1|0|0|0|0|2|0...   \n",
       "REACH000547     33|0|5|23|59|96|3|98|0|2|30|14|18|1|16|38|11|5...   \n",
       "REACH000561                                                     .   \n",
       "REACH000581                                                     .   \n",
       "\n",
       "                                                        methyl_h2  \\\n",
       "REACH000220                                                     .   \n",
       "REACH000234     0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...   \n",
       "REACH000365                                                     .   \n",
       "REACH000436_PB  98|99|98|93|96|84|38|4|0|0|0|1|4|9|14|81|95|85...   \n",
       "REACH000451     0|0|0|0|0|9|2|0|0|0|0|0|0|0|0|0|1|1|1|0|0|0|0|...   \n",
       "...                                                           ...   \n",
       "REACH000518     1|10|4|75|1|0|1|0|0|0|1|1|60|0|0|8|2|0|0|0|0|1...   \n",
       "REACH000519     1|68|41|2|1|0|79|82|0|64|4|2|23|0|0|5|0|3|2|0|...   \n",
       "REACH000547     100|55|92|100|99|45|86|99|99|91|99|100|100|100...   \n",
       "REACH000561                                                     .   \n",
       "REACH000581                                                     .   \n",
       "\n",
       "                                                        methyl_h0  \\\n",
       "REACH000220     10|12|16|13|4|5|2|4|1|0|0|0|0|0|0|0|1|0|0|0|0|...   \n",
       "REACH000234     78|99|99|100|99|97|98|95|26|5|1|10|12|5|29|26|...   \n",
       "REACH000365     0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...   \n",
       "REACH000436_PB                                                  .   \n",
       "REACH000451     82|95|98|97|82|82|62|72|33|14|32|56|29|5|3|4|5...   \n",
       "...                                                           ...   \n",
       "REACH000518     0|100|14|98|98|98|0|100|96|0|100|100|0|5|0|1|1...   \n",
       "REACH000519     100|53|0|0|0|0|3|14|0|0|2|8|0|40|0|2|1|0|29|0|...   \n",
       "REACH000547                                        0|0|1|29|8|0|0   \n",
       "REACH000561     7|65|0|0|0|0|7|0|0|0|0|0|0|0|9|0|0|0|0|1|2|57|...   \n",
       "REACH000581     5|27|100|100|98|100|100|100|84|0|0|0|99|69|0|0...   \n",
       "\n",
       "                                                       ps_h1  \\\n",
       "REACH000220                                                .   \n",
       "REACH000234                                      10185,10185   \n",
       "REACH000365                                                .   \n",
       "REACH000436_PB  10532334,10532334,10532334,10532334,10532334   \n",
       "REACH000451                                                .   \n",
       "...                                                      ...   \n",
       "REACH000518                          10185,10185,10185,10185   \n",
       "REACH000519                                            10185   \n",
       "REACH000547        12790,12790,12790,12790,12790,12790,12790   \n",
       "REACH000561                                                .   \n",
       "REACH000581                                                .   \n",
       "\n",
       "                                     ps_h2 read_mean_methyl_lik_h1  \\\n",
       "REACH000220                              .                       .   \n",
       "REACH000234              10185,10185,10185                    32,2   \n",
       "REACH000365                              .                       .   \n",
       "REACH000436_PB  10532334,10532334,10532334            39,9,55,2,67   \n",
       "REACH000451                          10185                       .   \n",
       "...                                    ...                     ...   \n",
       "REACH000518                    10185,10185               3,4,61,13   \n",
       "REACH000519                          10185                       2   \n",
       "REACH000547              12790,12790,12790      16,36,64,47,2,9,75   \n",
       "REACH000561                              .                       .   \n",
       "REACH000581                              .                       .   \n",
       "\n",
       "               read_mean_methyl_lik_h2            read_mean_methyl_lik_h0  \\\n",
       "REACH000220                          .                     2,44,2,22,51,1   \n",
       "REACH000234                     0,51,9                    52,7,7,26,32,73   \n",
       "REACH000365                          .                 0,72,0,73,1,0,2,11   \n",
       "REACH000436_PB                 50,0,48                                  .   \n",
       "REACH000451                          0             53,2,43,0,1,53,79,0,89   \n",
       "...                                ...                                ...   \n",
       "REACH000518                       8,12                32,5,10,7,70,7,17,2   \n",
       "REACH000519                         17  8,17,53,45,52,11,2,59,16,11,29,55   \n",
       "REACH000547                    73,2,22                                  5   \n",
       "REACH000561                          .    8,44,60,80,76,81,58,59,52,47,12   \n",
       "REACH000581                          .  77,0,8,75,68,7,15,6,89,51,6,48,21   \n",
       "\n",
       "               num_methyl_reads_h1 num_unmethyl_reads_h1 frac_hap_methyl_h1  \\\n",
       "REACH000220                      .                     .                  .   \n",
       "REACH000234                      0                     2                0.0   \n",
       "REACH000365                      .                     .                  .   \n",
       "REACH000436_PB                   2                     3                0.4   \n",
       "REACH000451                      .                     .                  .   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      1                     3               0.25   \n",
       "REACH000519                      0                     1                0.0   \n",
       "REACH000547                      2                     5           0.285714   \n",
       "REACH000561                      .                     .                  .   \n",
       "REACH000581                      .                     .                  .   \n",
       "\n",
       "               num_methyl_reads_h2 num_unmethyl_reads_h2 frac_hap_methyl_h2  \\\n",
       "REACH000220                      .                     .                  .   \n",
       "REACH000234                      1                     2           0.333333   \n",
       "REACH000365                      .                     .                  .   \n",
       "REACH000436_PB                   1                     2           0.333333   \n",
       "REACH000451                      0                     1                0.0   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      0                     2                0.0   \n",
       "REACH000519                      0                     1                0.0   \n",
       "REACH000547                      1                     2           0.333333   \n",
       "REACH000561                      .                     .                  .   \n",
       "REACH000581                      .                     .                  .   \n",
       "\n",
       "               num_methyl_reads_h0 num_unmethyl_reads_h0 frac_hap_methyl_h0  \\\n",
       "REACH000220                      1                     5           0.166667   \n",
       "REACH000234                      2                     4           0.333333   \n",
       "REACH000365                      2                     6               0.25   \n",
       "REACH000436_PB                   .                     .                  .   \n",
       "REACH000451                      4                     5           0.444444   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      1                     7              0.125   \n",
       "REACH000519                      4                     8           0.333333   \n",
       "REACH000547                      0                     1                0.0   \n",
       "REACH000561                      7                     4           0.636364   \n",
       "REACH000581                      5                     8           0.384615   \n",
       "\n",
       "                  region_id  num_reads_h1  num_reads_h2  Read count  \n",
       "REACH000220       MID1_5UTR             0             0           0  \n",
       "REACH000234       MID1_5UTR             2             3           5  \n",
       "REACH000365       MID1_5UTR             0             0           0  \n",
       "REACH000436_PB    MID1_5UTR             5             3           8  \n",
       "REACH000451       MID1_5UTR             0             1           1  \n",
       "...                     ...           ...           ...         ...  \n",
       "REACH000518     RAB39B_5UTR             4             2           6  \n",
       "REACH000519     RAB39B_5UTR             1             1           2  \n",
       "REACH000547     RAB39B_5UTR             7             3          10  \n",
       "REACH000561     RAB39B_5UTR             0             0           0  \n",
       "REACH000581     RAB39B_5UTR             0             0           0  \n",
       "\n",
       "[3749 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Make methylation data table of all regions for selected subjects\n",
    "samples_only = ['REACH000561', 'REACH000453', 'REACH000451', 'REACH000479', 'REACH000518', 'REACH000519', \n",
    "                'REACH000234', 'REACH000365', 'REACH000436_PB', 'REACH000616', 'REACH000628', 'REACH000672', 'REACH000674', \n",
    "                'REACH000702', 'REACH000107', 'REACH000404', 'REACH000417', 'REACH000435', 'REACH000482', 'REACH000510', \n",
    "                'REACH000547', 'REACH000581', 'REACH000220']\n",
    "df_methyl_sample = pd.DataFrame()\n",
    "### this loop may take some time to run...\n",
    "for i, region_id in enumerate(df_exnd_og.region_id.tolist()):\n",
    "    if (i % 10 == 1):\n",
    "        print(f'{i} {region_id}')\n",
    "    this_df = get_gene_methyl_data(region_id, df_exnd_og, sample_order_og, samples_only)\n",
    "    this_df['region_id'] = region_id\n",
    "    df_methyl_sample = pd.concat([df_methyl_sample, this_df])\n",
    "\n",
    "### correct FMR1 methylation data for REACH000561 from before corrections\n",
    "cols_corr = ['read_mean_methyl_lik_h1', 'read_mean_methyl_lik_h2', 'read_mean_methyl_lik_h0', \n",
    "             'num_methyl_reads_h1', 'num_methyl_reads_h2', 'num_methyl_reads_h0', \n",
    "             'num_unmethyl_reads_h1', 'num_unmethyl_reads_h2', 'num_unmethyl_reads_h0', \n",
    "             'frac_hap_methyl_h1', 'frac_hap_methyl_h2', 'frac_hap_methyl_h0']\n",
    "as_type = [str, str, str, int, int, str, int, int, str, float, float, str]\n",
    "for col, t in zip(cols_corr, as_type):\n",
    "    df_methyl_sample.loc[((df_methyl_sample.region_id=='FMR1_5UTR') & \n",
    "                          (df_methyl_sample.index=='REACH000561')), [col]] = df_methyl.loc[['REACH000561'],[col]].astype(t)\n",
    "\n",
    "df_methyl_sample['num_reads_h1'] = df_methyl_sample.apply(lambda row: int(row.num_methyl_reads_h1) + int(row.num_unmethyl_reads_h1) \n",
    "                                                  if row.num_methyl_reads_h1!='.' else 0, axis=1)\n",
    "df_methyl_sample['num_reads_h2'] = df_methyl_sample.apply(lambda row: int(row.num_methyl_reads_h2) + int(row.num_unmethyl_reads_h2) \n",
    "                                                  if row.num_methyl_reads_h2!='.' else 0, axis=1)\n",
    "df_methyl_sample['Read count'] = df_methyl_sample['num_reads_h1'] + df_methyl_sample['num_reads_h2']\n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "pd.set_option('display.max_columns', None)\n",
    "print('df_methyl_sample:')\n",
    "print(df_methyl_sample.shape)\n",
    "display(df_methyl_sample)\n",
    "pd.set_option('display.max_columns', 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "dfab8bae-f310-4626-82be-64bfcdd35e26",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_methyl_sample:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>methyl_h1</th>\n",
       "      <th>methyl_h2</th>\n",
       "      <th>methyl_h0</th>\n",
       "      <th>ps_h1</th>\n",
       "      <th>ps_h2</th>\n",
       "      <th>read_mean_methyl_lik_h1</th>\n",
       "      <th>read_mean_methyl_lik_h2</th>\n",
       "      <th>read_mean_methyl_lik_h0</th>\n",
       "      <th>num_methyl_reads_h1</th>\n",
       "      <th>num_unmethyl_reads_h1</th>\n",
       "      <th>frac_hap_methyl_h1</th>\n",
       "      <th>num_methyl_reads_h2</th>\n",
       "      <th>num_unmethyl_reads_h2</th>\n",
       "      <th>frac_hap_methyl_h2</th>\n",
       "      <th>num_methyl_reads_h0</th>\n",
       "      <th>num_unmethyl_reads_h0</th>\n",
       "      <th>frac_hap_methyl_h0</th>\n",
       "      <th>region_id</th>\n",
       "      <th>num_reads_h1</th>\n",
       "      <th>num_reads_h2</th>\n",
       "      <th>Read count</th>\n",
       "      <th>fmr1_mean_repeat_length_h1</th>\n",
       "      <th>fmr1_mean_repeat_length_h2</th>\n",
       "      <th>fmr1_mean_repeat_length_h0</th>\n",
       "      <th>frac_hap_methyl_large_rep</th>\n",
       "      <th>frac_hap_methyl_small_rep</th>\n",
       "      <th>fmr1_mean_large_rep</th>\n",
       "      <th>fmr1_mean_small_rep</th>\n",
       "      <th>repeat_group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>REACH000220</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>10|12|16|13|4|5|2|4|1|0|0|0|0|0|0|0|1|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2,44,2,22,51,1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000234</th>\n",
       "      <td>77|81|91|72|65|59|19|2|0|1|2|0|2|21|28|53|51|7...</td>\n",
       "      <td>0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...</td>\n",
       "      <td>78|99|99|100|99|97|98|95|26|5|1|10|12|5|29|26|...</td>\n",
       "      <td>10185,10185</td>\n",
       "      <td>10185,10185,10185</td>\n",
       "      <td>32,2</td>\n",
       "      <td>0,51,9</td>\n",
       "      <td>52,7,7,26,32,73</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>29.333333</td>\n",
       "      <td>30.600000</td>\n",
       "      <td>.</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.600000</td>\n",
       "      <td>29.333333</td>\n",
       "      <td>26-34 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000365</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0,72,0,73,1,0,2,11</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>0.25</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30.166667</td>\n",
       "      <td>29.250000</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>30.166667</td>\n",
       "      <td>29.250000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000436_PB</th>\n",
       "      <td>44|93|96|98|97|100|99|98|73|9|2|3|0|11|66|68|7...</td>\n",
       "      <td>98|99|98|93|96|84|38|4|0|0|0|1|4|9|14|81|95|85...</td>\n",
       "      <td>.</td>\n",
       "      <td>10532334,10532334,10532334,10532334,10532334</td>\n",
       "      <td>10532334,10532334,10532334</td>\n",
       "      <td>39,9,55,2,67</td>\n",
       "      <td>50,0,48</td>\n",
       "      <td>.</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>23.750000</td>\n",
       "      <td>29.833333</td>\n",
       "      <td>.</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.4</td>\n",
       "      <td>29.833333</td>\n",
       "      <td>23.750000</td>\n",
       "      <td>≤25 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000451</th>\n",
       "      <td>.</td>\n",
       "      <td>0|0|0|0|0|9|2|0|0|0|0|0|0|0|0|0|1|1|1|0|0|0|0|...</td>\n",
       "      <td>82|95|98|97|82|82|62|72|33|14|32|56|29|5|3|4|5...</td>\n",
       "      <td>.</td>\n",
       "      <td>10185</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>53,2,43,0,1,53,79,0,89</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>MID1_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>28.800000</td>\n",
       "      <td>49.400000</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>49.400000</td>\n",
       "      <td>28.800000</td>\n",
       "      <td>≥35 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000518</th>\n",
       "      <td>0|0|1|0|0|0|13|0|7|11|0|0|0|6|0|0|0|0|0|0|0|4|...</td>\n",
       "      <td>1|10|4|75|1|0|1|0|0|0|1|1|60|0|0|8|2|0|0|0|0|1...</td>\n",
       "      <td>0|100|14|98|98|98|0|100|96|0|100|100|0|5|0|1|1...</td>\n",
       "      <td>10185,10185,10185,10185</td>\n",
       "      <td>10185,10185</td>\n",
       "      <td>3,4,61,13</td>\n",
       "      <td>8,12</td>\n",
       "      <td>32,5,10,7,70,7,17,2</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0.125</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>35.000000</td>\n",
       "      <td>29.444444</td>\n",
       "      <td>28.5</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.000000</td>\n",
       "      <td>29.444444</td>\n",
       "      <td>≥35 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000519</th>\n",
       "      <td>7|0|3|0|0|0|6|41|0|0|0|3|0|0|0|0|1|0|0|0|0|2|0...</td>\n",
       "      <td>1|68|41|2|1|0|79|82|0|64|4|2|23|0|0|5|0|3|2|0|...</td>\n",
       "      <td>100|53|0|0|0|0|3|14|0|0|2|8|0|40|0|2|1|0|29|0|...</td>\n",
       "      <td>10185</td>\n",
       "      <td>10185</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>8,17,53,45,52,11,2,59,16,11,29,55</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>35.000000</td>\n",
       "      <td>20.833333</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.000000</td>\n",
       "      <td>20.833333</td>\n",
       "      <td>≥35 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000547</th>\n",
       "      <td>33|0|5|23|59|96|3|98|0|2|30|14|18|1|16|38|11|5...</td>\n",
       "      <td>100|55|92|100|99|45|86|99|99|91|99|100|100|100...</td>\n",
       "      <td>0|0|1|29|8|0|0</td>\n",
       "      <td>12790,12790,12790,12790,12790,12790,12790</td>\n",
       "      <td>12790,12790,12790</td>\n",
       "      <td>16,36,64,47,2,9,75</td>\n",
       "      <td>73,2,22</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>33.750000</td>\n",
       "      <td>.</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.285714</td>\n",
       "      <td>33.750000</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>26-34 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000561</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7|65|0|0|0|0|7|0|0|0|0|0|0|0|9|0|0|0|0|1|2|57|...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8,44,60,80,76,81,58,59,52,47,12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>0.636364</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28.250000</td>\n",
       "      <td>49.111111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>49.111111</td>\n",
       "      <td>28.250000</td>\n",
       "      <td>≥35 CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REACH000581</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5|27|100|100|98|100|100|100|84|0|0|0|99|69|0|0...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>77,0,8,75,68,7,15,6,89,51,6,48,21</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>0.384615</td>\n",
       "      <td>RAB39B_5UTR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31.500000</td>\n",
       "      <td>23.750000</td>\n",
       "      <td>28.333333</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>31.500000</td>\n",
       "      <td>23.750000</td>\n",
       "      <td>≤25 CGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3749 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        methyl_h1  \\\n",
       "REACH000220                                                     .   \n",
       "REACH000234     77|81|91|72|65|59|19|2|0|1|2|0|2|21|28|53|51|7...   \n",
       "REACH000365                                                     .   \n",
       "REACH000436_PB  44|93|96|98|97|100|99|98|73|9|2|3|0|11|66|68|7...   \n",
       "REACH000451                                                     .   \n",
       "...                                                           ...   \n",
       "REACH000518     0|0|1|0|0|0|13|0|7|11|0|0|0|6|0|0|0|0|0|0|0|4|...   \n",
       "REACH000519     7|0|3|0|0|0|6|41|0|0|0|3|0|0|0|0|1|0|0|0|0|2|0...   \n",
       "REACH000547     33|0|5|23|59|96|3|98|0|2|30|14|18|1|16|38|11|5...   \n",
       "REACH000561                                                     .   \n",
       "REACH000581                                                     .   \n",
       "\n",
       "                                                        methyl_h2  \\\n",
       "REACH000220                                                     .   \n",
       "REACH000234     0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...   \n",
       "REACH000365                                                     .   \n",
       "REACH000436_PB  98|99|98|93|96|84|38|4|0|0|0|1|4|9|14|81|95|85...   \n",
       "REACH000451     0|0|0|0|0|9|2|0|0|0|0|0|0|0|0|0|1|1|1|0|0|0|0|...   \n",
       "...                                                           ...   \n",
       "REACH000518     1|10|4|75|1|0|1|0|0|0|1|1|60|0|0|8|2|0|0|0|0|1...   \n",
       "REACH000519     1|68|41|2|1|0|79|82|0|64|4|2|23|0|0|5|0|3|2|0|...   \n",
       "REACH000547     100|55|92|100|99|45|86|99|99|91|99|100|100|100...   \n",
       "REACH000561                                                     .   \n",
       "REACH000581                                                     .   \n",
       "\n",
       "                                                        methyl_h0  \\\n",
       "REACH000220     10|12|16|13|4|5|2|4|1|0|0|0|0|0|0|0|1|0|0|0|0|...   \n",
       "REACH000234     78|99|99|100|99|97|98|95|26|5|1|10|12|5|29|26|...   \n",
       "REACH000365     0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|...   \n",
       "REACH000436_PB                                                  .   \n",
       "REACH000451     82|95|98|97|82|82|62|72|33|14|32|56|29|5|3|4|5...   \n",
       "...                                                           ...   \n",
       "REACH000518     0|100|14|98|98|98|0|100|96|0|100|100|0|5|0|1|1...   \n",
       "REACH000519     100|53|0|0|0|0|3|14|0|0|2|8|0|40|0|2|1|0|29|0|...   \n",
       "REACH000547                                        0|0|1|29|8|0|0   \n",
       "REACH000561     7|65|0|0|0|0|7|0|0|0|0|0|0|0|9|0|0|0|0|1|2|57|...   \n",
       "REACH000581     5|27|100|100|98|100|100|100|84|0|0|0|99|69|0|0...   \n",
       "\n",
       "                                                       ps_h1  \\\n",
       "REACH000220                                                .   \n",
       "REACH000234                                      10185,10185   \n",
       "REACH000365                                                .   \n",
       "REACH000436_PB  10532334,10532334,10532334,10532334,10532334   \n",
       "REACH000451                                                .   \n",
       "...                                                      ...   \n",
       "REACH000518                          10185,10185,10185,10185   \n",
       "REACH000519                                            10185   \n",
       "REACH000547        12790,12790,12790,12790,12790,12790,12790   \n",
       "REACH000561                                                .   \n",
       "REACH000581                                                .   \n",
       "\n",
       "                                     ps_h2 read_mean_methyl_lik_h1  \\\n",
       "REACH000220                              .                       .   \n",
       "REACH000234              10185,10185,10185                    32,2   \n",
       "REACH000365                              .                       .   \n",
       "REACH000436_PB  10532334,10532334,10532334            39,9,55,2,67   \n",
       "REACH000451                          10185                       .   \n",
       "...                                    ...                     ...   \n",
       "REACH000518                    10185,10185               3,4,61,13   \n",
       "REACH000519                          10185                       2   \n",
       "REACH000547              12790,12790,12790      16,36,64,47,2,9,75   \n",
       "REACH000561                              .                       .   \n",
       "REACH000581                              .                       .   \n",
       "\n",
       "               read_mean_methyl_lik_h2            read_mean_methyl_lik_h0  \\\n",
       "REACH000220                          .                     2,44,2,22,51,1   \n",
       "REACH000234                     0,51,9                    52,7,7,26,32,73   \n",
       "REACH000365                          .                 0,72,0,73,1,0,2,11   \n",
       "REACH000436_PB                 50,0,48                                  .   \n",
       "REACH000451                          0             53,2,43,0,1,53,79,0,89   \n",
       "...                                ...                                ...   \n",
       "REACH000518                       8,12                32,5,10,7,70,7,17,2   \n",
       "REACH000519                         17  8,17,53,45,52,11,2,59,16,11,29,55   \n",
       "REACH000547                    73,2,22                                  5   \n",
       "REACH000561                          .    8,44,60,80,76,81,58,59,52,47,12   \n",
       "REACH000581                          .  77,0,8,75,68,7,15,6,89,51,6,48,21   \n",
       "\n",
       "               num_methyl_reads_h1 num_unmethyl_reads_h1 frac_hap_methyl_h1  \\\n",
       "REACH000220                      .                     .                  .   \n",
       "REACH000234                      0                     2                0.0   \n",
       "REACH000365                      .                     .                  .   \n",
       "REACH000436_PB                   2                     3                0.4   \n",
       "REACH000451                      .                     .                  .   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      1                     3               0.25   \n",
       "REACH000519                      0                     1                0.0   \n",
       "REACH000547                      2                     5           0.285714   \n",
       "REACH000561                      .                     .                  .   \n",
       "REACH000581                      .                     .                  .   \n",
       "\n",
       "               num_methyl_reads_h2 num_unmethyl_reads_h2 frac_hap_methyl_h2  \\\n",
       "REACH000220                      .                     .                  .   \n",
       "REACH000234                      1                     2           0.333333   \n",
       "REACH000365                      .                     .                  .   \n",
       "REACH000436_PB                   1                     2           0.333333   \n",
       "REACH000451                      0                     1                0.0   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      0                     2                0.0   \n",
       "REACH000519                      0                     1                0.0   \n",
       "REACH000547                      1                     2           0.333333   \n",
       "REACH000561                      .                     .                  .   \n",
       "REACH000581                      .                     .                  .   \n",
       "\n",
       "               num_methyl_reads_h0 num_unmethyl_reads_h0 frac_hap_methyl_h0  \\\n",
       "REACH000220                      1                     5           0.166667   \n",
       "REACH000234                      2                     4           0.333333   \n",
       "REACH000365                      2                     6               0.25   \n",
       "REACH000436_PB                   .                     .                  .   \n",
       "REACH000451                      4                     5           0.444444   \n",
       "...                            ...                   ...                ...   \n",
       "REACH000518                      1                     7              0.125   \n",
       "REACH000519                      4                     8           0.333333   \n",
       "REACH000547                      0                     1                0.0   \n",
       "REACH000561                      7                     4           0.636364   \n",
       "REACH000581                      5                     8           0.384615   \n",
       "\n",
       "                  region_id  num_reads_h1  num_reads_h2  Read count  \\\n",
       "REACH000220       MID1_5UTR             0             0           0   \n",
       "REACH000234       MID1_5UTR             2             3           5   \n",
       "REACH000365       MID1_5UTR             0             0           0   \n",
       "REACH000436_PB    MID1_5UTR             5             3           8   \n",
       "REACH000451       MID1_5UTR             0             1           1   \n",
       "...                     ...           ...           ...         ...   \n",
       "REACH000518     RAB39B_5UTR             4             2           6   \n",
       "REACH000519     RAB39B_5UTR             1             1           2   \n",
       "REACH000547     RAB39B_5UTR             7             3          10   \n",
       "REACH000561     RAB39B_5UTR             0             0           0   \n",
       "REACH000581     RAB39B_5UTR             0             0           0   \n",
       "\n",
       "                fmr1_mean_repeat_length_h1  fmr1_mean_repeat_length_h2  \\\n",
       "REACH000220                      29.000000                   29.000000   \n",
       "REACH000234                      29.333333                   30.600000   \n",
       "REACH000365                      30.166667                   29.250000   \n",
       "REACH000436_PB                   23.750000                   29.833333   \n",
       "REACH000451                      28.800000                   49.400000   \n",
       "...                                    ...                         ...   \n",
       "REACH000518                      35.000000                   29.444444   \n",
       "REACH000519                      35.000000                   20.833333   \n",
       "REACH000547                      28.000000                   33.750000   \n",
       "REACH000561                      28.250000                   49.111111   \n",
       "REACH000581                      31.500000                   23.750000   \n",
       "\n",
       "               fmr1_mean_repeat_length_h0 frac_hap_methyl_large_rep  \\\n",
       "REACH000220                             .                         .   \n",
       "REACH000234                             .                  0.333333   \n",
       "REACH000365                             .                         .   \n",
       "REACH000436_PB                          .                  0.333333   \n",
       "REACH000451                             .                       0.0   \n",
       "...                                   ...                       ...   \n",
       "REACH000518                          28.5                      0.25   \n",
       "REACH000519                          27.0                       0.0   \n",
       "REACH000547                             .                  0.333333   \n",
       "REACH000561                             .                         .   \n",
       "REACH000581                     28.333333                         .   \n",
       "\n",
       "               frac_hap_methyl_small_rep  fmr1_mean_large_rep  \\\n",
       "REACH000220                            .            29.000000   \n",
       "REACH000234                          0.0            30.600000   \n",
       "REACH000365                            .            30.166667   \n",
       "REACH000436_PB                       0.4            29.833333   \n",
       "REACH000451                            .            49.400000   \n",
       "...                                  ...                  ...   \n",
       "REACH000518                          0.0            35.000000   \n",
       "REACH000519                          0.0            35.000000   \n",
       "REACH000547                     0.285714            33.750000   \n",
       "REACH000561                            .            49.111111   \n",
       "REACH000581                            .            31.500000   \n",
       "\n",
       "                fmr1_mean_small_rep repeat_group  \n",
       "REACH000220               29.000000    26-34 CGG  \n",
       "REACH000234               29.333333    26-34 CGG  \n",
       "REACH000365               29.250000    26-34 CGG  \n",
       "REACH000436_PB            23.750000      ≤25 CGG  \n",
       "REACH000451               28.800000      ≥35 CGG  \n",
       "...                             ...          ...  \n",
       "REACH000518               29.444444      ≥35 CGG  \n",
       "REACH000519               20.833333      ≥35 CGG  \n",
       "REACH000547               28.000000    26-34 CGG  \n",
       "REACH000561               28.250000      ≥35 CGG  \n",
       "REACH000581               23.750000      ≤25 CGG  \n",
       "\n",
       "[3749 rows x 29 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputs/methylation_table_select_genes.tsv\n"
     ]
    }
   ],
   "source": [
    "### write methylation data table for the 163 genes for the selected subjects\n",
    "df_methyl_sample['fmr1_mean_repeat_length_h1'] = df_methyl_sample.index.map(df_methyl.mean_repeat_length_h1).astype(float)\n",
    "df_methyl_sample['fmr1_mean_repeat_length_h2'] = df_methyl_sample.index.map(df_methyl.mean_repeat_length_h2).astype(float)\n",
    "df_methyl_sample['fmr1_mean_repeat_length_h0'] = df_methyl_sample.index.map(df_methyl.mean_repeat_length_h0)\n",
    "\n",
    "### for those subjects with both haplotypes present (must be females), compute large and small haplotype stats (frac, mean rep, skewness)\n",
    "### haplotype of col_in_1 must correspond to col_out_1 and the same for _2\n",
    "### output is large: first, small: second\n",
    "def get_large_small_vals(row, col_in_1, col_in_2, col_out_1, col_out_2):\n",
    "    ### no need to check for sex and '.' present in num_methyla_reads\n",
    "    if row[col_in_1] >= row[col_in_2]:\n",
    "        out = (row[col_out_1], row[col_out_2])\n",
    "    else:\n",
    "        out = (row[col_out_2], row[col_out_1])\n",
    "    return out\n",
    "    \n",
    "### figure out which hp is the large and with is the small CGG repeat\n",
    "col_in_1 = 'fmr1_mean_repeat_length_h1'\n",
    "col_in_2 = 'fmr1_mean_repeat_length_h2'\n",
    "col_out_1 = 'frac_hap_methyl_h1'\n",
    "col_out_2 = 'frac_hap_methyl_h2'\n",
    "df_methyl_sample[['frac_hap_methyl_large_rep', 'frac_hap_methyl_small_rep']] = df_methyl_sample.apply(lambda row: get_large_small_vals(row, col_in_1, col_in_2, \n",
    "                                                                                                                         col_out_1, col_out_2), axis=1, result_type = 'expand')\n",
    "\n",
    "col_in_1 = 'fmr1_mean_repeat_length_h1'\n",
    "col_in_2 = 'fmr1_mean_repeat_length_h2'\n",
    "col_out_1 = 'fmr1_mean_repeat_length_h1'\n",
    "col_out_2 = 'fmr1_mean_repeat_length_h2'\n",
    "df_methyl_sample[['fmr1_mean_large_rep', 'fmr1_mean_small_rep']] = df_methyl_sample.apply(lambda row: get_large_small_vals(row, col_in_1, col_in_2, \n",
    "                                                                                                   col_out_1, col_out_2), axis=1, result_type = 'expand')\n",
    "\n",
    "def get_rep_group2(row):\n",
    "    out = ('≥35 CGG' if (row.fmr1_mean_repeat_length_h1 >= 35 or row.fmr1_mean_repeat_length_h2 >= 35) else \n",
    "           ('≤25 CGG' if (row.fmr1_mean_repeat_length_h1 <= 25 or row.fmr1_mean_repeat_length_h2 <= 25) else '26-34 CGG'))\n",
    "    return out\n",
    "df_methyl_sample['repeat_group'] = df_methyl_sample.apply(lambda row: get_rep_group2(row), axis=1)\n",
    "\n",
    "pd.set_option('display.max_columns', None)\n",
    "print('df_methyl_sample:')\n",
    "display(df_methyl_sample)\n",
    "pd.set_option('display.max_columns', 10)\n",
    "\n",
    "### write the table\n",
    "file_name = f'{dir_out}/methylation_table_select_genes.tsv'\n",
    "print(file_name)\n",
    "df_methyl_sample.to_csv(file_name, sep='\\t', index=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
