{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "75ced091-1d6a-4845-93a5-d130cb1343bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=pd.errors.PerformanceWarning)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "91352a99-5e24-4603-973c-a324fb1ce3d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "dir_out = 'outputs/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "21875b94-58eb-449d-8686-5fcde4afda42",
   "metadata": {},
   "outputs": [],
   "source": [
    "def xfeature(row, biotype):\n",
    "\tret = 0\n",
    "\tfor bio, con in zip(row['BIOTYPE'].split(','), row['Consequence'].split(',')):\n",
    "\t\tif bio == biotype and 'upstream_gene_variant' not in con and 'downstream_gene_variant' not in con:\n",
    "\t\t\tret = 1\n",
    "\t\t\tbreak\n",
    "\treturn ret\n",
    "\n",
    "def xintergenic(row):\n",
    "    ret = 0\n",
    "    #cons_set_max = {'intergenic_variant', 'upstream_gene_variant', 'downstream_gene_variant'}\n",
    "    cons_set_max = {'intergenic_variant'}\n",
    "    cons_set = set(row['Consequence'].split(','))\n",
    "    if (len(cons_set - cons_set_max) == 0) and (row['GENCODE'] == '.'):\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def xgenic(row):\n",
    "    ret = 0\n",
    "    pli_all = []\n",
    "    for p, con in zip(row['GENES_PLI'].split(','), row['Consequence'].split(',')):\n",
    "        if (p == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "            continue\n",
    "        pli_all.append(float(p))\n",
    "    if len(pli_all) > 0:\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "### filtering INV calls which do not interupt genes\n",
    "def xpli(row, pli_thr):\n",
    "    ret = 0\n",
    "    lof_inv_genes = row['LOF_DUP_INV_GENES'].split(',')\n",
    "    for p, con, sym in zip(row['GENES_PLI'].split(','), row['Consequence'].split(','), row['SYMBOL'].split(',')):\n",
    "        if (p == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "            continue\n",
    "        if float(p) >= pli_thr and (row['SVTYPE']!='INV' or sym in lof_inv_genes):\n",
    "            ret = 1\n",
    "            break\n",
    "    return ret\n",
    "\n",
    "def xlowpli(row, pli_thr):\n",
    "    ret = 0\n",
    "    pli_all = []\n",
    "    for p, con in zip(row['GENES_PLI'].split(','), row['Consequence'].split(',')):\n",
    "        if (p == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "            continue\n",
    "        pli_all.append(float(p))\n",
    "    if len(pli_all) > 0 and max(pli_all) < pli_thr:\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "### filtering INV calls which do not interupt genes\n",
    "def xloeuf(row, l_thr):\n",
    "    ret = 0\n",
    "    lof_inv_genes = row['LOF_DUP_INV_GENES'].split(',')\n",
    "    for l, con, sym in zip(row['GENES_LOEUF'].split(','), row['Consequence'].split(','), row['SYMBOL'].split(',')):\n",
    "        if (l == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "            continue\n",
    "        if float(l) <= l_thr and (row['SVTYPE']!='INV' or sym in lof_inv_genes):\n",
    "            ret = 1\n",
    "            break\n",
    "    return ret\n",
    "\n",
    "### filtering INV calls which do not interupt genes\n",
    "def xfdr(row, f_type, f_thr):\n",
    "    ret = 0\n",
    "    lof_inv_genes = row['LOF_DUP_INV_GENES'].split(',')\n",
    "    for f, con, sym in zip(row['GENES_FDR_'+f_type].split(','), row['Consequence'].split(','), row['SYMBOL'].split(',')):\n",
    "        if (f == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "            continue\n",
    "        if float(f) <= f_thr and (row['SVTYPE']!='INV' or sym in lof_inv_genes):\n",
    "            ret = 1\n",
    "            break\n",
    "    return ret\n",
    "\n",
    "def xfb_pr_enh(row, fb_set):\n",
    "    ret = 0\n",
    "    if row['FB_PR_ENH_M'] != '.' and len( set(row['FB_PR_ENH_M'].split('&')).intersection(fb_set) ) != 0:\n",
    "        ret = 1\n",
    "    if row['FB_PR_ENH_F'] != '.' and len( set(row['FB_PR_ENH_F'].split('&')).intersection(fb_set) ) != 0:\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def xgnocchi(row, col, gn_thr):\n",
    "    ret = 0\n",
    "    #if (row['GNOCCHI'] != '.') and (float(row['GNOCCHI']) > gn_thr):\n",
    "    if (row[col] != '.') and (float(row[col]) > gn_thr):\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def xfantom(row):\n",
    "    ret = 0\n",
    "    if row['FANTOM_ENH'] == 'FanEnh':\n",
    "        ret = 1\n",
    "    return ret\n",
    "                \n",
    "def status_num(row, cols):\n",
    "    col_case_nums = []\n",
    "    col_control_nums = []\n",
    "    for c in cols:\n",
    "        col = f'case_{c}'\n",
    "        if row[col] == '.' or row[col] == '':\n",
    "            col_case_nums.append(0)\n",
    "            col_control_nums.append(0)\n",
    "            continue\n",
    "        n_case = len([x for x in row[col].split(',') if x == 'Yes'])\n",
    "        n_control = len([x for x in row[col].split(',') if x == 'No'])\n",
    "        col_case_nums.append(n_case)\n",
    "        col_control_nums.append(n_control)\n",
    "    return *col_case_nums, *col_control_nums\n",
    "    \n",
    "def xgencode(row, gen_list, feature):\t\n",
    "\tret = 0\n",
    "\tif row[feature] == 1 and row['GENCODE'] in (gen_list):\n",
    "\t\tret = 1\n",
    "\treturn ret\n",
    "\n",
    "def x2features(row, feature1, feature2):\n",
    "    ret = 0\n",
    "    if row[feature1] == 1 and row[feature2] == 1:\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def count_num(row, col):\n",
    "    ret = 0\n",
    "    if row[col] == '.' or row[col] == '':\n",
    "        return ret\n",
    "    ret = len(row[col].split(','))\n",
    "    return ret\n",
    "\n",
    "def lof(row):\n",
    "    ret = 0\n",
    "    if row['SVTYPE'] in ['DEL', 'INS'] and row['GENCODE'] in ['CDS', 'start_codon', 'stop_codon']:\n",
    "        ret = 1\n",
    "    if row['LOF_DUP_INV'] == 1:\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def xdev_br(row, dev_br_list):\n",
    "\tret = 0\n",
    "\tfor sym, con in zip(row['SYMBOL'].split(','), row['Consequence'].split(',')):\n",
    "\t\tif (sym == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "\t\t\tcontinue\n",
    "\t\tif sym in dev_br_list:\n",
    "\t\t\tret = 1\n",
    "\t\t\tbreak\n",
    "\treturn ret\n",
    "\n",
    "def get_freq(row, col_lr, parents_set_lr, parents_set_sr):\n",
    "    # also compute frequency of IL samples in addition to LR col\n",
    "    if row['PLATFORM'] == 'LR_IL':\n",
    "        #if row[col_lr] == '.' or row[col_lr] == '':\n",
    "        #    return 0\n",
    "        parents = [x for x in row[col_lr].split(',') if x in parents_set_lr]\n",
    "        freq = float(f'{len(parents) / len(parents_set_lr):.4f}')\n",
    "        return freq\n",
    "    elif row['PLATFORM'] == 'LR':\n",
    "        parents_lr = [x for x in row[col_lr].split(',') if x in parents_set_lr]\n",
    "        freq_lr = float(f'{len(parents_lr) / len(parents_set_lr):.4f}')\n",
    "        parents_il = [x for x in row['SV2_GT_SAMPLES'].split(',') if x in parents_set_lr]\n",
    "        freq_il = float(f'{len(parents_il) / len(parents_set_lr):.4f}')\n",
    "        freq = max(freq_lr, freq_il)\n",
    "        return freq\n",
    "    elif row['PLATFORM'] == 'IL':\n",
    "        parents = [x for x in (row['HET_SAMPLES'].split(',') + row['HOMALT_SAMPLES'].split(',')) if x in parents_set_sr]\n",
    "        freq = float(f'{len(parents) / len(parents_set_sr):.4f}')\n",
    "        return freq\n",
    "    else:\n",
    "        return 'WRONG_FQ'\n",
    "\n",
    "def xconserv(row, col, cons_thr):\n",
    "    ret = 0\n",
    "    if (row[col] != '.') and (float(row[col]) >= cons_thr):\n",
    "        ret = 1\n",
    "    return ret\n",
    "\n",
    "def xsHet(row, s_thr):\n",
    "\tret = 0\n",
    "\tfor s, con in zip(row['GENES_S_HET'].split(','), row['Consequence'].split(',')):\n",
    "\t\tif (s == '.') or ('upstream_gene_variant' in con) or ('downstream_gene_variant' in con):\n",
    "\t\t\tcontinue\n",
    "\t\tif float(s) >= s_thr:\n",
    "\t\t\tret = 1\n",
    "\t\t\tbreak\n",
    "\treturn ret\n",
    "\n",
    "def stratify_samples(row, col):\n",
    "    samples_hp12 = []\n",
    "    samples_ncom = []\n",
    "    ad1_m_alt_samples = row['AD1_M_ALT_SAMPLES'].split(',')\n",
    "    ad1_p_alt_samples = row['AD1_P_ALT_SAMPLES'].split(',')\n",
    "    for sample in row[col].split(','):\n",
    "        if (sample in ad1_m_alt_samples) and (sample in ad1_p_alt_samples):\n",
    "            samples_hp12.append(sample)\n",
    "        else:\n",
    "            samples_ncom.append(sample)\n",
    "    if len(samples_hp12) == 0:\n",
    "        samples_hp12.append('.')\n",
    "    if len(samples_ncom) == 0:\n",
    "        samples_ncom.append('.')\n",
    "    return ','.join(samples_hp12), ','.join(samples_ncom)\n",
    "\n",
    "def platform_redef(row, col_plat_old, col_lr):\n",
    "    plat_old = row[col_plat_old]\n",
    "    if plat_old == 'LR_IL':\n",
    "        return plat_old\n",
    "    if plat_old == 'LR':\n",
    "        if row['SV2_GT_SAMPLES'] != '.':\n",
    "            return 'LR_IL'\n",
    "        else:\n",
    "            return 'LR'\n",
    "    if plat_old == 'IL':\n",
    "        if row[col_lr] != '.':\n",
    "            return 'LR_IL'\n",
    "        else:\n",
    "            return 'IL'\n",
    "    return 'WRONG'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5bd8b7dd-4f89-4c71-b90c-d38cb02ce559",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_trans(row, col_lr, mom_kids_dict, dad_kids_dict, parent_kids_dict, mate_dict ,aff_dict):\n",
    "    num_tran_case = 0\n",
    "    num_tran_control = 0\n",
    "    num_nontran_case = 0\n",
    "    num_nontran_control = 0\n",
    "    if row['PLATFORM'] != 'IL':\n",
    "        moms = [x for x in row[col_lr].split(',') if x in mom_kids_dict]\n",
    "        dads = [x for x in row[col_lr].split(',') if x in dad_kids_dict]\n",
    "        mat_inh = [k for k in row['MAT_INH_LR_LC'].split(',')]\n",
    "        pat_inh = [k for k in row['PAT_INH_LR_LC'].split(',')]\n",
    "        ad1_m_ref_samples = row['AD1_M_REF_SAMPLES'].split(',')\n",
    "        ad1_p_ref_samples = row['AD1_P_REF_SAMPLES'].split(',')\n",
    "        for mom in moms:\n",
    "            num_tran_case += len([k for k in mom_kids_dict[mom] if ((k in mat_inh) and (aff_dict[k] == 'Yes'))])\n",
    "            num_tran_control += len([k for k in mom_kids_dict[mom] if ((k in mat_inh) and (aff_dict[k] == 'No'))])\n",
    "            num_nontran_case += len([k for k in mom_kids_dict[mom] if ((k not in mat_inh) and (k in ad1_m_ref_samples) and (aff_dict[k] == 'Yes'))])\n",
    "            num_nontran_control += len([k for k in mom_kids_dict[mom] if ((k not in mat_inh) and (k in ad1_m_ref_samples) and (aff_dict[k] == 'No'))])\n",
    "        for dad in dads:\n",
    "            num_tran_case += len([k for k in dad_kids_dict[dad] if ((k in pat_inh) and (aff_dict[k] == 'Yes'))])\n",
    "            num_tran_control += len([k for k in dad_kids_dict[dad] if ((k in pat_inh) and (aff_dict[k] == 'No'))])\n",
    "            num_nontran_case += len([k for k in dad_kids_dict[dad] if ((k not in pat_inh) and (k in ad1_p_ref_samples) and (aff_dict[k] == 'Yes'))])\n",
    "            num_nontran_control += len([k for k in dad_kids_dict[dad] if ((k not in pat_inh) and (k in ad1_p_ref_samples) and (aff_dict[k] == 'No'))])\n",
    "    else:\n",
    "        parents = [x for x in row['IL_SAMPLES_LR'].split(',') if x in parent_kids_dict]\n",
    "        het_samples = [x for x in row['HET_SAMPLES_LR'].split(',')]\n",
    "        homalt_samples = [x for x in row['HOMALT_SAMPLES_LR'].split(',')]\n",
    "        samples = het_samples + homalt_samples\n",
    "        parent_proc = set()\n",
    "        #tran_kids = [k for k in row['PMAT_INH_IL_LR'].split(',')]\n",
    "        for parent in parents:\n",
    "            mate = mate_dict[parent]\n",
    "            if parent in parent_proc:\n",
    "                continue\n",
    "            parent_proc.update([parent, mate])\n",
    "            if parent in homalt_samples and mate not in parents:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k in het_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 1\n",
    "            elif parent in homalt_samples and mate in het_samples:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k in het_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 1\n",
    "                            num_nontran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 1\n",
    "                            num_nontran_control += 1\n",
    "                    elif k in homalt_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 2\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 2\n",
    "            elif parent in homalt_samples and mate in homalt_samples:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k in homalt_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 2\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 2\n",
    "            elif parent in het_samples and mate not in parents:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k not in samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_nontran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_nontran_control += 1\n",
    "                    elif k in het_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 1\n",
    "            elif parent in het_samples and mate in het_samples:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k not in samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_nontran_case += 2\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_nontran_control += 2\n",
    "                    elif k in het_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 1\n",
    "                            num_nontran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 1\n",
    "                            num_nontran_control += 1\n",
    "                    elif k in homalt_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 2\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 2\n",
    "            elif parent in het_samples and mate in homalt_samples:\n",
    "                for k in parent_kids_dict[parent]:\n",
    "                    if k in het_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 1\n",
    "                            num_nontran_case += 1\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 1\n",
    "                            num_nontran_control += 1\n",
    "                    elif k in homalt_samples:\n",
    "                        if aff_dict[k] == 'Yes':\n",
    "                            num_tran_case += 2\n",
    "                        elif aff_dict[k] == 'No':\n",
    "                            num_tran_control += 2\n",
    "    return num_tran_case, num_tran_control, num_nontran_case, num_nontran_control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c20418bc-e8ea-4c43-aeb7-cb9b5679befa",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_big_table(file_name):\n",
    "    df = pd.read_table(file_name, sep='\\t', header=0)\n",
    "    \n",
    "    main_chroms = [f'chr{x}' for x in range(1,23)] + ['chrX', 'chrY']\n",
    "    df = pd.DataFrame(df.loc[df.CHROM.isin(main_chroms)])\n",
    "\n",
    "    ### to avoid recongizing NaN as float for empty entries\n",
    "    df.LOF_DUP_INV_GENES = df.LOF_DUP_INV_GENES.fillna('')\n",
    "    \n",
    "    col_gns = ['GNOCCHI_MAX', 'GNOCCHI_MEAN']\n",
    "    gn_thrs = [4, 3, 2, 1]\n",
    "    for col_gn in col_gns:\n",
    "        for gn_thr in gn_thrs:\n",
    "            print(f'{col_gn}_{gn_thr}...')\n",
    "            df[f'X_{col_gn}_{gn_thr}'] = df.apply(lambda row: xgnocchi(row, col_gn, gn_thr), axis=1)\n",
    "            feature = f'X_{col_gn}_{gn_thr}'\n",
    "            print(f'{col_gn}_{gn_thr} subsets...')\n",
    "            gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "            df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "            gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "            df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "            gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "            #gen_list = ['.']\n",
    "            df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    print('frequence calc...')\n",
    "    cols = ['AD2_SAMPLES', 'AD3_SAMPLES', 'AD4_SAMPLES', 'AD5_SAMPLES', \n",
    "            'SQ5_SAMPLES', 'SQ10_SAMPLES', 'SQ20_SAMPLES', 'SQ30_SAMPLES', 'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES']\n",
    "    for col in cols:\n",
    "        print(f'   {col}')\n",
    "        df[f'freq_{col}'] = df.apply(lambda row: get_freq(row, col, parents_set_lr, parents_set_sr), axis=1)\n",
    "\n",
    "    print('count the number of samples in filter columns...')\n",
    "    cols = ['ZERO_COV_SAMPLES', 'SQ5_SAMPLES', 'SQ10_SAMPLES', 'SQ20_SAMPLES', 'SQ30_SAMPLES', \n",
    "            'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES', \n",
    "            'AD2_SAMPLES', 'AD3_SAMPLES', 'AD4_SAMPLES', 'AD5_SAMPLES', 'HET_SAMPLES', 'HOMALT_SAMPLES', 'IL_SAMPLES', 'IL_GT_SAMPLES', 'SV2_GT_SAMPLES']\n",
    "    for col in cols:\n",
    "        print(col)\n",
    "        df[f'NUM_{col}'] = df.apply(lambda row: count_num(row, col), axis=1)\n",
    "\n",
    "    print('count number of case/control...')\n",
    "    cols = ['SQ5_SAMPLES', 'SQ10_SAMPLES', 'SQ20_SAMPLES', 'SQ30_SAMPLES', 'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES', \n",
    "            'AD2_SAMPLES', 'AD3_SAMPLES', 'AD4_SAMPLES', 'AD5_SAMPLES', 'HET_SAMPLES', 'HOMALT_SAMPLES', 'IL_SAMPLES', 'IL_GT_SAMPLES', 'SV2_GT_SAMPLES']\n",
    "    df[[f'{col}_CASE_NUM' for col in cols]+[f'{col}_CONTROL_NUM' for col in cols]] = df.apply(lambda row: status_num(row, cols), \n",
    "                                                                                                         axis=1, result_type='expand')\n",
    "\n",
    "\n",
    "    print('IL samples names to LR sample names...')\n",
    "    rename_IL_to_LR = {'REACH000236_IL': 'REACH000236_PB', \n",
    "                       'REACH000436_IL': 'REACH000436_PB', \n",
    "                       'REACH000530_IL': 'REACH000530_PB', \n",
    "                       'REACH000531_IL': 'REACH000531_ONT', \n",
    "                       'REACH000532_IL': 'REACH000532_ONT'}\n",
    "    df['IL_SAMPLES_LR'] = df.apply(lambda row: ','.join([x if x not in rename_IL_to_LR else rename_IL_to_LR[x] \n",
    "                                                         for x in row['IL_SAMPLES'].split(',')]), axis=1)\n",
    "    df['HET_SAMPLES_LR'] = df.apply(lambda row: ','.join([x if x not in rename_IL_to_LR else rename_IL_to_LR[x] \n",
    "                                                         for x in row['HET_SAMPLES'].split(',')]), axis=1)\n",
    "    df['HOMALT_SAMPLES_LR'] = df.apply(lambda row: ','.join([x if x not in rename_IL_to_LR else rename_IL_to_LR[x] \n",
    "                                                         for x in row['HOMALT_SAMPLES'].split(',')]), axis=1)\n",
    "    df['IL_GT_SAMPLES_LR'] = df.apply(lambda row: ','.join([x if x not in rename_IL_to_LR else rename_IL_to_LR[x] \n",
    "                                                         for x in row['IL_GT_SAMPLES'].split(',')]), axis=1)\n",
    "    cols = ['AD2_SAMPLES', 'AD3_SAMPLES', 'AD4_SAMPLES', 'AD5_SAMPLES', \n",
    "            'SQ5_SAMPLES', 'SQ10_SAMPLES', 'SQ20_SAMPLES', 'SQ30_SAMPLES', 'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES']\n",
    "    for col in cols:\n",
    "        df[col] = df.apply(lambda row: ','.join([x if x not in rename_IL_to_LR else rename_IL_to_LR[x] \n",
    "                                                 for x in row[col].split(',')]), axis=1)\n",
    "    \n",
    "    print('generate alternative PLATFOMR from cross validation with IL calls...')\n",
    "    col_plat_old = 'PLATFORM'\n",
    "    col_lr = 'AD2_SAMPLES'\n",
    "    df['PLATFORM_CV'] =  df.apply(lambda row: platform_redef(row, col_plat_old, col_lr), axis=1)\n",
    "\n",
    "    print('count number of transmissions...')\n",
    "    cols = ['AD2_SAMPLES', 'AD3_SAMPLES', 'AD4_SAMPLES', 'AD5_SAMPLES', \n",
    "            'SQ5_SAMPLES', 'SQ10_SAMPLES', 'SQ20_SAMPLES', 'SQ30_SAMPLES', 'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES']\n",
    "    for col in cols:\n",
    "        print(f'   {col}')\n",
    "        df[[f'NUM_tran_case_{col}', f'NUM_tran_control_{col}', \n",
    "        f'NUM_nontran_case_{col}', f'NUM_nontran_control_{col}']] = df.apply(lambda row: get_trans(row, col, mom_kids_dict, dad_kids_dict, \n",
    "                                                                                                   parent_kids_dict, mate_dict, aff_dict), \n",
    "                                                                            axis=1, result_type='expand')\n",
    "        #df[[f'NUM_tran_case_asd_{col}', f'NUM_tran_control_asd_{col}', \n",
    "        #f'NUM_nontran_case_asd_{col}', f'NUM_nontran_control_asd_{col}']] = df.apply(lambda row: get_trans(row, col, mom_kids_dict, dad_kids_dict, \n",
    "        #                                                                                           parent_kids_dict, mate_dict, aff_dict_asd), \n",
    "        #                                                                    axis=1, result_type='expand')\n",
    "    \n",
    "    biotypes = ['lncRNA' ,'miRNA' ,'misc_RNA' ,'processed_transcript' ,'snoRNA' ,'snRNA' ,'TEC', 'protein_coding']\n",
    "    for biotype in biotypes:\n",
    "        print(f'biotype: {biotype}')\n",
    "        df[f'X_{biotype}'] = df.apply(lambda row: xfeature(row, biotype), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('protein coding subsets...')\n",
    "    feature = 'X_protein_coding'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    print('pli...')\n",
    "    pli_thr = .99\n",
    "    df['X_PLI'] = df.apply(lambda row: xpli(row, pli_thr), axis=1)\n",
    "\n",
    "    print('pli >0.9...')\n",
    "    pli_thr = .9\n",
    "    df['X_PLIp9'] = df.apply(lambda row: xpli(row, pli_thr), axis=1)\n",
    "\n",
    "    print('low pli...')\n",
    "    pli_thr = .9\n",
    "    df['X_LOWPLI'] = df.apply(lambda row: xlowpli(row, pli_thr), axis=1)\n",
    "\n",
    "    print('loeuf...')\n",
    "    l_thr = .37\n",
    "    #l_thr = .1\n",
    "    df['X_LOEUF'] = df.apply(lambda row: xloeuf(row, l_thr), axis=1)\n",
    "\n",
    "    print('ASD...')\n",
    "    f_thr = .05\n",
    "    f_type = 'ASD'\n",
    "    df['X_FDR_ASD'] = df.apply(lambda row: xfdr(row, f_type, f_thr), axis=1)\n",
    "\n",
    "    print('DD...')\n",
    "    f_type = 'DD'\n",
    "    df['X_FDR_DD'] = df.apply(lambda row: xfdr(row, f_type, f_thr), axis=1)\n",
    "\n",
    "    print('NDD...')\n",
    "    f_type = 'NDD'\n",
    "    df['X_FDR_NDD'] = df.apply(lambda row: xfdr(row, f_type, f_thr), axis=1)\n",
    "\n",
    "    print('intergenic...')\n",
    "    df['X_intergenic'] = df.apply(lambda row: xintergenic(row), axis=1)\n",
    "\n",
    "    print('genic...')\n",
    "    df['X_genic'] = df.apply(lambda row: xgenic(row), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('genic subsets...')\n",
    "    feature = 'X_genic'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    ####################################\n",
    "    print('pli subsets...')\n",
    "    feature = 'X_PLI'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('pli >0.9 subsets...')\n",
    "    feature = 'X_PLIp9'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('low pli subsets...')\n",
    "    feature = 'X_LOWPLI'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    ####################################\n",
    "    print('loeuf subsets...')\n",
    "    feature = 'X_LOEUF'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('ASD subsets...')\n",
    "    feature = 'X_FDR_ASD'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('DD subsets...')\n",
    "    feature = 'X_FDR_DD'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('NDD subsets...')\n",
    "    feature = 'X_FDR_NDD'\n",
    "    gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "    df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "    df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "    gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "    #gen_list = ['.']\n",
    "    df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    print('LoF...')\n",
    "    df['LOF'] = df.apply(lambda row: lof(row), axis=1)\n",
    "\n",
    "    print('LoF constrained subsets...')\n",
    "    feature1 = 'LOF'\n",
    "    feature2 = 'X_PLI_cds'\n",
    "    df['X_LOF_PLI_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_PLIp9_cds'\n",
    "    df['X_LOF_PLIp9_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_LOWPLI_cds'\n",
    "    df['X_LOF_LOWPLI_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_LOEUF_cds'\n",
    "    df['X_LOF_LOEUF_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_ASD_cds'\n",
    "    df['X_LOF_FDR_ASD_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_DD_cds'\n",
    "    df['X_LOF_FDR_DD_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_NDD_cds'\n",
    "    df['X_LOF_FDR_NDD_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    \n",
    "    print('FB PR...')\n",
    "    fb_set = set(['1_TssA', '2_TssAFlnk', '10_TssBiv', '11_BivFlnk'])\n",
    "    df['X_FB_PR'] = df.apply(lambda row: xfb_pr_enh(row, fb_set), axis=1)\n",
    "\n",
    "    print('FB ENH...')\n",
    "    fb_set = set(['6_EnhG', '7_Enh', '12_EnhBiv'])\n",
    "    df['X_FB_ENH'] = df.apply(lambda row: xfb_pr_enh(row, fb_set), axis=1)\n",
    "\n",
    "    print('FANTOM...')\n",
    "    df['X_FANTOM'] = df.apply(lambda row: xfantom(row), axis=1)\n",
    "\n",
    "    ### placeholder for gnocchi calc\n",
    "    \n",
    "\n",
    "    print('protein_coding CDS subsets...')\n",
    "    feature1 = 'X_protein_coding_cds'\n",
    "    feature2 = 'X_PLI'\n",
    "    df['X_protein_coding_cds_pli'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_LOEUF'\n",
    "    df['X_protein_coding_cds_loeuf'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_ASD'\n",
    "    df['X_protein_coding_cds_fdr_asd'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_DD'\n",
    "    df['X_protein_coding_cds_fdr_dd'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "    feature2 = 'X_FDR_NDD'\n",
    "    df['X_protein_coding_cds_fdr_ndd'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "\n",
    "    print('brain span genes...')\n",
    "    df['X_DEV_BR_GENE_10'] = df.apply(lambda row: xdev_br(row, br_flt_10), axis=1)\n",
    "    df['X_DEV_BR_GENE_20'] = df.apply(lambda row: xdev_br(row, br_flt_20), axis=1)\n",
    "    df['X_DEV_BR_GENE_30'] = df.apply(lambda row: xdev_br(row, br_flt_30), axis=1)\n",
    "    df['X_DEV_BR_GENE_40'] = df.apply(lambda row: xdev_br(row, br_flt_40), axis=1)\n",
    "    df['X_DEV_BR_GENE_50'] = df.apply(lambda row: xdev_br(row, br_flt_50), axis=1)\n",
    "\n",
    "    ####################################\n",
    "    print('dev brain genes subsets...')\n",
    "    for rpkm in [10, 20, 30, 40, 50]:\n",
    "        feature = f'X_DEV_BR_GENE_{rpkm}'\n",
    "        gen_list = ['CDS', 'start_codon', 'stop_codon']\n",
    "        df[f'{feature}_cds'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "        gen_list = ['five_prime_UTR', 'three_prime_UTR', 'TSS']\n",
    "        df[f'{feature}_utr'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "    \n",
    "        gen_list = ['CTCF-only' ,'dELS' ,'DNase-H3K4me3' ,'exon' ,'pELS' ,'PLS', '.']\n",
    "        #gen_list = ['.']\n",
    "        df[f'{feature}_intron'] = df.apply(lambda row: xgencode(row, gen_list, feature), axis=1)\n",
    "\n",
    "    print('LoF dev brain...')\n",
    "    feature1 = 'LOF'\n",
    "    for rpkm in [10, 20, 30, 40, 50]:\n",
    "        feature2 = f'X_DEV_BR_GENE_{rpkm}_cds'\n",
    "        df[f'X_LOF_DEV_BR_{rpkm}_cds'] = df.apply(lambda row: x2features(row, feature1, feature2), axis=1)\n",
    "\n",
    "    print('dev brain and PLI...')\n",
    "    feature1 = 'X_PLIp9'\n",
    "    for feature2 in ['LOF_DEV_BR_10_cds', \n",
    "                     'DEV_BR_GENE_10', 'DEV_BR_GENE_10_cds', 'DEV_BR_GENE_10_utr', 'DEV_BR_GENE_10_intron']:\n",
    "        df[f'{feature1}_{feature2}'] = df.apply(lambda row: x2features(row, feature1, f'X_{feature2}'), axis=1)\n",
    "    \n",
    "    print('writing the output...')\n",
    "    df.to_csv(f'{dir_out}/test.tsv', sep='\\t', header=True, index=False)\n",
    "    print('+'*40)\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4fd0eeb0-54ff-48b5-8b3e-b6acc2ce8124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SAMPLE</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>MEAN_COVERAGE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG004</td>\n",
       "      <td>ONT</td>\n",
       "      <td>81.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REACH000626</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REACH000546</td>\n",
       "      <td>ONT</td>\n",
       "      <td>10.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REACH000683</td>\n",
       "      <td>ONT</td>\n",
       "      <td>9.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REACH000563</td>\n",
       "      <td>ONT</td>\n",
       "      <td>8.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>REACH000284</td>\n",
       "      <td>PB</td>\n",
       "      <td>3.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>REACH000672</td>\n",
       "      <td>PB</td>\n",
       "      <td>5.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>REACH000292</td>\n",
       "      <td>PB</td>\n",
       "      <td>9.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>REACH000066</td>\n",
       "      <td>PB</td>\n",
       "      <td>1.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>2602-2-1</td>\n",
       "      <td>PB</td>\n",
       "      <td>0.89</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>288 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          SAMPLE COHORT  MEAN_COVERAGE\n",
       "0          HG004    ONT          81.44\n",
       "1    REACH000626    ONT          10.51\n",
       "2    REACH000546    ONT          10.87\n",
       "3    REACH000683    ONT           9.97\n",
       "4    REACH000563    ONT           8.68\n",
       "..           ...    ...            ...\n",
       "283  REACH000284     PB           3.43\n",
       "284  REACH000672     PB           5.02\n",
       "285  REACH000292     PB           9.00\n",
       "286  REACH000066     PB           1.61\n",
       "287     2602-2-1     PB           0.89\n",
       "\n",
       "[288 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "file_cov = '../data/coverages_mod.tsv'\n",
    "df_cov = pd.read_table(file_cov, header=0, sep='\\t')\n",
    "multiplat_samples = ['REACH000236', 'REACH000530', 'REACH000531', 'REACH000532', 'REACH000436']\n",
    "df_cov['SAMPLE'] = df_cov.apply(lambda row: row['SAMPLE']+'_'+row['COHORT'] if row['SAMPLE'] in multiplat_samples else row['SAMPLE'], axis=1)\n",
    "display(df_cov)\n",
    "\n",
    "def get_plat(sample):\n",
    "    try:\n",
    "        ret = df_cov[df_cov.SAMPLE == sample]['COHORT'].values[0]\n",
    "    except:\n",
    "        print(f'problem with samples: {sample}')\n",
    "        ret = ''\n",
    "    return ret\n",
    "\n",
    "plat_dict = {sample: get_plat(sample) for sample in df_cov.SAMPLE.tolist()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "54ffde58-4478-414c-89c9-5a7f4abde50a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5824, 32)\n",
      "5824\n",
      "3045\n",
      "1983\n",
      "1392\n",
      "1078\n"
     ]
    }
   ],
   "source": [
    "file_brainspan = '../data/gene_list_RPKM_10.csv'\n",
    "dtype = {'gene_id': str, 'entrez_id': str}\n",
    "df_br = pd.read_table(file_brainspan, sep=',', header=0, dtype=dtype)\n",
    "br_flt_10 = df_br.loc[df_br.mean_rpkm>10].gene_symbol.tolist()\n",
    "br_flt_20 = df_br.loc[df_br.mean_rpkm>20].gene_symbol.tolist()\n",
    "br_flt_30 = df_br.loc[df_br.mean_rpkm>30].gene_symbol.tolist()\n",
    "br_flt_40 = df_br.loc[df_br.mean_rpkm>40].gene_symbol.tolist()\n",
    "br_flt_50 = df_br.loc[df_br.mean_rpkm>50].gene_symbol.tolist()\n",
    "print(df_br.shape)\n",
    "print(len(br_flt_10))\n",
    "print(len(br_flt_20))\n",
    "print(len(br_flt_30))\n",
    "print(len(br_flt_40))\n",
    "print(len(br_flt_50))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0e21df02-cd54-41b3-ae1d-2ac3c544ecf1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>dad</th>\n",
       "      <th>mom</th>\n",
       "      <th>sex</th>\n",
       "      <th>phen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2323</td>\n",
       "      <td>2323-2-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2602</td>\n",
       "      <td>2602-2-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3392</td>\n",
       "      <td>3392-2-3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3538</td>\n",
       "      <td>3538-2-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3939</td>\n",
       "      <td>3939-3-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000681</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000684</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000685</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>280 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     famid    sample_id          dad          mom  sex  phen\n",
       "0     2323     2323-2-1            0            0    2     2\n",
       "1     2602     2602-2-1            0            0    2     2\n",
       "2     3392     3392-2-3            0            0    2     2\n",
       "3     3538     3538-2-1            0            0    2     2\n",
       "4     3939     3939-3-1            0            0    2     2\n",
       "..     ...          ...          ...          ...  ...   ...\n",
       "275  F0270  REACH000681  REACH000683  REACH000682    1     2\n",
       "276  F0270  REACH000682            0            0    2     2\n",
       "277  F0270  REACH000683            0            0    1     2\n",
       "278  F0270  REACH000684  REACH000683  REACH000682    1     2\n",
       "279  F0270  REACH000685  REACH000683  REACH000682    1     1\n",
       "\n",
       "[280 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "complete trios:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>dad</th>\n",
       "      <th>mom</th>\n",
       "      <th>sex</th>\n",
       "      <th>phen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>REACH000270</td>\n",
       "      <td>REACH000269</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>REACH000440</td>\n",
       "      <td>REACH000439</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>F0065</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>REACH000067</td>\n",
       "      <td>REACH000066</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>REACH000088</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000089</td>\n",
       "      <td>REACH000088</td>\n",
       "      <td>REACH000087</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>F0266</td>\n",
       "      <td>REACH000660</td>\n",
       "      <td>REACH000662</td>\n",
       "      <td>REACH000661</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>F0266</td>\n",
       "      <td>REACH000663</td>\n",
       "      <td>REACH000662</td>\n",
       "      <td>REACH000661</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000681</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000684</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>F0270</td>\n",
       "      <td>REACH000685</td>\n",
       "      <td>REACH000683</td>\n",
       "      <td>REACH000682</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     famid    sample_id          dad          mom  sex  phen\n",
       "10   F0026  REACH000026  REACH000270  REACH000269    1     2\n",
       "11   F0058  REACH000058  REACH000440  REACH000439    1     2\n",
       "12   F0065  REACH000065  REACH000067  REACH000066    1     2\n",
       "15   F0078  REACH000086  REACH000088  REACH000087    1     2\n",
       "18   F0078  REACH000089  REACH000088  REACH000087    2     2\n",
       "..     ...          ...          ...          ...  ...   ...\n",
       "271  F0266  REACH000660  REACH000662  REACH000661    2     2\n",
       "274  F0266  REACH000663  REACH000662  REACH000661    1     1\n",
       "275  F0270  REACH000681  REACH000683  REACH000682    1     2\n",
       "278  F0270  REACH000684  REACH000683  REACH000682    1     2\n",
       "279  F0270  REACH000685  REACH000683  REACH000682    1     1\n",
       "\n",
       "[117 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sex\n",
      "1    74\n",
      "2    43\n",
      "Name: count, dtype: int64\n",
      "phen\n",
      "2    92\n",
      "1    25\n",
      "Name: count, dtype: int64\n",
      "len(parents_set_lr): 126\n",
      "len(parents_set_sr): 126\n",
      "len(parent_kids_dict): 126\n",
      "len(mom_kids_dict): 63\n",
      "len(dad_kids_dict): 63\n",
      "len(mate_dict): 126\n"
     ]
    }
   ],
   "source": [
    "file_psam = '../data/REACH_LR_platform.psam'\n",
    "df_p = pd.read_table(file_psam, sep='\\t', header=None, names = ['famid', 'sample_id', 'dad', 'mom', 'sex', 'phen'])\n",
    "display(df_p)\n",
    "\n",
    "# make mom and dad dictionary\n",
    "sample_dad_dict = {}\n",
    "sample_mom_dict = {}\n",
    "for sample, dad, mom in zip(df_p.sample_id, df_p.dad, df_p.mom):\n",
    "    #print(sample, dad, mom)\n",
    "    sample_dad_dict[sample] = dad\n",
    "    sample_mom_dict[sample] = mom\n",
    "\n",
    "print('complete trios:')\n",
    "df_trios = df_p.loc[df_p.dad.isin(df_p.sample_id) & df_p.mom.isin(df_p.sample_id)]\n",
    "display(df_trios)\n",
    "print(df_trios.sex.value_counts())\n",
    "print(df_trios.phen.value_counts())\n",
    "\n",
    "parents_set_lr = set(df_trios.dad.tolist()) | set(df_trios.mom.tolist())\n",
    "print(f'len(parents_set_lr): {len(parents_set_lr)}')\n",
    "#print(parents_set_lr)\n",
    "\n",
    "rename_samples_dict_IL = {'REACH000236_PB': 'REACH000236_IL', \n",
    "                          'REACH000436_PB': 'REACH000436_IL', \n",
    "                          'REACH000530_PB': 'REACH000530_IL', \n",
    "                          'REACH000531_ONT': 'REACH000531_IL', \n",
    "                          'REACH000532_ONT': 'REACH000532_IL'}\n",
    "parents_set_sr = {x if x not in rename_samples_dict_IL else rename_samples_dict_IL[x] for x in parents_set_lr}\n",
    "print(f'len(parents_set_sr): {len(parents_set_sr)}')\n",
    "#print(parents_set_sr)\n",
    "\n",
    "parent_kids_dict = {}\n",
    "mom_kids_dict = {}\n",
    "dad_kids_dict = {}\n",
    "for parent in parents_set_lr:\n",
    "    parent_kids_dict[parent] = df_trios.loc[(df_trios.dad == parent) | (df_trios.mom == parent)].sample_id.tolist()\n",
    "    if parent in df_trios.dad.tolist():\n",
    "        dad_kids_dict[parent] = df_trios.loc[(df_trios.dad == parent)].sample_id.tolist()\n",
    "    if parent in df_trios.mom.tolist():\n",
    "        mom_kids_dict[parent] = df_trios.loc[(df_trios.mom == parent)].sample_id.tolist()\n",
    "print(f'len(parent_kids_dict): {len(parent_kids_dict)}')\n",
    "#print(parent_kids_dict)\n",
    "print(f'len(mom_kids_dict): {len(mom_kids_dict)}')\n",
    "#print(mom_kids_dict)\n",
    "print(f'len(dad_kids_dict): {len(dad_kids_dict)}')\n",
    "#print(dad_kids_dict)\n",
    "\n",
    "mate_dict = {}\n",
    "for dad, mom in zip(df_trios.dad, df_trios.mom):\n",
    "    mate_dict[dad] = mom\n",
    "    mate_dict[mom] = dad\n",
    "print(f'len(mate_dict): {len(mate_dict)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1f545339-efcc-462d-9ed8-5732a1d0e445",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_file = '../data/REACH_sample_info.tsv'\n",
    "df_meta = pd.read_table(meta_file, sep='\\t', header=0)\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_meta['Sample_ID'] = df_meta.apply(lambda row: rename_samples_dict[row['Sample_ID']] \n",
    "                                     if row['Sample_ID'] in rename_samples_dict else row['Sample_ID'], axis=1)\n",
    "\n",
    "aff_dict = {}\n",
    "for sample, aff in zip(df_meta['Sample_ID'].tolist(), df_meta['Affected'].tolist()):\n",
    "    aff_dict[sample] = aff\n",
    "    \n",
    "aff_dict_asd = {}\n",
    "for sample, aff in zip(df_meta['Sample_ID'].tolist(), df_meta['Affected_ASD'].tolist()):\n",
    "    aff_dict_asd[sample] = aff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "91bf2207-b3ce-4252-a231-f92b0a33dfa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_fdr:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_gencodeV33</th>\n",
       "      <th>FDR_TADA_ASD</th>\n",
       "      <th>FDR_TADA_DD</th>\n",
       "      <th>FDR_TADA_NDD</th>\n",
       "      <th>p_TADA_ASD</th>\n",
       "      <th>p_TADA_DD</th>\n",
       "      <th>p_TADA_NDD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SYNGAP1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CHD8</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADNP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SHANK3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCN2A</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18123</th>\n",
       "      <td>PSG2</td>\n",
       "      <td>0.911036</td>\n",
       "      <td>0.913971</td>\n",
       "      <td>0.879392</td>\n",
       "      <td>0.999770</td>\n",
       "      <td>0.999656</td>\n",
       "      <td>0.999762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18124</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.911038</td>\n",
       "      <td>0.913973</td>\n",
       "      <td>0.879395</td>\n",
       "      <td>0.999827</td>\n",
       "      <td>0.999713</td>\n",
       "      <td>0.999821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18125</th>\n",
       "      <td>GJB2</td>\n",
       "      <td>0.911040</td>\n",
       "      <td>0.913975</td>\n",
       "      <td>0.879399</td>\n",
       "      <td>0.999885</td>\n",
       "      <td>0.999771</td>\n",
       "      <td>0.999881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18126</th>\n",
       "      <td>PSG1</td>\n",
       "      <td>0.911042</td>\n",
       "      <td>0.913979</td>\n",
       "      <td>0.879403</td>\n",
       "      <td>0.999942</td>\n",
       "      <td>0.999885</td>\n",
       "      <td>0.999940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18127</th>\n",
       "      <td>NBPF10</td>\n",
       "      <td>0.911044</td>\n",
       "      <td>0.913981</td>\n",
       "      <td>0.879407</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999943</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18128 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_gencodeV33  FDR_TADA_ASD  FDR_TADA_DD  FDR_TADA_NDD  p_TADA_ASD  \\\n",
       "0             SYNGAP1      0.000000     0.000000      0.000000    0.000000   \n",
       "1                CHD8      0.000000     0.000000      0.000000    0.000000   \n",
       "2                ADNP      0.000000     0.000000      0.000000    0.000000   \n",
       "3              SHANK3      0.000000     0.000000      0.000000    0.000000   \n",
       "4               SCN2A      0.000000     0.000000      0.000000    0.000000   \n",
       "...               ...           ...          ...           ...         ...   \n",
       "18123            PSG2      0.911036     0.913971      0.879392    0.999770   \n",
       "18124             NaN      0.911038     0.913973      0.879395    0.999827   \n",
       "18125            GJB2      0.911040     0.913975      0.879399    0.999885   \n",
       "18126            PSG1      0.911042     0.913979      0.879403    0.999942   \n",
       "18127          NBPF10      0.911044     0.913981      0.879407    1.000000   \n",
       "\n",
       "       p_TADA_DD  p_TADA_NDD  \n",
       "0       0.000000    0.000000  \n",
       "1       0.000000    0.000000  \n",
       "2       0.000000    0.000000  \n",
       "3       0.000000    0.000000  \n",
       "4       0.000000    0.000000  \n",
       "...          ...         ...  \n",
       "18123   0.999656    0.999762  \n",
       "18124   0.999713    0.999821  \n",
       "18125   0.999771    0.999881  \n",
       "18126   0.999885    0.999940  \n",
       "18127   0.999943    1.000000  \n",
       "\n",
       "[18128 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "asd: 185\n",
      "dd: 477\n",
      "ndd: 664\n",
      "df_pli:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lof.pLI</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.814400e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>6.691700e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>7.412300e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.996200e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>1.007500e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>3.084800e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>8.055900e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>3.807700e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>9.990700e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18644 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              lof.pLI\n",
       "gene                 \n",
       "A1BG     1.814400e-07\n",
       "A1CF     6.691700e-03\n",
       "A2M      7.412300e-02\n",
       "A2ML1    1.996200e-01\n",
       "A3GALT2  1.007500e-06\n",
       "...               ...\n",
       "ZYG11A   3.084800e-08\n",
       "ZYG11B   8.055900e-01\n",
       "ZYX      3.807700e-02\n",
       "ZZEF1    9.990700e-01\n",
       "ZZZ3     1.000000e+00\n",
       "\n",
       "[18644 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_lf:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lof.oe_ci.upper</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>1.530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11A</th>\n",
       "      <td>1.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>0.537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.290</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18644 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         lof.oe_ci.upper\n",
       "gene                    \n",
       "A1BG               1.257\n",
       "A1CF               0.789\n",
       "A2M                0.766\n",
       "A2ML1              0.829\n",
       "A3GALT2            1.530\n",
       "...                  ...\n",
       "ZYG11A             1.002\n",
       "ZYG11B             0.537\n",
       "ZYX                0.854\n",
       "ZZEF1              0.456\n",
       "ZZZ3               0.290\n",
       "\n",
       "[18644 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dnLOF and dnMIS SNV/indels\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(30, 22)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " fdr asd: 6\n",
      " fdr dd: 11\n",
      " fdr ndd: 13\n",
      " fdr brain genes: 21\n",
      "df_dn_snv:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_hg38</th>\n",
       "      <th>end_hg38</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>alternate_allele</th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>sex</th>\n",
       "      <th>consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>gene</th>\n",
       "      <th>gnomad_loeuf</th>\n",
       "      <th>mpc</th>\n",
       "      <th>cohort</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2173976</td>\n",
       "      <td>2173977</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000089</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>0.4</td>\n",
       "      <td>2.260710742</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99979</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16251664</td>\n",
       "      <td>16251665</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0279</td>\n",
       "      <td>REACH000718</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>FBXO42</td>\n",
       "      <td>0.274</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chr1</td>\n",
       "      <td>85464960</td>\n",
       "      <td>85464961</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0262</td>\n",
       "      <td>REACH000650</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDAH1</td>\n",
       "      <td>0.952</td>\n",
       "      <td>3.362419638</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.15438</td>\n",
       "      <td>0.862</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>chr11</td>\n",
       "      <td>61303998</td>\n",
       "      <td>61303999</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0285</td>\n",
       "      <td>REACH000741</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDB1</td>\n",
       "      <td>0.243</td>\n",
       "      <td>3.079873048</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>chr11</td>\n",
       "      <td>64773439</td>\n",
       "      <td>64773440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0162B</td>\n",
       "      <td>REACH000774</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>SF1</td>\n",
       "      <td>0.183</td>\n",
       "      <td>2.419035056</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>chr12</td>\n",
       "      <td>57269898</td>\n",
       "      <td>57269900</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>F0083</td>\n",
       "      <td>REACH000097</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>R3HDM2</td>\n",
       "      <td>0.176</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313</th>\n",
       "      <td>chr12</td>\n",
       "      <td>94298718</td>\n",
       "      <td>94298719</td>\n",
       "      <td>T</td>\n",
       "      <td>TTTGAC</td>\n",
       "      <td>F0274</td>\n",
       "      <td>REACH000700</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PLXNC1</td>\n",
       "      <td>0.273</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.75987</td>\n",
       "      <td>0.550</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>chr14</td>\n",
       "      <td>23034800</td>\n",
       "      <td>23034802</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>F0251</td>\n",
       "      <td>REACH000605</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>PSMB5</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99989</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>chr16</td>\n",
       "      <td>2056194</td>\n",
       "      <td>2056195</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0233</td>\n",
       "      <td>REACH000529</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>0.074</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>chr16</td>\n",
       "      <td>67629532</td>\n",
       "      <td>67629533</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>F0242</td>\n",
       "      <td>REACH000564</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained&amp;splice_region_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>0.148</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.205</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>chr17</td>\n",
       "      <td>8466944</td>\n",
       "      <td>8466945</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>F0203</td>\n",
       "      <td>REACH000404</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>...</td>\n",
       "      <td>NDEL1</td>\n",
       "      <td>0.296</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99993</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>chr17</td>\n",
       "      <td>59647575</td>\n",
       "      <td>59647576</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>F0245</td>\n",
       "      <td>REACH000578</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CLTC</td>\n",
       "      <td>0.104</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.100</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>chr17</td>\n",
       "      <td>67886275</td>\n",
       "      <td>67886278</td>\n",
       "      <td>AAG</td>\n",
       "      <td>A</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000482</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>BPTF</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.083</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6213087</td>\n",
       "      <td>6213088</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0208</td>\n",
       "      <td>REACH000423</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MLLT1</td>\n",
       "      <td>0.288</td>\n",
       "      <td>2.086213226</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>chr20</td>\n",
       "      <td>49374625</td>\n",
       "      <td>49374626</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0284</td>\n",
       "      <td>REACH000747</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>KCNB1</td>\n",
       "      <td>0.125</td>\n",
       "      <td>3.839540893</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.309</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>chr20</td>\n",
       "      <td>63442515</td>\n",
       "      <td>63442516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0215</td>\n",
       "      <td>REACH000457</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>KCNQ2</td>\n",
       "      <td>0.158</td>\n",
       "      <td>4.219752135</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.251</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>chr21</td>\n",
       "      <td>37490200</td>\n",
       "      <td>37490201</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0210</td>\n",
       "      <td>REACH000435</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>DYRK1A</td>\n",
       "      <td>0.214</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.172</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38300756</td>\n",
       "      <td>38300757</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0230</td>\n",
       "      <td>REACH000519</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CSNK1E</td>\n",
       "      <td>0.143</td>\n",
       "      <td>2.503748791</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.92771</td>\n",
       "      <td>0.555</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41129986</td>\n",
       "      <td>41129992</td>\n",
       "      <td>TAAGAG</td>\n",
       "      <td>T</td>\n",
       "      <td>F0226</td>\n",
       "      <td>REACH000504</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>EP300</td>\n",
       "      <td>0.099</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>chr3</td>\n",
       "      <td>70972633</td>\n",
       "      <td>70972634</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0140</td>\n",
       "      <td>REACH000189</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>FOXP1</td>\n",
       "      <td>0.175</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.250</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>chr4</td>\n",
       "      <td>46303465</td>\n",
       "      <td>46303466</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000482</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>GABRA2</td>\n",
       "      <td>0.229</td>\n",
       "      <td>2.615526082</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.98587</td>\n",
       "      <td>0.481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>chr4</td>\n",
       "      <td>109511995</td>\n",
       "      <td>109511996</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0237</td>\n",
       "      <td>REACH000544</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>SEC24B</td>\n",
       "      <td>0.194</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.418</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>chr5</td>\n",
       "      <td>10255973</td>\n",
       "      <td>10255974</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0248</td>\n",
       "      <td>REACH000589</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_lost&amp;NMD_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CCT5</td>\n",
       "      <td>0.188</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.210</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>chr5</td>\n",
       "      <td>150251807</td>\n",
       "      <td>150251808</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0267</td>\n",
       "      <td>REACH000664</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CAMK2A</td>\n",
       "      <td>0.251</td>\n",
       "      <td>2.923086944</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.141</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>chr5</td>\n",
       "      <td>173110640</td>\n",
       "      <td>173110641</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000754</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CREBRF</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2.210301239</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.174</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>chr6</td>\n",
       "      <td>43213912</td>\n",
       "      <td>43213913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>F0167</td>\n",
       "      <td>REACH000249</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>CUL9</td>\n",
       "      <td>0.28</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.321</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>chr7</td>\n",
       "      <td>74700299</td>\n",
       "      <td>74700300</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>F0191</td>\n",
       "      <td>REACH000349</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>GTF2I</td>\n",
       "      <td>0.27</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99996</td>\n",
       "      <td>0.195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>chr9</td>\n",
       "      <td>81585545</td>\n",
       "      <td>81585546</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>F0276</td>\n",
       "      <td>REACH000708</td>\n",
       "      <td>Control</td>\n",
       "      <td>Male</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>TLE1</td>\n",
       "      <td>0.239</td>\n",
       "      <td>2.626466068</td>\n",
       "      <td>REACH</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1324</th>\n",
       "      <td>chrX</td>\n",
       "      <td>41344087</td>\n",
       "      <td>41344090</td>\n",
       "      <td>CGA</td>\n",
       "      <td>C</td>\n",
       "      <td>F0221</td>\n",
       "      <td>REACH000479</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX3X</td>\n",
       "      <td>0.118</td>\n",
       "      <td>NA</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1325</th>\n",
       "      <td>chrX</td>\n",
       "      <td>41346519</td>\n",
       "      <td>41346520</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>F0240</td>\n",
       "      <td>REACH000555</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Female</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>DDX3X</td>\n",
       "      <td>0.118</td>\n",
       "      <td>2.375896712</td>\n",
       "      <td>REACH</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  start_hg38   end_hg38 reference_allele alternate_allele     fid  \\\n",
       "0      chr1     2173976    2173977                G                A   F0078   \n",
       "17     chr1    16251664   16251665                G                A   F0279   \n",
       "54     chr1    85464960   85464961                C                G   F0262   \n",
       "208   chr11    61303998   61303999                G                A   F0285   \n",
       "219   chr11    64773439   64773440                G                A  F0162B   \n",
       "299   chr12    57269898   57269900               TG                T   F0083   \n",
       "313   chr12    94298718   94298719                T           TTTGAC   F0274   \n",
       "382   chr14    23034800   23034802               CA                C   F0251   \n",
       "465   chr16     2056194    2056195                G                A   F0233   \n",
       "501   chr16    67629532   67629533                G                T   F0242   \n",
       "526   chr17     8466944    8466945                T                C   F0203   \n",
       "580   chr17    59647575   59647576                T               TC   F0245   \n",
       "596   chr17    67886275   67886278              AAG                A   F0221   \n",
       "647   chr19     6213087    6213088                G                A   F0208   \n",
       "838   chr20    49374625   49374626                G                A   F0284   \n",
       "857   chr20    63442515   63442516                A                G   F0215   \n",
       "870   chr21    37490200   37490201                C                T   F0210   \n",
       "890   chr22    38300756   38300757                G                A   F0230   \n",
       "896   chr22    41129986   41129992           TAAGAG                T   F0226   \n",
       "948    chr3    70972633   70972634                G                A   F0140   \n",
       "995    chr4    46303465   46303466                C                G   F0221   \n",
       "1021   chr4   109511995  109511996                C                T   F0237   \n",
       "1044   chr5    10255973   10255974                A                G   F0248   \n",
       "1087   chr5   150251807  150251808                G                A   F0267   \n",
       "1100   chr5   173110640  173110641                A                T   F0288   \n",
       "1136   chr6    43213912   43213913                G                A   F0167   \n",
       "1203   chr7    74700299   74700300                C                T   F0191   \n",
       "1283   chr9    81585545   81585546                A                G   F0276   \n",
       "1324   chrX    41344087   41344090              CGA                C   F0221   \n",
       "1325   chrX    41346519   41346520                C                G   F0240   \n",
       "\n",
       "              iid phenotype     sex                        consequence  ...  \\\n",
       "0     REACH000089       ASD  Female                   missense_variant  ...   \n",
       "17    REACH000718       ASD    Male                        stop_gained  ...   \n",
       "54    REACH000650       ASD    Male                   missense_variant  ...   \n",
       "208   REACH000741   Control    Male                   missense_variant  ...   \n",
       "219   REACH000774   Control    Male                   missense_variant  ...   \n",
       "299   REACH000097       ASD    Male                 frameshift_variant  ...   \n",
       "313   REACH000700   Control    Male                 frameshift_variant  ...   \n",
       "382   REACH000605       ASD    Male                 frameshift_variant  ...   \n",
       "465   REACH000529       ASD    Male            splice_acceptor_variant  ...   \n",
       "501   REACH000564       ASD    Male  stop_gained&splice_region_variant  ...   \n",
       "526   REACH000404       ASD  Female                          stop_lost  ...   \n",
       "580   REACH000578       ASD  Female                 frameshift_variant  ...   \n",
       "596   REACH000482       ASD  Female                 frameshift_variant  ...   \n",
       "647   REACH000423       ASD    Male                   missense_variant  ...   \n",
       "838   REACH000747       ASD    Male                   missense_variant  ...   \n",
       "857   REACH000457       ASD  Female                   missense_variant  ...   \n",
       "870   REACH000435       ASD  Female                        stop_gained  ...   \n",
       "890   REACH000519       ASD  Female                   missense_variant  ...   \n",
       "896   REACH000504       ASD  Female                 frameshift_variant  ...   \n",
       "948   REACH000189       ASD    Male                        stop_gained  ...   \n",
       "995   REACH000482       ASD  Female                   missense_variant  ...   \n",
       "1021  REACH000544       ASD    Male                        stop_gained  ...   \n",
       "1044  REACH000589       ASD    Male   stop_lost&NMD_transcript_variant  ...   \n",
       "1087  REACH000664       ASD    Male                   missense_variant  ...   \n",
       "1100  REACH000754       ASD    Male                   missense_variant  ...   \n",
       "1136  REACH000249       ASD    Male               splice_donor_variant  ...   \n",
       "1203  REACH000349       ASD    Male                        stop_gained  ...   \n",
       "1283  REACH000708   Control    Male                   missense_variant  ...   \n",
       "1324  REACH000479       ASD  Female                 frameshift_variant  ...   \n",
       "1325  REACH000555       ASD  Female                   missense_variant  ...   \n",
       "\n",
       "        gene  gnomad_loeuf          mpc cohort      PLI  LOEUF  FDR_ASD  \\\n",
       "0      PRKCZ           0.4  2.260710742  REACH  0.99979  0.415        0   \n",
       "17    FBXO42         0.274           NA  REACH  1.00000  0.271        0   \n",
       "54     DDAH1         0.952  3.362419638  REACH  0.15438  0.862        0   \n",
       "208     DDB1         0.243  3.079873048  REACH  1.00000  0.183        0   \n",
       "219      SF1         0.183  2.419035056  REACH  1.00000  0.220        0   \n",
       "299   R3HDM2         0.176           NA  REACH  1.00000  0.366        0   \n",
       "313   PLXNC1         0.273           NA  REACH  0.75987  0.550        0   \n",
       "382    PSMB5           0.3           NA  REACH  0.99989  0.238        0   \n",
       "465     TSC2         0.074           NA  REACH  1.00000  0.168        0   \n",
       "501     CTCF         0.148           NA  REACH  1.00000  0.205        1   \n",
       "526    NDEL1         0.296           NA  REACH  0.99993  0.324        0   \n",
       "580     CLTC         0.104           NA  REACH  1.00000  0.100        0   \n",
       "596     BPTF         0.099           NA  REACH  1.00000  0.083        0   \n",
       "647    MLLT1         0.288  2.086213226  REACH  0.99994  0.379        0   \n",
       "838    KCNB1         0.125  3.839540893  REACH  1.00000  0.309        1   \n",
       "857    KCNQ2         0.158  4.219752135  REACH  1.00000  0.251        0   \n",
       "870   DYRK1A         0.214           NA  REACH  1.00000  0.172        1   \n",
       "890   CSNK1E         0.143  2.503748791  REACH  0.92771  0.555        1   \n",
       "896    EP300         0.099           NA  REACH  1.00000  0.163        0   \n",
       "948    FOXP1         0.175           NA  REACH  1.00000  0.250        1   \n",
       "995   GABRA2         0.229  2.615526082  REACH  0.98587  0.481        0   \n",
       "1021  SEC24B         0.194           NA  REACH  0.99994  0.418        0   \n",
       "1044    CCT5         0.188           NA  REACH  1.00000  0.210        0   \n",
       "1087  CAMK2A         0.251  2.923086944  REACH  1.00000  0.141        1   \n",
       "1100  CREBRF           0.1  2.210301239  REACH  1.00000  0.174        0   \n",
       "1136    CUL9          0.28           NA  REACH  1.00000  0.321        0   \n",
       "1203   GTF2I          0.27           NA  REACH  0.99996  0.195        0   \n",
       "1283    TLE1         0.239  2.626466068  REACH  1.00000  0.269        0   \n",
       "1324   DDX3X         0.118           NA  REACH  0.99986  0.118        0   \n",
       "1325   DDX3X         0.118  2.375896712  REACH  0.99986  0.118        0   \n",
       "\n",
       "      FDR_DD  FDR_NDD  DEV_BR_GENE_10  \n",
       "0          0        0               1  \n",
       "17         0        0               0  \n",
       "54         0        0               1  \n",
       "208        0        1               1  \n",
       "219        0        1               1  \n",
       "299        0        0               1  \n",
       "313        0        0               0  \n",
       "382        0        0               1  \n",
       "465        1        1               1  \n",
       "501        1        1               1  \n",
       "526        0        0               1  \n",
       "580        1        1               1  \n",
       "596        1        1               1  \n",
       "647        0        0               1  \n",
       "838        1        1               0  \n",
       "857        1        1               1  \n",
       "870        1        1               1  \n",
       "890        1        1               1  \n",
       "896        1        1               1  \n",
       "948        1        1               0  \n",
       "995        0        0               0  \n",
       "1021       0        0               1  \n",
       "1044       0        0               1  \n",
       "1087       1        1               0  \n",
       "1100       0        0               0  \n",
       "1136       0        0               0  \n",
       "1203       0        0               0  \n",
       "1283       0        0               1  \n",
       "1324       0        0               1  \n",
       "1325       0        0               1  \n",
       "\n",
       "[30 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_dn:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(28, 9)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loss_of_function</th>\n",
       "      <th>missense</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99979</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99989</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.168</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.205</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99993</td>\n",
       "      <td>0.324</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.100</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.083</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.309</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.251</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.172</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.92771</td>\n",
       "      <td>0.555</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>948</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.250</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.98587</td>\n",
       "      <td>0.481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1021</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.418</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.210</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1087</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.141</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1100</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.174</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.321</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99996</td>\n",
       "      <td>0.195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.269</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1324</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1325</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99986</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000555</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      loss_of_function  missense      PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  \\\n",
       "0                    0         1  0.99979  0.415        0       0        0   \n",
       "17                   1         0  1.00000  0.271        0       0        0   \n",
       "208                  0         1  1.00000  0.183        0       0        1   \n",
       "219                  0         1  1.00000  0.220        0       0        1   \n",
       "299                  1         0  1.00000  0.366        0       0        0   \n",
       "382                  1         0  0.99989  0.238        0       0        0   \n",
       "465                  1         0  1.00000  0.168        0       1        1   \n",
       "501                  1         0  1.00000  0.205        1       1        1   \n",
       "526                  1         0  0.99993  0.324        0       0        0   \n",
       "580                  1         0  1.00000  0.100        0       1        1   \n",
       "596                  1         0  1.00000  0.083        0       1        1   \n",
       "647                  0         1  0.99994  0.379        0       0        0   \n",
       "838                  0         1  1.00000  0.309        1       1        1   \n",
       "857                  0         1  1.00000  0.251        0       1        1   \n",
       "870                  1         0  1.00000  0.172        1       1        1   \n",
       "890                  0         1  0.92771  0.555        1       1        1   \n",
       "896                  1         0  1.00000  0.163        0       1        1   \n",
       "948                  1         0  1.00000  0.250        1       1        1   \n",
       "995                  0         1  0.98587  0.481        0       0        0   \n",
       "1021                 1         0  0.99994  0.418        0       0        0   \n",
       "1044                 1         0  1.00000  0.210        0       0        0   \n",
       "1087                 0         1  1.00000  0.141        1       1        1   \n",
       "1100                 0         1  1.00000  0.174        0       0        0   \n",
       "1136                 1         0  1.00000  0.321        0       0        0   \n",
       "1203                 1         0  0.99996  0.195        0       0        0   \n",
       "1283                 0         1  1.00000  0.269        0       0        0   \n",
       "1324                 1         0  0.99986  0.118        0       0        0   \n",
       "1325                 0         1  0.99986  0.118        0       0        0   \n",
       "\n",
       "      DEV_BR_GENE_10  samples_all  \n",
       "0                  1  REACH000089  \n",
       "17                 0  REACH000718  \n",
       "208                1  REACH000741  \n",
       "219                1  REACH000774  \n",
       "299                1  REACH000097  \n",
       "382                1  REACH000605  \n",
       "465                1  REACH000529  \n",
       "501                1  REACH000564  \n",
       "526                1  REACH000404  \n",
       "580                1  REACH000578  \n",
       "596                1  REACH000482  \n",
       "647                1  REACH000423  \n",
       "838                0  REACH000747  \n",
       "857                1  REACH000457  \n",
       "870                1  REACH000435  \n",
       "890                1  REACH000519  \n",
       "896                1  REACH000504  \n",
       "948                0  REACH000189  \n",
       "995                0  REACH000482  \n",
       "1021               1  REACH000544  \n",
       "1044               1  REACH000589  \n",
       "1087               0  REACH000664  \n",
       "1100               0  REACH000754  \n",
       "1136               0  REACH000249  \n",
       "1203               0  REACH000349  \n",
       "1283               1  REACH000708  \n",
       "1324               1  REACH000479  \n",
       "1325               1  REACH000555  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "inhLOF SNV/indels\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(153, 21)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " fdr asd: 13\n",
      " fdr dd: 23\n",
      " fdr ndd: 29\n",
      " fdr brain genes: 56\n",
      "df_inh_snv:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start_hg38</th>\n",
       "      <th>end_hg38</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>alternate_allele</th>\n",
       "      <th>consequence</th>\n",
       "      <th>gene</th>\n",
       "      <th>gnomad_af</th>\n",
       "      <th>loeuf</th>\n",
       "      <th>cohort</th>\n",
       "      <th>...</th>\n",
       "      <th>cases</th>\n",
       "      <th>controls</th>\n",
       "      <th>fathers</th>\n",
       "      <th>mothers</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2029751</td>\n",
       "      <td>2029752</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>GABRD</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.245</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000101</td>\n",
       "      <td></td>\n",
       "      <td>REACH000103</td>\n",
       "      <td></td>\n",
       "      <td>0.001654</td>\n",
       "      <td>0.753</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21598040</td>\n",
       "      <td>21598041</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>RAP1GAP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.349</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000141</td>\n",
       "      <td></td>\n",
       "      <td>REACH000143</td>\n",
       "      <td></td>\n",
       "      <td>0.380850</td>\n",
       "      <td>0.549</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29291893</td>\n",
       "      <td>29291895</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>PTPRU</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.338</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000293</td>\n",
       "      <td></td>\n",
       "      <td>REACH000295</td>\n",
       "      <td></td>\n",
       "      <td>0.025440</td>\n",
       "      <td>0.518</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39412333</td>\n",
       "      <td>39412336</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>MACF1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000544,REACH000547</td>\n",
       "      <td></td>\n",
       "      <td>REACH000546</td>\n",
       "      <td></td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39571636</td>\n",
       "      <td>39571637</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>PABPC4</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.232</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>REACH000760</td>\n",
       "      <td>REACH000759</td>\n",
       "      <td></td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10606</th>\n",
       "      <td>chr9</td>\n",
       "      <td>136840512</td>\n",
       "      <td>136840513</td>\n",
       "      <td>G</td>\n",
       "      <td>GGAGCTCTAGGC</td>\n",
       "      <td>stop_gained&amp;frameshift_variant</td>\n",
       "      <td>RABL6</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>0.245</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000145</td>\n",
       "      <td></td>\n",
       "      <td>REACH000148</td>\n",
       "      <td></td>\n",
       "      <td>0.359240</td>\n",
       "      <td>0.570</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10615</th>\n",
       "      <td>chr9</td>\n",
       "      <td>136923873</td>\n",
       "      <td>136923874</td>\n",
       "      <td>C</td>\n",
       "      <td>CTA</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>TRAF2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.123</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>REACH000166,REACH000168</td>\n",
       "      <td></td>\n",
       "      <td>REACH000164</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.215</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10745</th>\n",
       "      <td>chrX</td>\n",
       "      <td>48522252</td>\n",
       "      <td>48522253</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>EBP</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.342</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000586</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000587</td>\n",
       "      <td>0.936450</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10774</th>\n",
       "      <td>chrX</td>\n",
       "      <td>67569002</td>\n",
       "      <td>67569003</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>AR</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.291</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000387</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000388</td>\n",
       "      <td>0.988370</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10819</th>\n",
       "      <td>chrX</td>\n",
       "      <td>100820619</td>\n",
       "      <td>100820620</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>CSTF2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.133</td>\n",
       "      <td>REACH</td>\n",
       "      <td>...</td>\n",
       "      <td>REACH000252</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>REACH000253</td>\n",
       "      <td>0.999780</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  start_hg38   end_hg38 reference_allele alternate_allele  \\\n",
       "2      chr1     2029751    2029752                C                T   \n",
       "140    chr1    21598040   21598041                C                T   \n",
       "189    chr1    29291893   29291895               AC                A   \n",
       "275    chr1    39412333   39412336              CTG                C   \n",
       "281    chr1    39571636   39571637                C                T   \n",
       "...     ...         ...        ...              ...              ...   \n",
       "10606  chr9   136840512  136840513                G     GGAGCTCTAGGC   \n",
       "10615  chr9   136923873  136923874                C              CTA   \n",
       "10745  chrX    48522252   48522253                G                A   \n",
       "10774  chrX    67569002   67569003                A                G   \n",
       "10819  chrX   100820619  100820620                G                T   \n",
       "\n",
       "                          consequence     gene  gnomad_af  loeuf cohort  ...  \\\n",
       "2                         stop_gained    GABRD   0.000008  0.245  REACH  ...   \n",
       "140           splice_acceptor_variant  RAP1GAP   0.000000  0.349  REACH  ...   \n",
       "189                frameshift_variant    PTPRU   0.000012  0.338  REACH  ...   \n",
       "275                frameshift_variant    MACF1   0.000000  0.084  REACH  ...   \n",
       "281                       stop_gained   PABPC4   0.000008  0.232  REACH  ...   \n",
       "...                               ...      ...        ...    ...    ...  ...   \n",
       "10606  stop_gained&frameshift_variant    RABL6   0.000054  0.245  REACH  ...   \n",
       "10615              frameshift_variant    TRAF2   0.000000  0.123  REACH  ...   \n",
       "10745            splice_donor_variant      EBP   0.000000  0.342  REACH  ...   \n",
       "10774                      start_lost       AR   0.000017  0.291  REACH  ...   \n",
       "10819                      start_lost    CSTF2   0.000000  0.133  REACH  ...   \n",
       "\n",
       "                         cases                 controls      fathers  \\\n",
       "2                  REACH000101                           REACH000103   \n",
       "140                REACH000141                           REACH000143   \n",
       "189                REACH000293                           REACH000295   \n",
       "275    REACH000544,REACH000547                           REACH000546   \n",
       "281                                         REACH000760  REACH000759   \n",
       "...                        ...                      ...          ...   \n",
       "10606              REACH000145                           REACH000148   \n",
       "10615                           REACH000166,REACH000168                \n",
       "10745              REACH000586                                         \n",
       "10774              REACH000387                                         \n",
       "10819              REACH000252                                         \n",
       "\n",
       "           mothers       PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  DEV_BR_GENE_10  \n",
       "2                   0.001654  0.753        0       0        0               0  \n",
       "140                 0.380850  0.549        0       0        0               1  \n",
       "189                 0.025440  0.518        0       0        0               0  \n",
       "275                 1.000000  0.126        0       0        0               1  \n",
       "281                 1.000000  0.295        0       0        0               1  \n",
       "...            ...       ...    ...      ...     ...      ...             ...  \n",
       "10606               0.359240  0.570        0       0        0               0  \n",
       "10615  REACH000164  1.000000  0.215        1       0        1               0  \n",
       "10745  REACH000587  0.936450  0.342        0       0        0               1  \n",
       "10774  REACH000388  0.988370  0.291        0       0        0               0  \n",
       "10819  REACH000253  0.999780  0.124        0       0        0               0  \n",
       "\n",
       "[153 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "this_df_inh:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(125, 7)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000544,REACH000547,REACH000546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000760,REACH000759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.131</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000650,REACH000652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.332</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000372,REACH000430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.151</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000668,REACH000670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10597</th>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.598</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000668,REACH000669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10615</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.215</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000166,REACH000168,REACH000164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10745</th>\n",
       "      <td>0.936450</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000586,REACH000587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10774</th>\n",
       "      <td>0.988370</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000387,REACH000388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10819</th>\n",
       "      <td>0.999780</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000252,REACH000253</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>125 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  DEV_BR_GENE_10  \\\n",
       "275    1.000000  0.126        0       0        0               1   \n",
       "281    1.000000  0.295        0       0        0               1   \n",
       "297    1.000000  0.131        0       1        1               1   \n",
       "589    1.000000  0.332        0       0        0               0   \n",
       "688    1.000000  0.151        1       1        1               1   \n",
       "...         ...    ...      ...     ...      ...             ...   \n",
       "10597  0.000007  0.598        0       1        1               0   \n",
       "10615  1.000000  0.215        1       0        1               0   \n",
       "10745  0.936450  0.342        0       0        0               1   \n",
       "10774  0.988370  0.291        0       0        0               0   \n",
       "10819  0.999780  0.124        0       0        0               0   \n",
       "\n",
       "                               samples_all  \n",
       "275    REACH000544,REACH000547,REACH000546  \n",
       "281                REACH000760,REACH000759  \n",
       "297                REACH000650,REACH000652  \n",
       "589                REACH000372,REACH000430  \n",
       "688                REACH000668,REACH000670  \n",
       "...                                    ...  \n",
       "10597              REACH000668,REACH000669  \n",
       "10615  REACH000166,REACH000168,REACH000164  \n",
       "10745              REACH000586,REACH000587  \n",
       "10774              REACH000387,REACH000388  \n",
       "10819              REACH000252,REACH000253  \n",
       "\n",
       "[125 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged denovo and inheritted one row per sample\n",
      "this_df_dn_inh:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loss_of_function</th>\n",
       "      <th>missense</th>\n",
       "      <th>PLI</th>\n",
       "      <th>LOEUF</th>\n",
       "      <th>FDR_ASD</th>\n",
       "      <th>FDR_DD</th>\n",
       "      <th>FDR_NDD</th>\n",
       "      <th>DEV_BR_GENE_10</th>\n",
       "      <th>samples_all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.99979</td>\n",
       "      <td>0.415</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.271</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.183</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.220</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.366</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.93645</td>\n",
       "      <td>0.342</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>REACH000587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98837</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98837</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99978</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99978</td>\n",
       "      <td>0.124</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>REACH000253</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     loss_of_function  missense      PLI  LOEUF  FDR_ASD  FDR_DD  FDR_NDD  \\\n",
       "0                 0.0       1.0  0.99979  0.415        0       0        0   \n",
       "1                 1.0       0.0  1.00000  0.271        0       0        0   \n",
       "2                 0.0       1.0  1.00000  0.183        0       0        1   \n",
       "3                 0.0       1.0  1.00000  0.220        0       0        1   \n",
       "4                 1.0       0.0  1.00000  0.366        0       0        0   \n",
       "..                ...       ...      ...    ...      ...     ...      ...   \n",
       "150               NaN       NaN  0.93645  0.342        0       0        0   \n",
       "151               NaN       NaN  0.98837  0.291        0       0        0   \n",
       "151               NaN       NaN  0.98837  0.291        0       0        0   \n",
       "152               NaN       NaN  0.99978  0.124        0       0        0   \n",
       "152               NaN       NaN  0.99978  0.124        0       0        0   \n",
       "\n",
       "     DEV_BR_GENE_10  samples_all  \n",
       "0                 1  REACH000089  \n",
       "1                 0  REACH000718  \n",
       "2                 1  REACH000741  \n",
       "3                 1  REACH000774  \n",
       "4                 1  REACH000097  \n",
       "..              ...          ...  \n",
       "150               1  REACH000587  \n",
       "151               0  REACH000387  \n",
       "151               0  REACH000388  \n",
       "152               0  REACH000252  \n",
       "152               0  REACH000253  \n",
       "\n",
       "[311 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### previously discovered variants\n",
    "file_fdr = '../data/fdr_pvals_tada.tsv'\n",
    "df_fdr = pd.read_table(file_fdr, sep='\\t', header=0, index_col=None)\n",
    "print('df_fdr:')\n",
    "display(df_fdr)\n",
    "asd_genes = df_fdr.loc[df_fdr.FDR_TADA_ASD <= 0.05].gene_gencodeV33.tolist()\n",
    "dd_genes = df_fdr.loc[df_fdr.FDR_TADA_DD <= 0.05].gene_gencodeV33.tolist()\n",
    "ndd_genes = df_fdr.loc[df_fdr.FDR_TADA_NDD <= 0.05].gene_gencodeV33.tolist()\n",
    "print(f'asd: {len(asd_genes)}')\n",
    "#print(f'asd: {(asd_genes)}')\n",
    "print(f'dd: {len(dd_genes)}')\n",
    "print(f'ndd: {len(ndd_genes)}')\n",
    "\n",
    "file_pli = '../data/gene_max_pli.tsv'\n",
    "df_pli = pd.read_table(file_pli, sep='\\t', header=0, index_col=None).set_index('gene')\n",
    "print('df_pli:')\n",
    "display(df_pli)\n",
    "\n",
    "file_lf = '../data/gene_min_loeuf.tsv'\n",
    "df_lf = pd.read_table(file_lf, sep='\\t', header=0, index_col=None).set_index('gene')\n",
    "print('df_lf:')\n",
    "display(df_lf)\n",
    "\n",
    "print('dnLOF and dnMIS SNV/indels')\n",
    "file_dn_snv = '../data/Antaki2022_supplementary_tables_ST5.tsv'\n",
    "df_dn_snv = pd.read_table(file_dn_snv, sep='\\t', header=0, index_col=None, keep_default_na=False)\n",
    "df_dn_snv['PLI'] = df_dn_snv.gene.map(df_pli['lof.pLI'])\n",
    "df_dn_snv['LOEUF'] = df_dn_snv.gene.map(df_lf['lof.oe_ci.upper'])\n",
    "df_dn_snv['FDR_ASD'] = df_dn_snv.gene.map(lambda x: 1 if x in asd_genes else 0)\n",
    "df_dn_snv['FDR_DD'] = df_dn_snv.gene.map(lambda x: 1 if x in dd_genes else 0)\n",
    "df_dn_snv['FDR_NDD'] = df_dn_snv.gene.map(lambda x: 1 if x in ndd_genes else 0)\n",
    "df_dn_snv['DEV_BR_GENE_10'] = df_dn_snv.gene.map(lambda x: 1 if x in br_flt_10 else 0)\n",
    "df_dn_snv = df_dn_snv.loc[(df_dn_snv.iid.str.contains('REACH'))]\n",
    "display(df_dn_snv.shape)\n",
    "print(f\" fdr asd: {df_dn_snv['FDR_ASD'].sum()}\")\n",
    "print(f\" fdr dd: {df_dn_snv['FDR_DD'].sum()}\")\n",
    "print(f\" fdr ndd: {df_dn_snv['FDR_NDD'].sum()}\")\n",
    "print(f\" fdr brain genes: {df_dn_snv['DEV_BR_GENE_10'].sum()}\")\n",
    "print('df_dn_snv:')\n",
    "display(df_dn_snv)\n",
    "this_df_dn = df_dn_snv[['loss_of_function', 'missense', 'PLI', 'LOEUF', 'FDR_ASD', 'FDR_DD', 'FDR_NDD', 'DEV_BR_GENE_10']].copy()\n",
    "this_df_dn['samples_all'] = df_dn_snv['iid']\n",
    "# filter constraint genes\n",
    "this_df_dn = this_df_dn.loc[((this_df_dn.PLI>=0.9) | (this_df_dn.LOEUF<=0.37) | \n",
    "                             (this_df_dn.FDR_ASD==1) | (this_df_dn.FDR_DD==1) | (this_df_dn.FDR_NDD==1))]\n",
    "print('this_df_dn:')\n",
    "display(this_df_dn.shape)\n",
    "display(this_df_dn)\n",
    "\n",
    "print('inhLOF SNV/indels')\n",
    "file_inh_snv = '../data/Antaki2022_supplementary_tables_ST8.tsv'\n",
    "df_inh_snv = pd.read_table(file_inh_snv, sep='\\t', header=0, index_col=None, keep_default_na=False)\n",
    "df_inh_snv['PLI'] = df_inh_snv.gene.map(df_pli['lof.pLI'])\n",
    "df_inh_snv['LOEUF'] = df_inh_snv.gene.map(df_lf['lof.oe_ci.upper'])\n",
    "df_inh_snv['FDR_ASD'] = df_inh_snv.gene.map(lambda x: 1 if x in asd_genes else 0)\n",
    "df_inh_snv['FDR_DD'] = df_inh_snv.gene.map(lambda x: 1 if x in dd_genes else 0)\n",
    "df_inh_snv['FDR_NDD'] = df_inh_snv.gene.map(lambda x: 1 if x in ndd_genes else 0)\n",
    "df_inh_snv['DEV_BR_GENE_10'] = df_inh_snv.gene.map(lambda x: 1 if x in br_flt_10 else 0)\n",
    "df_inh_snv = df_inh_snv.loc[(df_inh_snv.cases.str.contains('REACH')) | ((df_inh_snv.controls.str.contains('REACH')))]\n",
    "display(df_inh_snv.shape)\n",
    "print(f\" fdr asd: {df_inh_snv['FDR_ASD'].sum()}\")\n",
    "print(f\" fdr dd: {df_inh_snv['FDR_DD'].sum()}\")\n",
    "print(f\" fdr ndd: {df_inh_snv['FDR_NDD'].sum()}\")\n",
    "print(f\" fdr brain genes: {df_inh_snv['DEV_BR_GENE_10'].sum()}\")\n",
    "print('df_inh_snv:')\n",
    "display(df_inh_snv)\n",
    "this_df_inh = df_inh_snv[['PLI', 'LOEUF', 'FDR_ASD', 'FDR_DD', 'FDR_NDD', 'DEV_BR_GENE_10']].copy()\n",
    "this_df_inh['samples_all'] = df_inh_snv.apply(lambda row: ','.join([x for x in [row.cases, row.controls, \n",
    "                                                                         row.fathers, row.mothers] if x != '']), axis=1)\n",
    "# filter constraint genes\n",
    "this_df_inh = this_df_inh.loc[((this_df_inh.PLI>=0.9) | (this_df_inh.LOEUF<=0.37) | \n",
    "                             (this_df_inh.FDR_ASD==1) | (this_df_inh.FDR_DD==1) | (this_df_inh.FDR_NDD==1))]\n",
    "print('this_df_inh:')\n",
    "display(this_df_inh.shape)\n",
    "display(this_df_inh)\n",
    "\n",
    "print('Merged denovo and inheritted one row per sample')\n",
    "this_df_dn_inh = pd.concat([this_df_dn, this_df_inh], axis=0, ignore_index=True)\n",
    "this_df_dn_inh['samples_all'] = this_df_dn_inh.samples_all.str.split(',')\n",
    "this_df_dn_inh = this_df_dn_inh.explode('samples_all')\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "this_df_dn_inh['samples_all'] = this_df_dn_inh.apply(lambda row: rename_samples_dict[row['samples_all']] \n",
    "                                             if row['samples_all'] in rename_samples_dict else row['samples_all'], axis=1)\n",
    "\n",
    "print('this_df_dn_inh:')\n",
    "display(this_df_dn_inh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "df7715f7-460a-4542-ba7f-d323cc59d4ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_pca_flt:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phen</th>\n",
       "      <th>sex</th>\n",
       "      <th>cohort</th>\n",
       "      <th>duo</th>\n",
       "      <th>family</th>\n",
       "      <th>is_eur</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>...</th>\n",
       "      <th>PC2_MAT</th>\n",
       "      <th>PC3_MAT</th>\n",
       "      <th>PC4_MAT</th>\n",
       "      <th>PC5_MAT</th>\n",
       "      <th>PC6_MAT</th>\n",
       "      <th>PC7_MAT</th>\n",
       "      <th>PC8_MAT</th>\n",
       "      <th>PC9_MAT</th>\n",
       "      <th>PC10_MAT</th>\n",
       "      <th>rel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.008627</td>\n",
       "      <td>-0.005607</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.002845</td>\n",
       "      <td>-0.019312</td>\n",
       "      <td>-0.013769</td>\n",
       "      <td>-0.014384</td>\n",
       "      <td>0.007249</td>\n",
       "      <td>0.002096</td>\n",
       "      <td>0.005399</td>\n",
       "      <td>-0.014983</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.095364</td>\n",
       "      <td>-0.045230</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047780</td>\n",
       "      <td>-0.017435</td>\n",
       "      <td>0.153686</td>\n",
       "      <td>0.005292</td>\n",
       "      <td>-0.009487</td>\n",
       "      <td>-0.020773</td>\n",
       "      <td>0.034190</td>\n",
       "      <td>0.015895</td>\n",
       "      <td>-0.011925</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003289</td>\n",
       "      <td>0.003757</td>\n",
       "      <td>-0.004828</td>\n",
       "      <td>0.007650</td>\n",
       "      <td>0.005633</td>\n",
       "      <td>0.032103</td>\n",
       "      <td>0.009510</td>\n",
       "      <td>0.020007</td>\n",
       "      <td>0.016383</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>F0065</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.252774</td>\n",
       "      <td>0.107933</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049799</td>\n",
       "      <td>0.158347</td>\n",
       "      <td>0.011411</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.007690</td>\n",
       "      <td>-0.006605</td>\n",
       "      <td>0.008261</td>\n",
       "      <td>-0.012000</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0.044785</td>\n",
       "      <td>0.038065</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000762</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.027901</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>-0.009075</td>\n",
       "      <td>-0.014136</td>\n",
       "      <td>-0.005499</td>\n",
       "      <td>-0.008555</td>\n",
       "      <td>0.015644</td>\n",
       "      <td>0.004869</td>\n",
       "      <td>0.011004</td>\n",
       "      <td>-0.019684</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000765</td>\n",
       "      <td>CON</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.030809</td>\n",
       "      <td>-0.001867</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>-0.009075</td>\n",
       "      <td>-0.014136</td>\n",
       "      <td>-0.005499</td>\n",
       "      <td>-0.008555</td>\n",
       "      <td>0.015644</td>\n",
       "      <td>0.004869</td>\n",
       "      <td>0.011004</td>\n",
       "      <td>-0.019684</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000766</td>\n",
       "      <td>CON</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.028469</td>\n",
       "      <td>-0.005167</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>-0.009075</td>\n",
       "      <td>-0.014136</td>\n",
       "      <td>-0.005499</td>\n",
       "      <td>-0.008555</td>\n",
       "      <td>0.015644</td>\n",
       "      <td>0.004869</td>\n",
       "      <td>0.011004</td>\n",
       "      <td>-0.019684</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000767</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.087690</td>\n",
       "      <td>-0.012273</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>-0.021358</td>\n",
       "      <td>0.151522</td>\n",
       "      <td>-0.035297</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.005679</td>\n",
       "      <td>0.020179</td>\n",
       "      <td>-0.022316</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000770</td>\n",
       "      <td>CON</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.102736</td>\n",
       "      <td>-0.007714</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>-0.021358</td>\n",
       "      <td>0.151522</td>\n",
       "      <td>-0.035297</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.005679</td>\n",
       "      <td>0.020179</td>\n",
       "      <td>-0.022316</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       fid          iid phen     sex  cohort   duo  family  is_eur       PC1  \\\n",
       "136  F0001  REACH000001  ASD    Male       1  trio       1       1  0.008627   \n",
       "137  F0026  REACH000026  ASD    Male       1  trio       1       0  0.095364   \n",
       "139  F0058  REACH000058  ASD    Male       1  trio       1       1  0.002422   \n",
       "140  F0065  REACH000065  ASD    Male       1  trio       1       0  0.252774   \n",
       "146  F0078  REACH000086  ASD    Male       1  trio       2       0  0.044785   \n",
       "..     ...          ...  ...     ...     ...   ...     ...     ...       ...   \n",
       "408  F0290  REACH000762  ASD    Male       1  trio       1       1 -0.027901   \n",
       "409  F0290  REACH000765  CON  Female       1  trio       1       1 -0.030809   \n",
       "410  F0290  REACH000766  CON  Female       1  trio       1       1 -0.028469   \n",
       "411  F0291  REACH000767  ASD    Male       1  trio       1       0  0.087690   \n",
       "412  F0291  REACH000770  CON  Female       1  trio       1       0  0.102736   \n",
       "\n",
       "          PC2  ...   PC2_MAT   PC3_MAT   PC4_MAT   PC5_MAT   PC6_MAT  \\\n",
       "136 -0.005607  ...  0.000410  0.002845 -0.019312 -0.013769 -0.014384   \n",
       "137 -0.045230  ... -0.047780 -0.017435  0.153686  0.005292 -0.009487   \n",
       "139 -0.009216  ... -0.003289  0.003757 -0.004828  0.007650  0.005633   \n",
       "140  0.107933  ...  0.049799  0.158347  0.011411  0.004728  0.007690   \n",
       "146  0.038065  ...  0.080387  0.132056  0.029583  0.020156  0.007971   \n",
       "..        ...  ...       ...       ...       ...       ...       ...   \n",
       "408  0.000946  ...  0.003052 -0.009075 -0.014136 -0.005499 -0.008555   \n",
       "409 -0.001867  ...  0.003052 -0.009075 -0.014136 -0.005499 -0.008555   \n",
       "410 -0.005167  ...  0.003052 -0.009075 -0.014136 -0.005499 -0.008555   \n",
       "411 -0.012273  ... -0.016917 -0.021358  0.151522 -0.035297  0.008383   \n",
       "412 -0.007714  ... -0.016917 -0.021358  0.151522 -0.035297  0.008383   \n",
       "\n",
       "      PC7_MAT   PC8_MAT   PC9_MAT  PC10_MAT  rel  \n",
       "136  0.007249  0.002096  0.005399 -0.014983    C  \n",
       "137 -0.020773  0.034190  0.015895 -0.011925    C  \n",
       "139  0.032103  0.009510  0.020007  0.016383    C  \n",
       "140 -0.006605  0.008261 -0.012000 -0.009906    C  \n",
       "146 -0.002383  0.014468  0.006253 -0.018303    C  \n",
       "..        ...       ...       ...       ...  ...  \n",
       "408  0.015644  0.004869  0.011004 -0.019684    C  \n",
       "409  0.015644  0.004869  0.011004 -0.019684    C  \n",
       "410  0.015644  0.004869  0.011004 -0.019684    C  \n",
       "411  0.005679  0.020179 -0.022316  0.002755    C  \n",
       "412  0.005679  0.020179 -0.022316  0.002755    C  \n",
       "\n",
       "[117 rows x 39 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_parents:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phen</th>\n",
       "      <th>sex</th>\n",
       "      <th>cohort</th>\n",
       "      <th>duo</th>\n",
       "      <th>is_eur</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>PC6</th>\n",
       "      <th>PC7</th>\n",
       "      <th>PC8</th>\n",
       "      <th>PC9</th>\n",
       "      <th>PC10</th>\n",
       "      <th>rel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000107</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.022403</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.002845</td>\n",
       "      <td>-0.019312</td>\n",
       "      <td>-0.013769</td>\n",
       "      <td>-0.014384</td>\n",
       "      <td>0.007249</td>\n",
       "      <td>0.002096</td>\n",
       "      <td>0.005399</td>\n",
       "      <td>-0.014983</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000226</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>0.024023</td>\n",
       "      <td>-0.013482</td>\n",
       "      <td>-0.003936</td>\n",
       "      <td>0.129597</td>\n",
       "      <td>0.030587</td>\n",
       "      <td>-0.007727</td>\n",
       "      <td>-0.002212</td>\n",
       "      <td>-0.022159</td>\n",
       "      <td>0.005150</td>\n",
       "      <td>-0.000824</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000269</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.100551</td>\n",
       "      <td>-0.047780</td>\n",
       "      <td>-0.017435</td>\n",
       "      <td>0.153686</td>\n",
       "      <td>0.005292</td>\n",
       "      <td>-0.009487</td>\n",
       "      <td>-0.020773</td>\n",
       "      <td>0.034190</td>\n",
       "      <td>0.015895</td>\n",
       "      <td>-0.011925</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000270</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.091671</td>\n",
       "      <td>-0.037847</td>\n",
       "      <td>-0.022134</td>\n",
       "      <td>0.163193</td>\n",
       "      <td>0.007864</td>\n",
       "      <td>-0.006841</td>\n",
       "      <td>-0.023716</td>\n",
       "      <td>0.040800</td>\n",
       "      <td>0.006666</td>\n",
       "      <td>-0.012916</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000439</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.025327</td>\n",
       "      <td>-0.003289</td>\n",
       "      <td>0.003757</td>\n",
       "      <td>-0.004828</td>\n",
       "      <td>0.007650</td>\n",
       "      <td>0.005633</td>\n",
       "      <td>0.032103</td>\n",
       "      <td>0.009510</td>\n",
       "      <td>0.020007</td>\n",
       "      <td>0.016383</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.204041</td>\n",
       "      <td>0.127559</td>\n",
       "      <td>0.190286</td>\n",
       "      <td>0.013553</td>\n",
       "      <td>0.006706</td>\n",
       "      <td>-0.001266</td>\n",
       "      <td>-0.010150</td>\n",
       "      <td>0.006605</td>\n",
       "      <td>-0.006867</td>\n",
       "      <td>0.006560</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.030630</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>-0.009075</td>\n",
       "      <td>-0.014136</td>\n",
       "      <td>-0.005499</td>\n",
       "      <td>-0.008555</td>\n",
       "      <td>0.015644</td>\n",
       "      <td>0.004869</td>\n",
       "      <td>0.011004</td>\n",
       "      <td>-0.019684</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.027327</td>\n",
       "      <td>-0.004878</td>\n",
       "      <td>-0.003881</td>\n",
       "      <td>0.000495</td>\n",
       "      <td>0.003727</td>\n",
       "      <td>-0.001647</td>\n",
       "      <td>-0.008831</td>\n",
       "      <td>0.016730</td>\n",
       "      <td>-0.005315</td>\n",
       "      <td>0.006958</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.091442</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>-0.021358</td>\n",
       "      <td>0.151522</td>\n",
       "      <td>-0.035297</td>\n",
       "      <td>0.008383</td>\n",
       "      <td>0.005679</td>\n",
       "      <td>0.020179</td>\n",
       "      <td>-0.022316</td>\n",
       "      <td>0.002755</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.086006</td>\n",
       "      <td>-0.003008</td>\n",
       "      <td>-0.018691</td>\n",
       "      <td>0.150184</td>\n",
       "      <td>-0.024530</td>\n",
       "      <td>0.006159</td>\n",
       "      <td>-0.002022</td>\n",
       "      <td>0.016434</td>\n",
       "      <td>-0.002075</td>\n",
       "      <td>-0.013056</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      fid          iid phen     sex  cohort duo  is_eur       PC1       PC2  \\\n",
       "0   F0001  REACH000107    .  Female       1   .       1 -0.022403  0.000410   \n",
       "0   F0001  REACH000226    .    Male       1   .       1  0.024023 -0.013482   \n",
       "0   F0026  REACH000269    .  Female       1   .       0  0.100551 -0.047780   \n",
       "0   F0026  REACH000270    .    Male       1   .       0  0.091671 -0.037847   \n",
       "0   F0058  REACH000439    .  Female       1   .       1 -0.025327 -0.003289   \n",
       "..    ...          ...  ...     ...     ...  ..     ...       ...       ...   \n",
       "0   F0288  REACH000756    .    Male       1   .       0  0.204041  0.127559   \n",
       "0   F0290  REACH000763    .  Female       1   .       1 -0.030630  0.003052   \n",
       "0   F0290  REACH000764    .    Male       1   .       1 -0.027327 -0.004878   \n",
       "0   F0291  REACH000768    .  Female       1   .       0  0.091442 -0.016917   \n",
       "0   F0291  REACH000769    .    Male       1   .       0  0.086006 -0.003008   \n",
       "\n",
       "         PC3       PC4       PC5       PC6       PC7       PC8       PC9  \\\n",
       "0   0.002845 -0.019312 -0.013769 -0.014384  0.007249  0.002096  0.005399   \n",
       "0  -0.003936  0.129597  0.030587 -0.007727 -0.002212 -0.022159  0.005150   \n",
       "0  -0.017435  0.153686  0.005292 -0.009487 -0.020773  0.034190  0.015895   \n",
       "0  -0.022134  0.163193  0.007864 -0.006841 -0.023716  0.040800  0.006666   \n",
       "0   0.003757 -0.004828  0.007650  0.005633  0.032103  0.009510  0.020007   \n",
       "..       ...       ...       ...       ...       ...       ...       ...   \n",
       "0   0.190286  0.013553  0.006706 -0.001266 -0.010150  0.006605 -0.006867   \n",
       "0  -0.009075 -0.014136 -0.005499 -0.008555  0.015644  0.004869  0.011004   \n",
       "0  -0.003881  0.000495  0.003727 -0.001647 -0.008831  0.016730 -0.005315   \n",
       "0  -0.021358  0.151522 -0.035297  0.008383  0.005679  0.020179 -0.022316   \n",
       "0  -0.018691  0.150184 -0.024530  0.006159 -0.002022  0.016434 -0.002075   \n",
       "\n",
       "        PC10 rel  \n",
       "0  -0.014983   P  \n",
       "0  -0.000824   P  \n",
       "0  -0.011925   P  \n",
       "0  -0.012916   P  \n",
       "0   0.016383   P  \n",
       "..       ...  ..  \n",
       "0   0.006560   P  \n",
       "0  -0.019684   P  \n",
       "0   0.006958   P  \n",
       "0   0.002755   P  \n",
       "0  -0.013056   P  \n",
       "\n",
       "[126 rows x 18 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_pca_flt:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fid</th>\n",
       "      <th>iid</th>\n",
       "      <th>phen</th>\n",
       "      <th>sex</th>\n",
       "      <th>cohort</th>\n",
       "      <th>duo</th>\n",
       "      <th>family</th>\n",
       "      <th>is_eur</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>...</th>\n",
       "      <th>PC2_MAT</th>\n",
       "      <th>PC3_MAT</th>\n",
       "      <th>PC4_MAT</th>\n",
       "      <th>PC5_MAT</th>\n",
       "      <th>PC6_MAT</th>\n",
       "      <th>PC7_MAT</th>\n",
       "      <th>PC8_MAT</th>\n",
       "      <th>PC9_MAT</th>\n",
       "      <th>PC10_MAT</th>\n",
       "      <th>rel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>F0001</td>\n",
       "      <td>REACH000001</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.008627</td>\n",
       "      <td>-0.005607</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.002845</td>\n",
       "      <td>-0.019312</td>\n",
       "      <td>-0.013769</td>\n",
       "      <td>-0.014384</td>\n",
       "      <td>0.007249</td>\n",
       "      <td>0.002096</td>\n",
       "      <td>0.005399</td>\n",
       "      <td>-0.014983</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>F0026</td>\n",
       "      <td>REACH000026</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.095364</td>\n",
       "      <td>-0.045230</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047780</td>\n",
       "      <td>-0.017435</td>\n",
       "      <td>0.153686</td>\n",
       "      <td>0.005292</td>\n",
       "      <td>-0.009487</td>\n",
       "      <td>-0.020773</td>\n",
       "      <td>0.034190</td>\n",
       "      <td>0.015895</td>\n",
       "      <td>-0.011925</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>F0058</td>\n",
       "      <td>REACH000058</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002422</td>\n",
       "      <td>-0.009216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.003289</td>\n",
       "      <td>0.003757</td>\n",
       "      <td>-0.004828</td>\n",
       "      <td>0.007650</td>\n",
       "      <td>0.005633</td>\n",
       "      <td>0.032103</td>\n",
       "      <td>0.009510</td>\n",
       "      <td>0.020007</td>\n",
       "      <td>0.016383</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>F0065</td>\n",
       "      <td>REACH000065</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.252774</td>\n",
       "      <td>0.107933</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049799</td>\n",
       "      <td>0.158347</td>\n",
       "      <td>0.011411</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.007690</td>\n",
       "      <td>-0.006605</td>\n",
       "      <td>0.008261</td>\n",
       "      <td>-0.012000</td>\n",
       "      <td>-0.009906</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>F0078</td>\n",
       "      <td>REACH000086</td>\n",
       "      <td>ASD</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>trio</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.044785</td>\n",
       "      <td>0.038065</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080387</td>\n",
       "      <td>0.132056</td>\n",
       "      <td>0.029583</td>\n",
       "      <td>0.020156</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>-0.002383</td>\n",
       "      <td>0.014468</td>\n",
       "      <td>0.006253</td>\n",
       "      <td>-0.018303</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0288</td>\n",
       "      <td>REACH000756</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.204041</td>\n",
       "      <td>0.127559</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000763</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.030630</td>\n",
       "      <td>0.003052</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0290</td>\n",
       "      <td>REACH000764</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.027327</td>\n",
       "      <td>-0.004878</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000768</td>\n",
       "      <td>.</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.091442</td>\n",
       "      <td>-0.016917</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F0291</td>\n",
       "      <td>REACH000769</td>\n",
       "      <td>.</td>\n",
       "      <td>Male</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0.086006</td>\n",
       "      <td>-0.003008</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       fid          iid phen     sex  cohort   duo  family  is_eur       PC1  \\\n",
       "136  F0001  REACH000001  ASD    Male       1  trio     1.0       1  0.008627   \n",
       "137  F0026  REACH000026  ASD    Male       1  trio     1.0       0  0.095364   \n",
       "139  F0058  REACH000058  ASD    Male       1  trio     1.0       1  0.002422   \n",
       "140  F0065  REACH000065  ASD    Male       1  trio     1.0       0  0.252774   \n",
       "146  F0078  REACH000086  ASD    Male       1  trio     2.0       0  0.044785   \n",
       "..     ...          ...  ...     ...     ...   ...     ...     ...       ...   \n",
       "0    F0288  REACH000756    .    Male       1     .     NaN       0  0.204041   \n",
       "0    F0290  REACH000763    .  Female       1     .     NaN       1 -0.030630   \n",
       "0    F0290  REACH000764    .    Male       1     .     NaN       1 -0.027327   \n",
       "0    F0291  REACH000768    .  Female       1     .     NaN       0  0.091442   \n",
       "0    F0291  REACH000769    .    Male       1     .     NaN       0  0.086006   \n",
       "\n",
       "          PC2  ...   PC2_MAT   PC3_MAT   PC4_MAT   PC5_MAT   PC6_MAT  \\\n",
       "136 -0.005607  ...  0.000410  0.002845 -0.019312 -0.013769 -0.014384   \n",
       "137 -0.045230  ... -0.047780 -0.017435  0.153686  0.005292 -0.009487   \n",
       "139 -0.009216  ... -0.003289  0.003757 -0.004828  0.007650  0.005633   \n",
       "140  0.107933  ...  0.049799  0.158347  0.011411  0.004728  0.007690   \n",
       "146  0.038065  ...  0.080387  0.132056  0.029583  0.020156  0.007971   \n",
       "..        ...  ...       ...       ...       ...       ...       ...   \n",
       "0    0.127559  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0    0.003052  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.004878  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.016917  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "0   -0.003008  ...       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "      PC7_MAT   PC8_MAT   PC9_MAT  PC10_MAT  rel  \n",
       "136  0.007249  0.002096  0.005399 -0.014983    C  \n",
       "137 -0.020773  0.034190  0.015895 -0.011925    C  \n",
       "139  0.032103  0.009510  0.020007  0.016383    C  \n",
       "140 -0.006605  0.008261 -0.012000 -0.009906    C  \n",
       "146 -0.002383  0.014468  0.006253 -0.018303    C  \n",
       "..        ...       ...       ...       ...  ...  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "0         NaN       NaN       NaN       NaN    P  \n",
       "\n",
       "[243 rows x 39 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# samples with SNV vars: 73\n"
     ]
    }
   ],
   "source": [
    "file_pca = '../data/master_phen_4.SBayesR.20210915.csv'\n",
    "\n",
    "df_pca = pd.read_table(file_pca, sep=',', header=0)\n",
    "cols = ['fid', 'iid', 'phen','sex','cohort','duo','family','is_eur','PC1','PC2','PC3','PC4','PC5','PC6','PC7','PC8','PC9','PC10', \n",
    "        'PC1_PAT','PC2_PAT','PC3_PAT','PC4_PAT','PC5_PAT','PC6_PAT','PC7_PAT','PC8_PAT','PC9_PAT','PC10_PAT', \n",
    "        'PC1_MAT','PC2_MAT','PC3_MAT','PC4_MAT','PC5_MAT','PC6_MAT','PC7_MAT','PC8_MAT','PC9_MAT','PC10_MAT']\n",
    "df_pca = df_pca[cols]\n",
    "\n",
    "rename_samples_dict = {'REACH000236': 'REACH000236_PB', \n",
    "                       'REACH000436': 'REACH000436_PB', \n",
    "                       'REACH000530': 'REACH000530_PB', \n",
    "                       'REACH000531': 'REACH000531_ONT', \n",
    "                       'REACH000532': 'REACH000532_ONT'}\n",
    "df_pca['iid'] = df_pca.apply(lambda row: rename_samples_dict[row['iid']] if row['iid'] in rename_samples_dict else row['iid'], axis=1)\n",
    "#display(df_pca)\n",
    "\n",
    "# first filter for the kids, then add parents to them\n",
    "df_pca_flt = df_pca.loc[df_pca.iid.isin(df_trios.sample_id)].copy()\n",
    "df_pca_flt['rel'] = 'C'\n",
    "print('df_pca_flt:')\n",
    "display(df_pca_flt)\n",
    "\n",
    "df_parents = pd.DataFrame()\n",
    "for index, row in df_pca_flt.iterrows():\n",
    "    sample = row['iid']\n",
    "    fid = row['fid']\n",
    "    phen = row['phen']\n",
    "    is_eur = row['is_eur']\n",
    "    pc_mat = [row[f'PC{x}_MAT'] for x in range(1,11)]\n",
    "    pc_pat = [row[f'PC{x}_PAT'] for x in range(1,11)]\n",
    "    dad = sample_dad_dict[sample]\n",
    "    mom = sample_mom_dict[sample]\n",
    "    \n",
    "    dict_mat = {'fid': [fid], 'iid': [mom], 'phen': ['.'], 'sex': ['Female'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_mat.update({f'PC{x}': [pc_mat[x-1]] for x in range(1,11)})\n",
    "    df_mat = pd.DataFrame(dict_mat)\n",
    "\n",
    "    dict_pat = {'fid': [fid], 'iid': [dad], 'phen': ['.'], 'sex': ['Male'], 'cohort': [1], 'duo': ['.'], 'is_eur': [is_eur]}\n",
    "    dict_pat.update({f'PC{x}': [pc_pat[x-1]] for x in range(1,11)})\n",
    "    df_pat = pd.DataFrame(dict_pat)\n",
    "    \n",
    "    if df_parents.shape[0] == 0 or mom not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_mat])\n",
    "    if df_parents.shape[0] == 0 or dad not in df_parents.iid.tolist():\n",
    "        df_parents = pd.concat([df_parents, df_pat])\n",
    "\n",
    "df_parents['rel'] = 'P'\n",
    "print('df_parents:')\n",
    "display(df_parents)\n",
    "\n",
    "df_pca_flt = pd.concat([df_pca_flt, df_parents])\n",
    "print('df_pca_flt:')\n",
    "display(df_pca_flt)\n",
    "\n",
    "df_pca_flt = pd.merge(df_pca_flt, df_meta[['Sample_ID', 'Affected', 'Affected_ASD']], how='inner', left_on='iid', right_on='Sample_ID')\n",
    "df_pca_flt = pd.merge(df_pca_flt, df_cov[['SAMPLE', 'MEAN_COVERAGE']], how='inner', left_on='iid', right_on='SAMPLE')\n",
    "\n",
    "df_pca_flt['Phenotype'] = df_pca_flt.apply(lambda row: 1 if row.Affected == 'Yes' else 0, axis=1)\n",
    "df_pca_flt['Phenotype_ASD'] = df_pca_flt.apply(lambda row: 1 if row.Affected_ASD == 'Yes' else 0, axis=1)\n",
    "df_pca_flt['Platform'] = df_pca_flt.apply(lambda row: get_plat(row.iid), axis=1)\n",
    "#df_pca_flt.to_csv('investigate2.tsv', sep='\\t', index=None)\n",
    "\n",
    "df_pca_flt['SNV_CAR'] = df_pca_flt.apply(lambda row: 1 if row['iid'] in this_df_dn_inh.samples_all.tolist() else 0, axis=1)\n",
    "print(f\"# samples with SNV vars: {df_pca_flt['SNV_CAR'].sum()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b5815866-0917-49a1-a1fe-4a81bedef82a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/scratch/smmortazavi/job_40601570/ipykernel_3544310/1777979188.py:2: DtypeWarning: Columns (25,65,66,70,72,73,77,78,79,80,81,82,83,84,88,89,90,91,92,93,94,95,99,100,101,102,103,104,105,106,110,111,112,113,114,115,116,117,121,122,123,124,125,126,127,128) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_table(file_name, sep='\\t', header=0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GNOCCHI_MAX_4...\n",
      "GNOCCHI_MAX_4 subsets...\n",
      "GNOCCHI_MAX_3...\n",
      "GNOCCHI_MAX_3 subsets...\n",
      "GNOCCHI_MAX_2...\n",
      "GNOCCHI_MAX_2 subsets...\n",
      "GNOCCHI_MAX_1...\n",
      "GNOCCHI_MAX_1 subsets...\n",
      "GNOCCHI_MEAN_4...\n",
      "GNOCCHI_MEAN_4 subsets...\n",
      "GNOCCHI_MEAN_3...\n",
      "GNOCCHI_MEAN_3 subsets...\n",
      "GNOCCHI_MEAN_2...\n",
      "GNOCCHI_MEAN_2 subsets...\n",
      "GNOCCHI_MEAN_1...\n",
      "GNOCCHI_MEAN_1 subsets...\n",
      "frequence calc...\n",
      "   AD2_SAMPLES\n",
      "   AD3_SAMPLES\n",
      "   AD4_SAMPLES\n",
      "   AD5_SAMPLES\n",
      "   SQ5_SAMPLES\n",
      "   SQ10_SAMPLES\n",
      "   SQ20_SAMPLES\n",
      "   SQ30_SAMPLES\n",
      "   SQ40_SAMPLES\n",
      "   SQ50_SAMPLES\n",
      "   SQ60_SAMPLES\n",
      "   SQ70_SAMPLES\n",
      "count the number of samples in filter columns...\n",
      "ZERO_COV_SAMPLES\n",
      "SQ5_SAMPLES\n",
      "SQ10_SAMPLES\n",
      "SQ20_SAMPLES\n",
      "SQ30_SAMPLES\n",
      "SQ40_SAMPLES\n",
      "SQ50_SAMPLES\n",
      "SQ60_SAMPLES\n",
      "SQ70_SAMPLES\n",
      "AD2_SAMPLES\n",
      "AD3_SAMPLES\n",
      "AD4_SAMPLES\n",
      "AD5_SAMPLES\n",
      "HET_SAMPLES\n",
      "HOMALT_SAMPLES\n",
      "IL_SAMPLES\n",
      "IL_GT_SAMPLES\n",
      "SV2_GT_SAMPLES\n",
      "count number of case/control...\n",
      "IL samples names to LR sample names...\n",
      "generate alternative PLATFOMR from cross validation with IL calls...\n",
      "count number of transmissions...\n",
      "   AD2_SAMPLES\n",
      "   AD3_SAMPLES\n",
      "   AD4_SAMPLES\n",
      "   AD5_SAMPLES\n",
      "   SQ5_SAMPLES\n",
      "   SQ10_SAMPLES\n",
      "   SQ20_SAMPLES\n",
      "   SQ30_SAMPLES\n",
      "   SQ40_SAMPLES\n",
      "   SQ50_SAMPLES\n",
      "   SQ60_SAMPLES\n",
      "   SQ70_SAMPLES\n",
      "biotype: lncRNA\n",
      "biotype: miRNA\n",
      "biotype: misc_RNA\n",
      "biotype: processed_transcript\n",
      "biotype: snoRNA\n",
      "biotype: snRNA\n",
      "biotype: TEC\n",
      "biotype: protein_coding\n",
      "protein coding subsets...\n",
      "pli...\n",
      "pli >0.9...\n",
      "low pli...\n",
      "loeuf...\n",
      "ASD...\n",
      "DD...\n",
      "NDD...\n",
      "intergenic...\n",
      "genic...\n",
      "genic subsets...\n",
      "pli subsets...\n",
      "pli >0.9 subsets...\n",
      "low pli subsets...\n",
      "loeuf subsets...\n",
      "ASD subsets...\n",
      "DD subsets...\n",
      "NDD subsets...\n",
      "LoF...\n",
      "LoF constrained subsets...\n",
      "FB PR...\n",
      "FB ENH...\n",
      "FANTOM...\n",
      "protein_coding CDS subsets...\n",
      "brain span genes...\n",
      "dev brain genes subsets...\n",
      "LoF dev brain...\n",
      "dev brain and PLI...\n",
      "writing the output...\n",
      "++++++++++++++++++++++++++++++++++++++++\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>END</th>\n",
       "      <th>ID</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>PLATFORM</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>SRC</th>\n",
       "      <th>GENCODE</th>\n",
       "      <th>denovo_LR</th>\n",
       "      <th>...</th>\n",
       "      <th>X_LOF_DEV_BR_10_cds</th>\n",
       "      <th>X_LOF_DEV_BR_20_cds</th>\n",
       "      <th>X_LOF_DEV_BR_30_cds</th>\n",
       "      <th>X_LOF_DEV_BR_40_cds</th>\n",
       "      <th>X_LOF_DEV_BR_50_cds</th>\n",
       "      <th>X_PLIp9_LOF_DEV_BR_10_cds</th>\n",
       "      <th>X_PLIp9_DEV_BR_GENE_10</th>\n",
       "      <th>X_PLIp9_DEV_BR_GENE_10_cds</th>\n",
       "      <th>X_PLIp9_DEV_BR_GENE_10_utr</th>\n",
       "      <th>X_PLIp9_DEV_BR_GENE_10_intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>180998</td>\n",
       "      <td>14346589</td>\n",
       "      <td>DUP</td>\n",
       "      <td>LR</td>\n",
       "      <td>170998</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10468</td>\n",
       "      <td>180108</td>\n",
       "      <td>Sniffles2.DEL.1DM0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>LR</td>\n",
       "      <td>-169640</td>\n",
       "      <td>Sniffles_Lumpy</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>14848</td>\n",
       "      <td>182483</td>\n",
       "      <td>14346592</td>\n",
       "      <td>DUP</td>\n",
       "      <td>LR</td>\n",
       "      <td>167635</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32024</td>\n",
       "      <td>32024</td>\n",
       "      <td>Sniffles2.INS.1M0</td>\n",
       "      <td>INS</td>\n",
       "      <td>LR</td>\n",
       "      <td>64</td>\n",
       "      <td>Sniffles</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35141</td>\n",
       "      <td>35141</td>\n",
       "      <td>948107_1</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>24860</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>exon</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344741</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56887898</td>\n",
       "      <td>56887898</td>\n",
       "      <td>6390796_2</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>.</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344742</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56887898</td>\n",
       "      <td>56887898</td>\n",
       "      <td>6390795_2</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>.</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344743</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56887898</td>\n",
       "      <td>56887898</td>\n",
       "      <td>10218686_2</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>.</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344744</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56887902</td>\n",
       "      <td>56887902</td>\n",
       "      <td>13088295_2</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>.</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344745</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56887902</td>\n",
       "      <td>56887902</td>\n",
       "      <td>13088296_2</td>\n",
       "      <td>BND</td>\n",
       "      <td>LR</td>\n",
       "      <td>.</td>\n",
       "      <td>Lumpy</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>344746 rows × 434 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS       END                  ID SVTYPE PLATFORM    SVLEN  \\\n",
       "0       chr1     10000    180998            14346589    DUP       LR   170998   \n",
       "1       chr1     10468    180108  Sniffles2.DEL.1DM0    DEL       LR  -169640   \n",
       "2       chr1     14848    182483            14346592    DUP       LR   167635   \n",
       "3       chr1     32024     32024   Sniffles2.INS.1M0    INS       LR       64   \n",
       "4       chr1     35141     35141            948107_1    BND       LR    24860   \n",
       "...      ...       ...       ...                 ...    ...      ...      ...   \n",
       "344741  chrY  56887898  56887898           6390796_2    BND       LR        .   \n",
       "344742  chrY  56887898  56887898           6390795_2    BND       LR        .   \n",
       "344743  chrY  56887898  56887898          10218686_2    BND       LR        .   \n",
       "344744  chrY  56887902  56887902          13088295_2    BND       LR        .   \n",
       "344745  chrY  56887902  56887902          13088296_2    BND       LR        .   \n",
       "\n",
       "                   SRC      GENCODE denovo_LR  ... X_LOF_DEV_BR_10_cds  \\\n",
       "0                Lumpy  start_codon         .  ...                   0   \n",
       "1       Sniffles_Lumpy  start_codon         .  ...                   0   \n",
       "2                Lumpy  start_codon         .  ...                   0   \n",
       "3             Sniffles            .         .  ...                   0   \n",
       "4                Lumpy         exon         .  ...                   0   \n",
       "...                ...          ...       ...  ...                 ...   \n",
       "344741           Lumpy            .         .  ...                   0   \n",
       "344742           Lumpy            .         .  ...                   0   \n",
       "344743           Lumpy            .         .  ...                   0   \n",
       "344744           Lumpy            .         .  ...                   0   \n",
       "344745           Lumpy            .         .  ...                   0   \n",
       "\n",
       "       X_LOF_DEV_BR_20_cds X_LOF_DEV_BR_30_cds X_LOF_DEV_BR_40_cds  \\\n",
       "0                        0                   0                   0   \n",
       "1                        0                   0                   0   \n",
       "2                        0                   0                   0   \n",
       "3                        0                   0                   0   \n",
       "4                        0                   0                   0   \n",
       "...                    ...                 ...                 ...   \n",
       "344741                   0                   0                   0   \n",
       "344742                   0                   0                   0   \n",
       "344743                   0                   0                   0   \n",
       "344744                   0                   0                   0   \n",
       "344745                   0                   0                   0   \n",
       "\n",
       "       X_LOF_DEV_BR_50_cds X_PLIp9_LOF_DEV_BR_10_cds X_PLIp9_DEV_BR_GENE_10  \\\n",
       "0                        0                         0                      0   \n",
       "1                        0                         0                      0   \n",
       "2                        0                         0                      0   \n",
       "3                        0                         0                      0   \n",
       "4                        0                         0                      0   \n",
       "...                    ...                       ...                    ...   \n",
       "344741                   0                         0                      0   \n",
       "344742                   0                         0                      0   \n",
       "344743                   0                         0                      0   \n",
       "344744                   0                         0                      0   \n",
       "344745                   0                         0                      0   \n",
       "\n",
       "       X_PLIp9_DEV_BR_GENE_10_cds X_PLIp9_DEV_BR_GENE_10_utr  \\\n",
       "0                               0                          0   \n",
       "1                               0                          0   \n",
       "2                               0                          0   \n",
       "3                               0                          0   \n",
       "4                               0                          0   \n",
       "...                           ...                        ...   \n",
       "344741                          0                          0   \n",
       "344742                          0                          0   \n",
       "344743                          0                          0   \n",
       "344744                          0                          0   \n",
       "344745                          0                          0   \n",
       "\n",
       "       X_PLIp9_DEV_BR_GENE_10_intron  \n",
       "0                                  0  \n",
       "1                                  0  \n",
       "2                                  0  \n",
       "3                                  0  \n",
       "4                                  0  \n",
       "...                              ...  \n",
       "344741                             0  \n",
       "344742                             0  \n",
       "344743                             0  \n",
       "344744                             0  \n",
       "344745                             0  \n",
       "\n",
       "[344746 rows x 434 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### # generate the big table\n",
    "### if this file is not present in folder: make_raw_data, you should generate with code in that folder\n",
    "### generates the output in folder: outputs\n",
    "file_name = 'make_raw_data/combined_table.tsv'\n",
    "df = get_big_table(file_name)\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c5844426-4c91-4320-8e62-9abff8cfdc79",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_df_flt(df, flt_name, svtypes, ext_col=None, flt_sd_art=True, freq_col=None, freq_thr=None):\n",
    "    df_flt = df.loc[((df[f'NUM_{flt_name}'] > 0) | (df['NUM_IL_SAMPLES'] > 0) | (df['NUM_IL_GT_SAMPLES'] > 0) | (df['NUM_SV2_GT_SAMPLES'] > 0)) & (df.SVTYPE.isin(svtypes))]\n",
    "    #| (df['NUM_SV2_GT_SAMPLES'] > 0)\n",
    "    if flt_sd_art:\n",
    "        df_flt = df_flt.loc[~((df_flt.SD_ART==1) & (df_flt.PLATFORM=='LR'))]\n",
    "    if freq_thr != None:\n",
    "        df_flt = df_flt.loc[df_flt[f'freq_{freq_col}']<=freq_thr]\n",
    "    return df_flt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "69ae09c3-edb4-4441-82cf-7f55a5215efb",
   "metadata": {},
   "outputs": [],
   "source": [
    "biotypes = ['lncRNA' ,'miRNA' ,'misc_RNA' ,'processed_transcript' ,'snoRNA' ,'snRNA' ,'TEC', 'protein_coding']\n",
    "pc_subsec = ['protein_coding_cds', 'protein_coding_utr', 'protein_coding_intron', \n",
    "             'protein_coding_cds_pli', 'protein_coding_cds_loeuf', \n",
    "             'protein_coding_cds_fdr_asd', 'protein_coding_cds_fdr_dd', 'protein_coding_cds_fdr_ndd']\n",
    "consts_1 = ['PLI', 'LOEUF', 'FDR_ASD', 'FDR_DD', 'FDR_NDD', 'intergenic', 'genic']\n",
    "consts_2 = ['FB_PR', 'FB_ENH', 'FANTOM']\n",
    "consts_3 = ['GNOCCHI_MAX_1', 'GNOCCHI_MAX_2', 'GNOCCHI_MAX_3', 'GNOCCHI_MAX_4', \n",
    "            'GNOCCHI_MEAN_1', 'GNOCCHI_MEAN_2', 'GNOCCHI_MEAN_3', 'GNOCCHI_MEAN_4']\n",
    "consts_3_subsec = ['GNOCCHI_MAX_4_cds', 'GNOCCHI_MAX_4_utr', 'GNOCCHI_MAX_4_intron', \n",
    "                   'GNOCCHI_MAX_3_cds', 'GNOCCHI_MAX_3_utr', 'GNOCCHI_MAX_3_intron', \n",
    "                   'GNOCCHI_MAX_2_cds', 'GNOCCHI_MAX_2_utr', 'GNOCCHI_MAX_2_intron']\n",
    "consts_4 = ['EV_CONS_EL_BP_1', 'EV_CONS_EL_BP_50', 'EV_CONS_EL_BP_100', \n",
    "            'EV_CONS_EL_FRAC_2', 'EV_CONS_EL_FRAC_5', 'EV_CONS_EL_FRAC_10', 'S_HET_1', 'S_HET_10', 'S_HET_100']\n",
    "consts_4_subsec1 = ['EV_CONS_EL_BP_1_cds', 'EV_CONS_EL_BP_1_utr', 'EV_CONS_EL_BP_1_ncd', \n",
    "                   'EV_CONS_EL_BP_50_cds', 'EV_CONS_EL_BP_50_utr', 'EV_CONS_EL_BP_50_ncd', \n",
    "                   'EV_CONS_EL_BP_100_cds', 'EV_CONS_EL_BP_100_utr', 'EV_CONS_EL_BP_100_ncd',\n",
    "                   'S_HET_1_cds', 'S_HET_1_utr', 'S_HET_1_intron', \n",
    "                    'S_HET_10_cds', 'S_HET_10_utr', 'S_HET_10_intron', \n",
    "                    'S_HET_100_cds', 'S_HET_100_utr', 'S_HET_100_intron']\n",
    "consts_4_subsec2 = ['EV_CONS_EL_FRAC_2_cds', 'EV_CONS_EL_FRAC_2_utr', 'EV_CONS_EL_FRAC_2_ncd', \n",
    "                   'EV_CONS_EL_FRAC_5_cds', 'EV_CONS_EL_FRAC_5_utr', 'EV_CONS_EL_FRAC_5_ncd', \n",
    "                   'EV_CONS_EL_FRAC_10_cds', 'EV_CONS_EL_FRAC_10_utr', 'EV_CONS_EL_FRAC_10_ncd']\n",
    "consts_subsec_1 = ['PLI_cds', 'PLI_utr', 'PLI_intron', \n",
    "                   'LOEUF_cds', 'LOEUF_utr', 'LOEUF_intron', \n",
    "                   'genic_cds', 'genic_utr', 'genic_intron']\n",
    "consts_subsec_2 = ['FDR_ASD_cds', 'FDR_ASD_utr', 'FDR_ASD_intron', \n",
    "                   'FDR_DD_cds', 'FDR_DD_utr', 'FDR_DD_intron', \n",
    "                   'FDR_NDD_cds', 'FDR_NDD_utr', 'FDR_NDD_intron']\n",
    "consts_subsec_3 = ['PLIp9_cds', 'PLIp9_utr', 'PLIp9_intron',\n",
    "                   'LOWPLI_cds', 'LOWPLI_utr', 'LOWPLI_intron']\n",
    "dev_br_10 = ['DEV_BR_GENE_10', 'DEV_BR_GENE_10_cds', 'DEV_BR_GENE_10_utr', 'DEV_BR_GENE_10_intron']\n",
    "dev_br_20 = ['DEV_BR_GENE_20', 'DEV_BR_GENE_20_cds', 'DEV_BR_GENE_20_utr', 'DEV_BR_GENE_20_intron']\n",
    "dev_br_30 = ['DEV_BR_GENE_30', 'DEV_BR_GENE_30_cds', 'DEV_BR_GENE_30_utr', 'DEV_BR_GENE_30_intron']\n",
    "dev_br_40 = ['DEV_BR_GENE_40', 'DEV_BR_GENE_40_cds', 'DEV_BR_GENE_40_utr', 'DEV_BR_GENE_40_intron']\n",
    "dev_br_50 = ['DEV_BR_GENE_50', 'DEV_BR_GENE_50_cds', 'DEV_BR_GENE_50_utr', 'DEV_BR_GENE_50_intron']\n",
    "lof_dev_br = ['LOF_DEV_BR_50_cds', 'LOF_DEV_BR_40_cds', 'LOF_DEV_BR_30_cds', 'LOF_DEV_BR_20_cds', 'LOF_DEV_BR_10_cds']\n",
    "lof_plus_consts = ['LOF_DEV_BR_10_cds', 'LOF_PLI_cds', 'PLI_cds', 'PLI_utr', 'PLI_intron',\n",
    "                   'LOF_LOEUF_cds', 'LOEUF_cds', 'LOEUF_utr', 'LOEUF_intron']\n",
    "lof_plus_consts2 = ['LOF_DEV_BR_10_cds', 'LOF_PLI_cds', 'PLI_cds', 'PLI_utr', 'PLI_intron',\n",
    "                   'LOF_LOWPLI_cds',  'LOWPLI_cds', 'LOWPLI_utr', 'LOWPLI_intron']\n",
    "lof_plus_consts3 = ['LOF_PLIp9_cds', 'PLIp9_cds', 'PLIp9_utr', 'PLIp9_intron']\n",
    "lof_plus_consts4 = ['LOF_FDR_ASD_cds', 'LOF_FDR_DD_cds', 'LOF_FDR_NDD_cds']\n",
    "lof_plus_consts5 = ['PLIp9_LOF_DEV_BR_10_cds', \n",
    "                    'PLIp9_DEV_BR_GENE_10', \n",
    "                    'PLIp9_DEV_BR_GENE_10_cds', 'PLIp9_DEV_BR_GENE_10_utr', 'PLIp9_DEV_BR_GENE_10_intron']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cf43c442-4fc8-48eb-8c5c-73b59db91438",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SQ20_SAMPLES\n",
      "total: (344746, 434)\n",
      "filtered: (27279, 434)\n",
      "writing outputs//main_df_toR_freq05_LRIL_SQ20_SAMPLES_AD2_SAMPLES_DELINS_GW_plat.csv\n",
      "++++++++++++++++++++++++++++++++++++++++\n"
     ]
    }
   ],
   "source": [
    "### write counts for each sample for R script\n",
    "# with ins/dup and del separated\n",
    "# add genome wide burden with ins/dup and del stratification\n",
    "\n",
    "fts = (consts_1 + consts_2 + consts_3 + consts_3_subsec + consts_subsec_1 + consts_subsec_2 + \n",
    "       dev_br_10 + dev_br_20 + dev_br_30 + dev_br_40 + dev_br_50 + lof_dev_br + \n",
    "       lof_plus_consts + lof_plus_consts2 + lof_plus_consts3 + lof_plus_consts4 + lof_plus_consts5)\n",
    "\n",
    "#flt_names = ['AD5_SAMPLES', 'AD4_SAMPLES', 'AD3_SAMPLES', \n",
    "#             'SQ20_SAMPLES', 'SQ30_SAMPLES', 'SQ40_SAMPLES', 'SQ50_SAMPLES', 'SQ60_SAMPLES', 'SQ70_SAMPLES']\n",
    "flt_names = ['SQ20_SAMPLES']\n",
    "\n",
    "include_svtypes = ['INS', 'DEL', 'DUP', 'INV', '.']\n",
    "include_svtypes_del = ['DEL']\n",
    "include_svtypes_ins = ['INS', 'DUP', '.']\n",
    "\n",
    "samples_col = 'AD2_SAMPLES'\n",
    "\n",
    "freq_col = 'AD2_SAMPLES'\n",
    "\n",
    "for flt_name in flt_names:\n",
    "    print(flt_name)\n",
    "    # filter for main svtypes and quality\n",
    "    df_flt = get_df_flt(df, flt_name, include_svtypes, flt_sd_art=True, freq_col=freq_col, freq_thr=0.05)\n",
    "    print(f'total: {df.shape}')\n",
    "    print(f'filtered: {df_flt.shape}')\n",
    "    df_main = pd.DataFrame(df_pca_flt)\n",
    "    ############## compute genomewide burden ##############\n",
    "    ### using IL genotypes for LR_IL calls\n",
    "    temp = (df_flt.loc[df_flt.PLATFORM_CV=='LR'][samples_col].str.split(',').tolist() +\n",
    "            df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['IL_GT_SAMPLES_LR'].str.split(',').tolist() + \n",
    "            df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['IL_SAMPLES_LR'].str.split(',').tolist() + \n",
    "            df_flt.loc[df_flt.PLATFORM_CV=='IL']['IL_SAMPLES_LR'].str.split(',').tolist() +\n",
    "            df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['SV2_GT_SAMPLES'].str.split(',').tolist())\n",
    "    ### filtering LR-only calls all together\n",
    "    temp_del = df_flt.loc[df_flt.SVTYPE.isin(include_svtypes_del)][samples_col].str.split(',').tolist() + df_flt.loc[df_flt.SVTYPE.isin(include_svtypes_del)]['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "    temp_ins = df_flt.loc[df_flt.SVTYPE.isin(include_svtypes_ins)][samples_col].str.split(',').tolist() + df_flt.loc[df_flt.SVTYPE.isin(include_svtypes_ins)]['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "    temp_lr = df_flt.loc[df_flt.PLATFORM_CV=='LR'][samples_col].str.split(',').tolist()\n",
    "    temp_lr_il_lrgt = df_flt.loc[df_flt.PLATFORM_CV=='LR_IL'][samples_col].str.split(',').tolist()\n",
    "    ### IL_GT_SAMPLES_LR: for calls coming from LR_IL, IL_SAMPLES_LR: for calls coming from IL, they shouldn't be both present\n",
    "    temp_lr_il_ilgt = (df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['IL_GT_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                       df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['IL_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                       df_flt.loc[df_flt.PLATFORM_CV=='LR_IL']['SV2_GT_SAMPLES'].str.split(',').tolist())\n",
    "    temp_il = df_flt.loc[df_flt.PLATFORM_CV=='IL']['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "    this_samples_list = []\n",
    "    this_samples_del_list = []\n",
    "    this_samples_ins_list = []\n",
    "    this_samples_lr_list = []\n",
    "    this_samples_lr_il_lrgt_list = []\n",
    "    this_samples_lr_il_ilgt_list = []\n",
    "    this_samples_il_list = []\n",
    "    for x in temp:\n",
    "        this_samples_list.extend(x)\n",
    "    for x in temp_del:\n",
    "        this_samples_del_list.extend(x)\n",
    "    for x in temp_ins:\n",
    "        this_samples_ins_list.extend(x)\n",
    "    for x in temp_lr:\n",
    "        this_samples_lr_list.extend(x)\n",
    "    for x in temp_lr_il_lrgt:\n",
    "        this_samples_lr_il_lrgt_list.extend(x)\n",
    "    for x in temp_lr_il_ilgt:\n",
    "        this_samples_lr_il_ilgt_list.extend(x)\n",
    "    for x in temp_il:\n",
    "        this_samples_il_list.extend(x)\n",
    "    count_col = []\n",
    "    count_col_del = []\n",
    "    count_col_ins = []\n",
    "    count_col_lr = []\n",
    "    count_col_lr_il_lrgt = []\n",
    "    count_col_lr_il_ilgt = []\n",
    "    count_col_il = []\n",
    "    for sample in df_pca_flt.iid.tolist():\n",
    "        count_col.append(this_samples_list.count(sample))\n",
    "        count_col_del.append(this_samples_del_list.count(sample))\n",
    "        count_col_ins.append(this_samples_ins_list.count(sample))\n",
    "        count_col_lr.append(this_samples_lr_list.count(sample))\n",
    "        count_col_lr_il_lrgt.append(this_samples_lr_il_lrgt_list.count(sample))\n",
    "        count_col_lr_il_ilgt.append(this_samples_lr_il_ilgt_list.count(sample))\n",
    "        count_col_il.append(this_samples_il_list.count(sample))\n",
    "    df_main[f'count_genomewide'] = count_col\n",
    "    df_main[f'count_genomewide_del'] = count_col_del\n",
    "    df_main[f'count_genomewide_ins'] = count_col_ins\n",
    "    df_main[f'count_genomewide_lr'] = count_col_lr\n",
    "    df_main[f'count_genomewide_lr_il_lrgt'] = count_col_lr_il_lrgt\n",
    "    df_main[f'count_genomewide_lr_il_ilgt'] = count_col_lr_il_ilgt\n",
    "    df_main[f'count_genomewide_il'] = count_col_il\n",
    "    #######################################################\n",
    "    for ft in fts:\n",
    "        #print(ft)\n",
    "        this_df = df_flt.loc[df_flt[f'X_{ft}'] == 1]\n",
    "        ### using IL genotypes for LR_IL calls\n",
    "        temp = (this_df.loc[this_df.PLATFORM_CV=='LR'][samples_col].str.split(',').tolist() +\n",
    "                this_df.loc[this_df.PLATFORM_CV=='LR_IL']['IL_GT_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                this_df.loc[this_df.PLATFORM_CV=='LR_IL']['IL_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                this_df.loc[this_df.PLATFORM_CV=='IL']['IL_SAMPLES_LR'].str.split(',').tolist() +\n",
    "                this_df.loc[this_df.PLATFORM_CV=='LR_IL']['SV2_GT_SAMPLES'].str.split(',').tolist())\n",
    "        ### filtering LR-only calls all together\n",
    "        temp_del = this_df.loc[this_df.SVTYPE.isin(include_svtypes_del)][samples_col].str.split(',').tolist() + this_df.loc[this_df.SVTYPE.isin(include_svtypes_del)]['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "        temp_ins = this_df.loc[this_df.SVTYPE.isin(include_svtypes_ins)][samples_col].str.split(',').tolist() + this_df.loc[this_df.SVTYPE.isin(include_svtypes_ins)]['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "        temp_lr = this_df.loc[this_df.PLATFORM_CV=='LR'][samples_col].str.split(',').tolist()\n",
    "        temp_lr_il_lrgt = this_df.loc[this_df.PLATFORM_CV=='LR_IL'][samples_col].str.split(',').tolist()\n",
    "        ### IL_GT_SAMPLES_LR: for calls coming from LR_IL, IL_SAMPLES_LR: for calls coming from IL, they shouldn't be both present\n",
    "        temp_lr_il_ilgt = (this_df.loc[this_df.PLATFORM_CV=='LR_IL']['IL_GT_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                           this_df.loc[this_df.PLATFORM_CV=='LR_IL']['IL_SAMPLES_LR'].str.split(',').tolist() + \n",
    "                           this_df.loc[this_df.PLATFORM_CV=='LR_IL']['SV2_GT_SAMPLES'].str.split(',').tolist())\n",
    "        temp_il = this_df.loc[this_df.PLATFORM_CV=='IL']['IL_SAMPLES_LR'].str.split(',').tolist()\n",
    "        this_samples_list = []\n",
    "        this_samples_del_list = []\n",
    "        this_samples_ins_list = []\n",
    "        this_samples_lr_list = []\n",
    "        this_samples_lr_il_lrgt_list = []\n",
    "        this_samples_lr_il_ilgt_list = []\n",
    "        this_samples_il_list = []\n",
    "        for x in temp:\n",
    "            this_samples_list.extend(x)\n",
    "        for x in temp_del:\n",
    "            this_samples_del_list.extend(x)\n",
    "        for x in temp_ins:\n",
    "            this_samples_ins_list.extend(x)\n",
    "        for x in temp_lr:\n",
    "            this_samples_lr_list.extend(x)\n",
    "        for x in temp_lr_il_lrgt:\n",
    "            this_samples_lr_il_lrgt_list.extend(x)\n",
    "        for x in temp_lr_il_ilgt:\n",
    "            this_samples_lr_il_ilgt_list.extend(x)\n",
    "        for x in temp_il:\n",
    "            this_samples_il_list.extend(x)\n",
    "        count_col = []\n",
    "        count_col_del = []\n",
    "        count_col_ins = []\n",
    "        count_col_lr = []\n",
    "        count_col_lr_il_lrgt = []\n",
    "        count_col_lr_il_ilgt = []\n",
    "        count_col_il = []\n",
    "        for sample in df_pca_flt.iid.tolist():\n",
    "            count_col.append(this_samples_list.count(sample))\n",
    "            count_col_del.append(this_samples_del_list.count(sample))\n",
    "            count_col_ins.append(this_samples_ins_list.count(sample))\n",
    "            count_col_lr.append(this_samples_lr_list.count(sample))\n",
    "            count_col_lr_il_lrgt.append(this_samples_lr_il_lrgt_list.count(sample))\n",
    "            count_col_lr_il_ilgt.append(this_samples_lr_il_ilgt_list.count(sample))\n",
    "            count_col_il.append(this_samples_il_list.count(sample))\n",
    "        df_main[f'count_{ft}'] = count_col\n",
    "        df_main[f'count_{ft}_del'] = count_col_del\n",
    "        df_main[f'count_{ft}_ins'] = count_col_ins\n",
    "        df_main[f'count_{ft}_lr'] = count_col_lr\n",
    "        df_main[f'count_{ft}_lr_il_lrgt'] = count_col_lr_il_lrgt\n",
    "        df_main[f'count_{ft}_lr_il_ilgt'] = count_col_lr_il_ilgt\n",
    "        df_main[f'count_{ft}_il'] = count_col_il\n",
    "\n",
    "    #display(df_main)\n",
    "    file_out = f'{dir_out}/main_df_toR_freq05_LRIL_{flt_name}_{samples_col}_DELINS_GW_plat.csv'\n",
    "    print(f'writing {file_out}')\n",
    "    df_main.to_csv(file_out, sep=',', header=True, index=False)\n",
    "print('+'*40)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
